--- gmp/mpn/arm/udiv.asm.orig	2013-08-11 17:38:14.542690389 +0200
+++ gmp/mpn/arm/udiv.asm	2013-08-11 17:38:18.078690240 +0200
@@ -50,7 +50,7 @@
 	teq	r12, #0
 	bne	L(oop)
 
-	str	n1, [ rem_ptr ]		C store remainder
+	str	n1, [rem_ptr]		C store remainder
 	adc	r0, n0, n0		C quotient: add last carry from divstep
 	mov	pc, lr
 
@@ -89,7 +89,7 @@
 	addcs	n0, n0, #1		C adjust quotient
 
 L(_even_divisor):
-	str	n1, [ rem_ptr ]		C store remainder
+	str	n1, [rem_ptr]		C store remainder
 	mov	r0, n0			C quotient
 	ldmfd	sp!, { r8, pc }
 EPILOGUE(mpn_udiv_qrnnd)
