--- gmp-4.2.1/gmp-h.in.orig	2013-03-26 14:53:42.132617397 +0100
+++ gmp-4.2.1/gmp-h.in	2013-03-26 14:53:47.424617630 +0100
@@ -420,8 +420,7 @@
    for an inline too, so as to correctly specify "dllimport" on windows, in
    case the function is called rather than inlined.  */
 #ifdef __GNUC__
-#define __GMP_EXTERN_INLINE      extern __inline__
-#define __GMP_INLINE_PROTOTYPES  1
+#define __GMP_EXTERN_INLINE      static __inline__
 #endif
 
 /* DEC C (eg. version 5.9) supports "static __inline foo()", even in -std1
