--- gmp/longlong.h	2006-03-14 10:57:54.000000000 -0500
+++ gmp/longlong.h	2013-08-12 12:03:50.080931954 -0400
@@ -1011,7 +1011,19 @@
 #endif /* __m88000__ */
 
 #if defined (__mips) && W_TYPE_SIZE == 32
-#if __GNUC__ > 2 || __GNUC_MINOR__ >= 7
+
+#define GCC_VERSION (__GNUC__ * 10000 \
+                   + __GNUC_MINOR__ * 100 \
+                   + __GNUC_PATCHLEVEL__)
+
+#if GCC_VERSION >= 40400
+#define umul_ppmm(w1, w0, u, v) \
+    do { \
+    uint64_t res = u*v; \
+    w1 = res >> 32; \
+    w0 = res; \
+    } while(0);
+#elif GCC_VERSION >= 20700
 #define umul_ppmm(w1, w0, u, v) \
   __asm__ ("multu %2,%3" : "=l" (w0), "=h" (w1) : "d" (u), "d" (v))
 #else
