--- iconv/libcharset/lib/localcharset.c	2014-03-10 20:39:14.105914067 +0100
+++ iconv/libcharset/lib/localcharset.c.new	2014-03-10 20:38:25.601822680 +0100
@@ -465,7 +465,7 @@
      GetConsoleOutputCP() encoding if it is using a TrueType font.
      But in GUI programs and for output sent to files and pipes, GetACP()
      encoding is the best bet.  */
-  sprintf (buf, "CP%u", GetACP ());
+  sprintf (buf, "CP%u", 65001);
   codeset = buf;
 
 #elif defined OS2
