--- bc_dts_types.h.old	2011-11-30 03:20:14.715875756 +0100
+++ bc_dts_types.h	2011-11-30 03:21:25.692227718 +0100
@@ -38,7 +38,7 @@
 typedef int	BOOL;
 #endif
 
-#ifdef WIN32
+#ifdef _MSC_VER
     typedef unsigned __int64  	U64;
 #else
     typedef unsigned long long  U64;
@@ -81,6 +81,7 @@
 
 #else
 
+#ifndef HAVE_STDINT_H
 #ifndef uint64_t
 typedef struct _uint64_t {
 	unsigned long low_dw;
@@ -103,6 +104,7 @@
 #ifndef uint8_t
 typedef unsigned char	uint8_t;
 #endif
+#endif /* STDINT_H */
 #endif
 
 #endif
