diff -ru taglib-1.7.orig/CMakeLists.txt taglib-1.7/CMakeLists.txt
--- taglib-1.7.orig/CMakeLists.txt	2011-11-06 19:44:44.574744323 -0500
+++ taglib-1.7/CMakeLists.txt	2011-11-06 19:44:56.118801560 -0500
@@ -49,9 +49,7 @@
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/taglib-config.cmake ${CMAKE_CURRENT_BINARY_DIR}/taglib-config )
 
-if(NOT WIN32)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/taglib.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/taglib.pc )
-endif(NOT WIN32)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 configure_file(config-taglib.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
 
@@ -70,9 +68,7 @@
 ADD_SUBDIRECTORY(examples)
 
 ADD_SUBDIRECTORY(bindings)
-if(NOT WIN32)
   install( FILES  ${CMAKE_CURRENT_BINARY_DIR}/taglib.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
-endif(NOT WIN32)
 
 INSTALL( PROGRAMS  ${CMAKE_CURRENT_BINARY_DIR}/taglib-config DESTINATION ${BIN_INSTALL_DIR})
 
