From b713ca0fb7272b5a22d86ad0e823dbc40cc22a59 Mon Sep 17 00:00:00 2001
From: Ronnie Sahlberg <ronniesahlberg@gmail.com>
Date: Mon, 16 Dec 2024 18:49:59 +1000
Subject: [PATCH 1/5] cant have win32.h referenced from a header we install

Signed-off-by: Ronnie Sahlberg <ronniesahlberg@gmail.com>
---
 include/nfsc/libnfs-raw.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/nfsc/libnfs-raw.h b/include/nfsc/libnfs-raw.h
index 80d31f4..0bd88a2 100644
--- a/include/nfsc/libnfs-raw.h
+++ b/include/nfsc/libnfs-raw.h
@@ -39,10 +39,6 @@
 #include "ps3_compat.h"
 #endif
 
-#ifdef WIN32
-#include <win32/win32_compat.h>
-#endif
-
 #include <stdint.h>
 
 #if defined(HAVE_SYS_UIO_H) || (defined(__APPLE__) && defined(__MACH__))
-- 
2.45.0.windows.1

