diff -ru nfs/Makefile.am nfs-fixed/Makefile.am
--- nfs/Makefile.am	2015-08-02 19:18:25.000000000 +0200
+++ nfs-fixed/Makefile.am	2016-01-14 14:33:06.000000000 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = doc mount nfs nlm nsm portmap rquota lib include utils . $(MAYBE_EXAMPLES)
+SUBDIRS = doc mount nfs nlm nsm portmap rquota lib include $(MAYBE_UTILS) . $(MAYBE_EXAMPLES)
 ACLOCAL_AMFLAGS = -I m4
 
 pkgconfigdir = $(libdir)/pkgconfig
diff -ru nfs/configure.ac nfs-fixed/configure.ac
--- nfs/configure.ac	2015-08-02 19:18:25.000000000 +0200
+++ nfs-fixed/configure.ac	2016-01-14 14:43:43.000000000 +0100
@@ -24,6 +24,14 @@
 # We always want 64 bit file offsets
 CFLAGS="${CFLAGS} -D_FILE_OFFSET_BITS=64"
 
+#option: utils
+AC_ARG_ENABLE([utils],
+              [AC_HELP_STRING([--enable-utils],
+                              [Build util programs])],
+	      [MAYBE_UTILS=""],
+	      [MAYBE_UTILS="utils"])
+AC_SUBST(MAYBE_UTILS)
+
 #option: examples
 AC_ARG_ENABLE([examples],
               [AC_HELP_STRING([--enable-examples],
