--- libxml2-2.9.14/Makefile.am.orig	2022-05-02 12:52:41.000000000 +0200
+++ libxml2-2.9.14/Makefile.am	2022-05-26 19:38:23.251503583 +0200
@@ -2,7 +2,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include . doc example fuzz xstc $(PYTHON_SUBDIR)
+SUBDIRS = include . fuzz xstc $(PYTHON_SUBDIR)
 
 DIST_SUBDIRS = include . doc example fuzz python xstc
 
@@ -15,7 +15,7 @@
                testReader testapi testModule runtest runsuite testchar \
 	       testdict runxmlconf testrecurse testlimits
 
-bin_PROGRAMS = xmllint xmlcatalog
+bin_PROGRAMS =
 
 bin_SCRIPTS=xml2-config
 
