--- libxml2/Makefile.am.orig	2011-11-26 20:53:51.147060747 -0500
+++ libxml2/Makefile.am	2011-11-26 20:52:36.013140366 -0500
@@ -8,10 +8,10 @@
 
 INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
 
-noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
-                testThreads testC14N testAutomata testRegexp \
-                testReader testapi testModule runtest runsuite testchar \
-		testdict runxmlconf testrecurse
+#noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
+#                testThreads testC14N testAutomata testRegexp \
+#                testReader testapi testModule runtest runsuite testchar \
+#		testdict runxmlconf testrecurse
 
 bin_PROGRAMS = xmllint xmlcatalog
 
