--- libxml2-2.9.0/Makefile.am.orig	2012-11-01 18:55:04.214524833 +0100
+++ libxml2-2.9.0/Makefile.am	2012-11-01 18:55:56.838557494 +0100
@@ -10,10 +10,10 @@
 
 AM_CFLAGS = $(THREAD_CFLAGS) $(Z_CFLAGS) $(LZMA_CFLAGS)
 
-noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
-                testThreads testC14N testAutomata testRegexp \
-                testReader testapi testModule runtest runsuite testchar \
-		testdict runxmlconf testrecurse testlimits
+# noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
+#                 testThreads testC14N testAutomata testRegexp \
+#                 testReader testapi testModule runtest runsuite testchar \
+# 		testdict runxmlconf testrecurse testlimits
 
 bin_PROGRAMS = xmllint xmlcatalog
 
