--- libxml2/configure.ac.orig	2018-03-05 16:54:00.000000000 +0100
+++ libxml2/configure.ac	2018-03-07 17:29:50.093410103 +0100
@@ -1036,7 +1036,7 @@
        *mingw*) if test "$with_threads" != "pthread" && test "$with_threads" != "no"; then
                WITH_THREADS="1"
                THREADS_W32="1"
-               THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_WIN32_THREADS"
+               THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_WIN32_THREADS -DHAVE_COMPILER_TLS"
            fi
        ;;
     esac
