diff -urN pthreads-orig/context.h pthreads/context.h
--- pthreads-orig/context.h	2011-03-04 01:37:20.000000000 +0200
+++ pthreads/context.h	2018-03-18 13:50:07.362676429 +0200
@@ -67,6 +67,10 @@
 #define PTW32_PROGCTR(Context)  ((Context).Pc)
 #endif
 
+#if defined(_ARM64_) || defined(__aarch64__)
+#define PTW32_PROGCTR(Context)  ((Context).Pc)
+#endif
+
 #if !defined(PTW32_PROGCTR)
 #error Module contains CPU-specific code; modify and recompile.
 #endif
