diff -urN pthreads-orig/ptw32_spinlock_check_need_init.c pthreads/ptw32_spinlock_check_need_init.c
--- pthreads-orig/ptw32_spinlock_check_need_init.c	2011-03-10 13:40:17.000000000 +0000
+++ pthreads/ptw32_spinlock_check_need_init.c	2018-01-06 07:22:18.399696662 +0000
@@ -38,7 +38,7 @@
 #include "implement.h"
 
 
-INLINE int
+int
 ptw32_spinlock_check_need_init (pthread_spinlock_t * lock)
 {
   int result = 0;
