From b159a783ac11cb6c2c8ec035deaeb3d8a73e0e2f Mon Sep 17 00:00:00 2001
From: Francois Cartegnie <fcvlcdev@free.fr>
Date: Mon, 9 Nov 2020 20:54:14 +0100
Subject: [PATCH] core: ifdef MSG_TRUNC nixes fix

---
 srtcore/channel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/srtcore/channel.cpp b/srtcore/channel.cpp
index 3473e75..3af852e 100644
--- a/srtcore/channel.cpp
+++ b/srtcore/channel.cpp
@@ -801,6 +801,7 @@ EReadStatus CChannel::recvfrom(sockaddr_any& w_addr, CPacket& w_packet) const
         goto Return_error;
     }
 
+#ifndef _WIN32
     // Fix for an issue with Linux Kernel found during tests at Tencent.
     //
     // There was a bug in older Linux Kernel which caused that when the internal
@@ -825,6 +826,7 @@ EReadStatus CChannel::recvfrom(sockaddr_any& w_addr, CPacket& w_packet) const
         status = RST_AGAIN;
         goto Return_error;
     }
+#endif
 
     w_packet.setLength(recv_size - CPacket::HDR_SIZE);
 
-- 
2.26.2

