--- srt/common/srt_compat.h_old	2018-12-20 02:07:53.000000000 +0100
+++ srt/common/srt_compat.h	2018-12-20 02:09:23.000000000 +0100
@@ -82,7 +82,7 @@
 {
 #if defined(__SRT_OSX_CLOCK_GETTIME_AVAILABILITY) \
    && (__SRT_OSX_CLOCK_GETTIME_AVAILABILITY == 1)
-   if (&clock_gettime != NULL)
+   if (__builtin_available(macOS 10.12, iOS 10.0, tvOS 10.0, *))
    {
       return clock_gettime(clock_id, ts);
    }
@@ -95,7 +95,7 @@
 {
 #if defined(__SRT_OSX_CLOCK_GETTIME_AVAILABILITY) \
    && (__SRT_OSX_CLOCK_GETTIME_AVAILABILITY == 1)
-   if (&clock_getres != NULL)
+   if (__builtin_available(macOS 10.12, iOS 10.0, tvOS 10.0, *))
    {
       return clock_getres(clock_id, ts);
    }
