--- srt-1.2.3/CMakeLists.txt	2018-05-16 15:11:24.490534700 +0200
+++ srt-1.2.3/CMakeLists.txt.no-pthread	2018-05-16 15:12:19.940696200 +0200
@@ -423,13 +423,20 @@ endif()
 # obtained by `pkg-config --libs`.
 
 message("Adding the following implicit link libraries: ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES}")
-foreach(LIB ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES})
+foreach(LIB ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES} ${PLATFORM_LIBS})
     if(IS_ABSOLUTE ${LIB} AND EXISTS ${LIB})
-	    set(SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ${LIB})
+        list(APPEND PLIBLIST "${LIB}")
     else()
-        set(SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} "-l${LIB}")
+        list(APPEND PLIBLIST "-l${LIB}")
     endif()
 endforeach()
+if(PLIBLIST)
+    # blacklist of libraries that should not be in Libs.private
+    list(REMOVE_ITEM PLIBLIST "-lc" "-lpthread")
+    string(REPLACE ";" " " SRT_LIBS_PRIVATE "${PLIBLIST}")
+else()
+    set(SRT_LIBS_PRIVATE "")
+endif(PLIBLIST)
 
 join_arguments(SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE})
 
