From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Mon, 29 Jun 2026 11:48:00 +0000
Subject: [PATCH] w32-utils: fix init utils syntax

---
 src/w32-utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/w32-utils.c b/src/w32-utils.c
index 692a384..2eaa781 100644
--- a/src/w32-utils.c
+++ b/src/w32-utils.c
@@ -159,6 +159,6 @@ _gpgrt_w32__init_utils
 void
 _gpgrt_w32__init_utils (void)
 {
-  w32_utils_init ().
+  w32_utils_init ();
 }
 #endif
-- 
2.45.1.windows.1
