diff -ur libgpg-error.orig/src/Makefile.in libgpg-error/src/Makefile.in
--- libgpg-error.orig/src/Makefile.in	2012-06-25 22:14:30.745691001 +0200
+++ libgpg-error/src/Makefile.in	2012-06-25 22:15:16.897691001 +0200
@@ -848,7 +848,7 @@
 # It is correct to use $(CPP).  We want the host's idea of the error codes.
 mkerrcodes.h: Makefile mkerrcodes.awk $(gpg_extra_headers)
 	$(AWK) -f $(srcdir)/mkerrcodes1.awk $(srcdir)/errnos.in >_$@
-	$(CPP) $(extra_cppflags) _$@ | grep GPG_ERR_ | \
+	$(CPP) $(CPPFLAGS) $(extra_cppflags) _$@ | grep GPG_ERR_ | \
                $(AWK) -f $(srcdir)/mkerrcodes.awk >$@
 	-rm _$@
 
