From 8e1e760bef6aa94e6e71000ead2bc21716a06d19 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 23 Oct 2025 12:18:54 +0200
Subject: [PATCH 06/14] don't use GetFileSize on UWP

---
 src/w32-gettext.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/w32-gettext.c b/src/w32-gettext.c
index 42aa01c..bf085cd 100644
--- a/src/w32-gettext.c
+++ b/src/w32-gettext.c
@@ -1233,7 +1233,15 @@ load_domain (const char *filename)
   if (fh == INVALID_HANDLE_VALUE)
     return NULL;
 
+#if !WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP)
+  FILE_STANDARD_INFO fStdInfo;
+  if (GetFileInformationByHandleEx(fh, FileStandardInfo, (LPVOID)&fStdInfo, sizeof(fStdInfo)))
+    size = fStdInfo.EndOfFile.LowPart;
+  else
+    size = INVALID_FILE_SIZE;
+#else
   size = GetFileSize (fh, NULL);
+#endif
   if (size == INVALID_FILE_SIZE)
     {
       CloseHandle (fh);
-- 
2.45.1.windows.1

