--- libgpg-error/src/gpgrt-int.h.orig	2014-09-17 15:23:22.992333706 +0200
+++ libgpg-error/src/gpgrt-int.h	2014-09-17 15:23:52.772783925 +0200
@@ -105,9 +105,9 @@
 
 int _gpgrt_fflush (gpgrt_stream_t stream);
 int _gpgrt_fseek (gpgrt_stream_t stream, long int offset, int whence);
-int _gpgrt_fseeko (gpgrt_stream_t stream, off_t offset, int whence);
+int _gpgrt_fseeko (gpgrt_stream_t stream, gpgrt_off_t offset, int whence);
 long int _gpgrt_ftell (gpgrt_stream_t stream);
-off_t _gpgrt_ftello (gpgrt_stream_t stream);
+gpgrt_off_t _gpgrt_ftello (gpgrt_stream_t stream);
 void _gpgrt_rewind (gpgrt_stream_t stream);
 
 int _gpgrt_fgetc (gpgrt_stream_t stream);
