--- a/loader/CMakeLists.txt	2018-05-16 14:05:33.046991100 +0300
+++ b/loader/CMakeLists.txt	2018-05-16 14:07:39.464013100 +0300
@@ -86,7 +86,7 @@
 set(ASM_FAILURE_MSG "${ASM_FAILURE_MSG} extensions will suffer from a corrupted stack.")
 if(WIN32)
     enable_language(ASM_MASM)
-    if(CMAKE_ASM_MASM_COMPILER_WORKS)
+    if (MSVC AND CMAKE_ASM_MASM_COMPILER_WORKS)
         if(NOT CMAKE_CL_64)
             set(CMAKE_ASM_MASM_FLAGS ${CMAKE_ASM_MASM_FLAGS} /safeseh)
         endif()
--- a/loader/loader.c	2018-05-16 14:03:52.750814700 +0300
+++ b/loader/loader.c	2018-05-16 14:04:08.631642600 +0300
@@ -672,6 +672,10 @@
 //
 // *reg_data contains a string list of filenames as pointer.
 // When done using the returned string list, the caller should free the pointer.
+#ifdef __MINGW32__
+#define CM_GETIDLIST_FILTER_PRESENT            0x00000100
+#define CM_GETIDLIST_FILTER_CLASS              0x00000200
+#endif
 VkResult loaderGetDeviceRegistryFiles(const struct loader_instance *inst, char **reg_data, PDWORD reg_data_size,
                                       LPCSTR value_name) {
     static const wchar_t *softwareComponentGUID = L"{5c4c3332-344d-483c-8739-259e934c9cc8}";
--- a/loader/loader.rc
+++ b/loader/loader.rc
@@ -43,8 +43,6 @@
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////
 
-#include "winres.h"
-
 #define VER_FILE_VERSION            VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH, VERSION_BUILDNO
 
 #define STRINGIZE2(s) #s
@@ -77,7 +75,6 @@
 #endif
 
  FILEOS 0x00000L
- FILETYPE VFT_DLL
  FILESUBTYPE 0x0L
 BEGIN
     BLOCK "StringFileInfo"
