From e9e54996189115fd4868cc7be55e2390419e0c57 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sat, 22 Jan 2022 09:51:26 +0100
Subject: [PATCH 2/3] disable suffix in static lib

---
 loader/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 0552c5bdd..98da79ab8 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -337,7 +337,7 @@ if(PKG_CONFIG_FOUND)
     foreach(LIB ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES} ${PLATFORM_LIBS})
         set(PRIVATE_LIBS "${PRIVATE_LIBS} -l${LIB}")
     endforeach()
-    if(WIN32)
+    if(MSVC)
         set(VULKAN_LIB_SUFFIX "-1")
         # Set libdir path as in cmake's FindVulkan.cmake
         # https://github.com/KhronosGroup/Vulkan-Loader/issues/668
-- 
2.45.0.windows.1

