--- fluidsynth-1.1.6/fluidsynth.pc.in.orig	2015-04-14 15:06:24.867271101 +0200
+++ fluidsynth-1.1.6/fluidsynth.pc.in	2015-04-14 15:07:23.088291727 +0200
@@ -6,5 +6,5 @@
 Name: FluidSynth
 Description: Software SoundFont synth
 Version: @VERSION@
-Libs: -L${libdir} -lfluidsynth -lgthread-2.0 -lglib-2.0 -lm
+Libs: -L${libdir} -lfluidsynth -lgthread-2.0 -lglib-2.0 -ldsound -lwinmm -lm
 Cflags: -I${includedir}
