--- trunk/configure.ac	2015/09/06 15:59:04	3820
+++ trunk/configure.ac	2015/09/07 07:59:55	3827
@@ -93,6 +93,8 @@
 AM_PROG_CC_C_O
 AC_PROG_CPP
 AC_PROG_INSTALL
+AC_CHECK_LIBM
+AC_SUBST(LIBM)
 dnl "Checking for egrep is broken after removal of libltdl stuff... checks use $EGREP, so searching it here."
 AC_PROG_EGREP
 AC_C_CONST
--- mpg123/libmpg123.pc.in.orig	2015-08-12 01:33:47.000000000 -0400
+++ mpg123/libmpg123.pc.in	2016-02-03 20:40:25.335615001 -0500
@@ -7,5 +7,5 @@
 Description: An optimised MPEG Audio decoder
 Requires: 
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lmpg123 
+Libs: -L${libdir} -lmpg123 @LIBM@
 Cflags: -I${includedir} 
