diff -Naur mpg123/src/Makemodule.am mpg123.new/src/Makemodule.am
--- mpg123/src/Makemodule.am	2016-09-26 23:20:25.000000000 +0200
+++ mpg123.new/src/Makemodule.am	2016-12-21 17:40:12.917251022 +0100
@@ -4,12 +4,6 @@
 include src/libmpg123/Makemodule.am
 include src/libout123/Makemodule.am
 
-bin_PROGRAMS += \
-  src/mpg123 \
-  src/out123 \
-  src/mpg123-id3dump \
-  src/mpg123-strip
-
 src_mpg123_LDADD = \
   src/compat/libcompat.la \
   src/libmpg123/libmpg123.la \
