From db238381db9763ca1a7fc066a613a087b37d87e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Wed, 3 Dec 2014 15:47:33 +0100
Subject: [PATCH 3/7] contribs: Fix mpcdec build with MSVC

---
 libmpcpsy/psy_tab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmpcpsy/psy_tab.c b/libmpcpsy/psy_tab.c
index 35c56ac..698601c 100644
--- a/libmpcpsy/psy_tab.c
+++ b/libmpcpsy/psy_tab.c
@@ -187,7 +187,7 @@ Ruhehoerschwelle ( PsyModel* m,
     }
 }
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && _MSC_VER < 1800
 static double
 asinh ( double x )
 {
-- 
2.45.0.windows.1

