From 22bfcda42e120ffe18b727a53b346fb489ebd88f Mon Sep 17 00:00:00 2001
From: Pavlov Konstantin <thresh@videolan.org>
Date: Tue, 17 Apr 2012 13:18:34 +0200
Subject: [PATCH 2/7] don't build binaries

---
 CMakeLists.txt | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6478c5..06d4363 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,11 +23,4 @@ add_subdirectory(libmpcdec)
 add_subdirectory(libmpcpsy)
 add_subdirectory(libmpcenc)
 add_subdirectory(libwavformat)
-add_subdirectory(mpcgain)
-add_subdirectory(mpcdec)
-add_subdirectory(mpcenc)
-add_subdirectory(mpc2sv8)
-add_subdirectory(mpccut)
-add_subdirectory(mpcchap)
-add_subdirectory(wavcmp)
 add_subdirectory(include)
-- 
2.45.0.windows.1

