diff -ru musepack.orig/CMakeLists.txt musepack/CMakeLists.txt
--- musepack.orig/CMakeLists.txt	2011-08-08 10:21:17.000000000 +0300
+++ musepack/CMakeLists.txt	2011-08-08 10:21:03.000000000 +0300
@@ -16,7 +16,7 @@
 add_definitions(-DFAST_MATH -DCVD_FASTLOG)
 
 if(NOT MSVC)
-  set(CMAKE_C_FLAGS "-O3 -Wall -fomit-frame-pointer -pipe")
+  add_definitions(-O3 -Wall -fomit-frame-pointer -pipe)
 endif(NOT MSVC)
 
 add_subdirectory(libmpcdec)
