--- postproc/configure.ranlib	2019-02-15 16:40:11.186212500 +0100
+++ postproc/configure	2019-02-15 13:43:31.904482900 +0100
@@ -95,6 +95,7 @@ Advanced options (experts only):
   --ar=AR                  use archive tool AR [$ar_default]
   --as=AS                  use assembler AS [$as_default]
   --cc=CC                  use C compiler CC [$cc_default]
+  --ranlib=RANLIB          use assembler AS [$ranlib_default]
   --ld=LD                  use linker LD
   --host-cc=HOSTCC         use host C compiler HOSTCC
   --host-cflags=HCFLAGS    use HCFLAGS when compiling for host
@@ -994,6 +995,7 @@ CMDLINE_SET="
     nm
     optflags
     pkg_config
+    ranlib
     samples
     sysinclude
     sysroot
@@ -1453,7 +1455,7 @@ ln_s="ln -sf"
 nm_default="nm"
 objformat="elf"
 pkg_config_default=pkg-config
-ranlib="ranlib"
+ranlib_default="ranlib"
 yasmexe="yasm"
 
 nm_opts='-g'
@@ -1673,7 +1675,7 @@ ar_default="${cross_prefix}${ar_default}
 cc_default="${cross_prefix}${cc_default}"
 nm_default="${cross_prefix}${nm_default}"
 pkg_config_default="${cross_prefix}${pkg_config_default}"
-ranlib="${cross_prefix}${ranlib}"
+ranlib_default="${cross_prefix}${ranlib_default}"
 
 sysinclude_default="${sysroot}/usr/include"
 
@@ -1930,7 +1932,7 @@ test -n "$cc_type" && enable $cc_type ||
 : ${as_default:=$cc}
 : ${dep_cc_default:=$cc}
 : ${ld_default:=$cc}
-set_default ar as dep_cc ld
+set_default ar as dep_cc ld ranlib
 
 test -n "$CC_DEPFLAGS" || CCDEP=$DEPEND_CMD
 test -n "$AS_DEPFLAGS" || ASDEP=$DEPEND_CMD
