diff -urN postproc-orig/configure postproc/configure
--- postproc-orig/configure	2012-08-21 23:27:39.000000000 +0300
+++ postproc/configure	2018-02-26 10:36:48.956225615 +0200
@@ -2419,8 +2419,14 @@
 enable_pic() {
     enable pic
     add_cppflags -DPIC
-    add_cflags   -fPIC
-    add_asflags  -fPIC
+    case "$target_os" in
+    mingw*|cygwin*|win*)
+        ;;
+    *)
+        add_cflags   -fPIC
+        add_asflags  -fPIC
+        ;;
+    esac
 }
 
 enabled pic && enable_pic
