--- libogg-1.3.3/configure.ac	2017-11-07 15:33:00.000000000 -0500
+++ libogg-1.3.3-new/configure.ac	2017-11-15 16:00:23.786278061 -0500
@@ -27,17 +27,17 @@
         case $host in 
         *-*-irix*)
                 DEBUG="-g -signed" 
-                CFLAGS="-O2 -w -signed"
+                EXTRA_CFLAGS="-O2 -w -signed"
                 PROFILE="-p -g3 -O2 -signed"
 		;;
         sparc-sun-solaris*)
                 DEBUG="-v -g"
-                CFLAGS="-xO4 -fast -w -fsimple -native -xcg92"
+                EXTRA_CFLAGS="-xO4 -fast -w -fsimple -native -xcg92"
                 PROFILE="-v -xpg -g -xO4 -fast -native -fsimple -xcg92 -Dsuncc"
 		;;
         *)
                 DEBUG="-g"
-                CFLAGS="-O"
+                EXTRA_CFLAGS="-O"
                 PROFILE="-g -p" 
 		;;
         esac
@@ -45,27 +45,27 @@
         case $host in 
         *-*-linux*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
+                EXTRA_CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
                 PROFILE="-Wall -W -pg -g -O20 -ffast-math -fsigned-char"
 		;;
         sparc-sun-*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -ffast-math -fsigned-char"
+                EXTRA_CFLAGS="-O20 -ffast-math -fsigned-char"
                 PROFILE="-pg -g -O20 -fsigned-char" 
 		;;
 	*-*-darwin*)
 		DEBUG="-fno-common -g -Wall -fsigned-char"
-		CFLAGS="-fno-common -O4 -Wall -fsigned-char -ffast-math"
+		EXTRA_CFLAGS="-fno-common -O3 -Wall -fsigned-char -ffast-math"
 		PROFILE="-fno-common -O4 -Wall -pg -g -fsigned-char -ffast-math"
 		;;
         *)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -fsigned-char"
+                EXTRA_CFLAGS="-O20 -fsigned-char"
                 PROFILE="-O20 -g -pg -fsigned-char" 
 		;;
         esac
 fi
-CFLAGS="$CFLAGS $cflags_save"
+CFLAGS="$EXTRA_CFLAGS $cflags_save"
 DEBUG="$DEBUG $cflags_save"
 PROFILE="$PROFILE $cflags_save"
 
