From 91ace89a505bc2070af2d585cfadaa9f97d99a8e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 6 Aug 2012 18:18:51 +0200
Subject: [PATCH 1/2] fixed compilation for iOS

---
 imdct_l_arm.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imdct_l_arm.S b/imdct_l_arm.S
index 92d67e1..35f1d55 100644
--- a/imdct_l_arm.S
+++ b/imdct_l_arm.S
@@ -202,7 +202,7 @@
 
 
     .text
-    .align
+    .align 2
 
     .global III_imdct_l
     .global _III_imdct_l
-- 
2.52.0.windows.1

