From 0267cddc89ccd5c8483d2731b50652578ca42d81 Mon Sep 17 00:00:00 2001
From: BogDan Vatra <bogdan@kde.org>
Date: Tue, 3 May 2022 13:16:07 +0300
Subject: [PATCH] Fix compile on Android

Fixes: #890
---
 libarchive/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libarchive/CMakeLists.txt b/libarchive/CMakeLists.txt
index e1d76a51..44895ce3 100644
--- a/libarchive/CMakeLists.txt
+++ b/libarchive/CMakeLists.txt
@@ -5,6 +5,10 @@
 #
 ############################################
 
+if (ANDROID)
+  include_directories(${PROJECT_SOURCE_DIR}/contrib/android/include)
+endif()
+
 # Public headers
 SET(include_HEADERS
   archive.h
-- 
2.37.3.windows.1

