From 338f388576f6ddadc9abe68bcb7841cf62c2abaa Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Fri, 3 Apr 2026 13:31:37 +0200
Subject: [PATCH 1/3] Disable CreateHardLinkW usage on old UWP targets

---
 libarchive/archive_write_disk_windows.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libarchive/archive_write_disk_windows.c b/libarchive/archive_write_disk_windows.c
index a6abd223..4a910b9f 100644
--- a/libarchive/archive_write_disk_windows.c
+++ b/libarchive/archive_write_disk_windows.c
@@ -584,6 +584,9 @@ static int
 la_CreateHardLinkW(wchar_t *linkname, wchar_t *target)
 {
 	BOOL ret;
+#if defined(WINAPI_FAMILY_PARTITION) && !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && NTDDI_VERSION < NTDDI_WIN10_19H1
+	ret = FALSE;
+#else
 	ret = CreateHardLinkW(linkname, target, NULL);
 	if (!ret) {
 		/* Under windows 2000, it is necessary to remove
@@ -605,6 +608,7 @@ la_CreateHardLinkW(wchar_t *linkname, wchar_t *target)
 #undef IS_UNC
 		ret = CreateHardLinkW(linkname, target, NULL);
 	}
+#endif
 	return (ret);
 }
 
-- 
2.52.0.windows.1

