--- libarchive/configure.ac.old	2017-06-01 16:09:59.930902994 +0200
+++ libarchive/configure.ac	2017-06-01 13:50:23.594103080 +0200
@@ -91,6 +91,7 @@
   *mingw* ) inc_windows_files=yes ;;
   *cygwin* | *msys*) inc_cygwin_files=yes ;;
 esac
+inc_windows_files=no
 AM_CONDITIONAL([INC_WINDOWS_FILES], [test $inc_windows_files = yes])
 AM_CONDITIONAL([INC_CYGWIN_FILES], [test $inc_cygwin_files = yes])
 
