--- libarchive/libarchive/filter_fork_windows.c.orig	2016-06-20 19:24:11.105360670 +0200
+++ libarchive/libarchive/filter_fork_windows.c	2016-06-20 19:25:04.049377660 +0200
@@ -25,7 +25,7 @@
 
 #include "archive_platform.h"
 
-#if defined(_WIN32) && !defined(__CYGWIN__)
+#if defined(_WIN32) && defined(HAVE_FORK)
 #include "archive_cmdline_private.h"
 #include "archive_string.h"
 
@@ -187,4 +187,4 @@
 	Sleep(100);
 }
 
-#endif /* _WIN32 && !__CYGWIN__ */
+#endif /* _WIN32 && HAVE_FORK */
