--- chromaprint/src/fft_lib_avfft.h	2012-09-05 20:05:36.000000000 +0200
+++ chromaprint.new/src/fft_lib_avfft.h	2013-04-05 17:21:56.830255210 +0200
@@ -23,6 +23,7 @@
 
 #include <math.h>
 extern "C" {
+#include <libavutil/mem.h>
 #include <libavcodec/avcodec.h>
 #include <libavcodec/avfft.h>
 }
--- chromaprint/libchromaprint.pc.cmake.orig	2013-04-11 16:03:15.878116880 +0200
+++ chromaprint/libchromaprint.pc.cmake	2013-04-11 16:03:26.178116971 +0200
@@ -7,6 +7,6 @@
 Description: Audio fingerprint library
 URL: http://wiki.acoustid.org/wiki/Chromaprint
 Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -lchromaprint
+Libs: -L${LIB_INSTALL_DIR} -lchromaprint -lavcodec -lavutil
 Cflags: -I${INCLUDE_INSTALL_DIR}
 
