diff -p1 -ru protobuf.orig/src/Makefile.am protobuf/src/Makefile.am
--- protobuf.orig/src/Makefile.am	2022-01-19 19:57:56.774686401 +0200
+++ protobuf/src/Makefile.am	2022-01-19 20:02:13.977240991 +0200
@@ -176,3 +176,4 @@ nobase_include_HEADERS =
 
-lib_LTLIBRARIES = libprotobuf-lite.la libprotobuf.la libprotoc.la
+lib_LTLIBRARIES = libprotobuf-lite.la libprotobuf.la
+check_LTLIBRARIES = libprotoc.la
 
@@ -483,3 +484,3 @@ libprotoc_la_SOURCES =
 
-bin_PROGRAMS = protoc
+#bin_PROGRAMS = protoc
 protoc_LDADD = $(PTHREAD_LIBS) libprotobuf.la libprotoc.la
@@ -661,3 +662,3 @@ protoc_outputs =
 
-BUILT_SOURCES = $(protoc_outputs)
+#BUILT_SOURCES = $(protoc_outputs)
 
