--- protobuf/src/google/protobuf/repeated_field.h.orig	2022-02-17 12:44:54.526527423 +0200
+++ protobuf/src/google/protobuf/repeated_field.h	2022-02-17 12:44:58.190446363 +0200
@@ -46,11 +46,7 @@
 #ifndef GOOGLE_PROTOBUF_REPEATED_FIELD_H__
 #define GOOGLE_PROTOBUF_REPEATED_FIELD_H__
 
-#ifdef _MSC_VER
-// This is required for min/max on VS2013 only.
 #include <algorithm>
-#endif
-
 #include <string>
 #include <iterator>
 #include <google/protobuf/stubs/casts.h>
