--- protobuf/configure.ac.gmock	2016-09-24 04:12:20.000000000 +0200
+++ protobuf/configure.ac	2020-03-24 14:46:28.900969300 +0100
@@ -183,7 +183,7 @@ AM_CONDITIONAL([OBJC_CONFORMANCE_TEST],
 #   too.
 export CFLAGS
 export CXXFLAGS
-AC_CONFIG_SUBDIRS([gmock])
+dnl AC_CONFIG_SUBDIRS([gmock])
 
 AC_CONFIG_FILES([Makefile src/Makefile benchmarks/Makefile conformance/Makefile protobuf.pc protobuf-lite.pc])
 AC_OUTPUT
