--- protobuf/cmake/install.cmake.pkgconfig	2019-02-14 13:49:06.917114800 +0100
+++ protobuf/cmake/install.cmake	2019-02-14 13:50:02.717359800 +0100
@@ -99,6 +99,24 @@ configure_file(protobuf-module.cmake.in
 configure_file(protobuf-options.cmake
   ${CMAKE_INSTALL_CMAKEDIR}/protobuf-options.cmake @ONLY)
 
+# pkg-config files
+if(NOT MSVC)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix "\${prefix}")
+  set(libdir "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
+  set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+  set(VERSION ${protobuf_VERSION})
+  if (CMAKE_USE_PTHREADS_INIT)
+    set(PTHREAD_CFLAGS "-pthread")
+  endif()
+  #set(LIBS "${ZLIB_LIBRARIES}")
+  configure_file(../protobuf.pc.in protobuf.pc @ONLY)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/protobuf.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+  configure_file(../protobuf-lite.pc.in protobuf-lite.pc @ONLY)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/protobuf-lite.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
+
 # Allows the build directory to be used as a find directory.
 export(TARGETS libprotobuf-lite libprotobuf libprotoc protoc
   NAMESPACE protobuf::
