diff -ruN live/groupsock/GroupsockHelper.cpp live.n/groupsock/GroupsockHelper.cpp
--- live/groupsock/GroupsockHelper.cpp	2012-08-20 09:33:31.000000000 +0200
+++ live.n/groupsock/GroupsockHelper.cpp	2012-08-20 10:58:28.598217212 +0200
@@ -466,7 +466,7 @@
   if (!IsMulticastAddress(groupAddress)) return True; // ignore this case
 
   struct ip_mreq_source imr;
-#ifdef ANDROID
+#ifdef __ANDROID__
     imr.imr_multiaddr = groupAddress;
     imr.imr_sourceaddr = sourceFilterAddr;
     imr.imr_interface = ReceivingInterfaceAddr;
@@ -490,7 +490,7 @@
   if (!IsMulticastAddress(groupAddress)) return True; // ignore this case
 
   struct ip_mreq_source imr;
-#ifdef ANDROID
+#ifdef __ANDROID__
     imr.imr_multiaddr = groupAddress;
     imr.imr_sourceaddr = sourceFilterAddr;
     imr.imr_interface = ReceivingInterfaceAddr;
