--- live.orig/groupsock/GroupsockHelper.cpp	2016-11-28 22:42:18.000000000 +0100
+++ live/groupsock/GroupsockHelper.cpp	2021-02-09 10:27:15.669908139 +0100
@@ -616,7 +616,7 @@
     // that other nodes will think our address is the same as we do.)
     do {
       loopbackWorks = 0; // until we learn otherwise
-
+#ifndef DISABLE_LOOPBACK_IP_ADDRESS_CHECK
       testAddr.s_addr = our_inet_addr("228.67.43.91"); // arbitrary
       Port testPort(15947); // ditto
 
@@ -653,6 +653,7 @@
 
       // We use this packet's source address, if it's good:
       loopbackWorks = !badAddressForUs(fromAddr.sin_addr.s_addr);
+#endif
     } while (0);
 
     if (sock >= 0) {
