--- live555/groupsock/GroupsockHelper.cpp.orig	2016-03-31 10:17:46.284928287 +0200
+++ live555/groupsock/GroupsockHelper.cpp	2016-03-31 10:20:20.428341877 +0200
@@ -27,6 +27,7 @@
 #include <stdarg.h>
 #include <time.h>
 #include <fcntl.h>
+#include <sys/time.h>
 #define initializeWinsockIfNecessary() 1
 #endif
 #if defined(__WIN32__) || defined(_WIN32) || defined(_QNX4)
--- live555/groupsock/include/GroupsockHelper.hh.orig	2016-03-31 10:22:56.667834195 +0200
+++ live555/groupsock/include/GroupsockHelper.hh	2016-03-31 10:23:37.411710728 +0200
@@ -133,6 +133,8 @@
 #if (defined(__WIN32__) || defined(_WIN32)) && !defined(__MINGW32__)
 // For Windoze, we need to implement our own gettimeofday()
 extern int gettimeofday(struct timeval*, int*);
+#else
+#include <sys/time.h>
 #endif
 
 // The following are implemented in inet.c:
