diff -ru sparkle/Sparkle/SUAppcast.m sparkle-fixed/Sparkle/SUAppcast.m
--- sparkle/Sparkle/SUAppcast.m	2013-12-09 15:23:42.000000000 +0100
+++ sparkle-fixed/Sparkle/SUAppcast.m	2014-01-31 13:23:11.000000000 +0100
@@ -15,6 +15,12 @@
 #import "SUConstants.h"
 #import "SULog.h"
 
+#ifndef MAC_OS_X_VERSION_10_7
+enum {
+    NSXMLNodeLoadExternalEntitiesSameOriginOnly = 1UL << 15
+};
+#endif
+
 @interface NSXMLElement (SUAppcastExtensions)
 @property (readonly, copy) NSDictionary *attributesAsDictionary;
 @end
@@ -32,7 +38,7 @@
 }
 @end
 
-@interface SUAppcast () <NSURLDownloadDelegate>
+@interface SUAppcast () // implicit NSURLDownloadDelegate
 @property (copy) NSString *downloadFilename;
 @property (strong) NSURLDownload *download;
 @property (copy) NSArray *items;
diff -ru sparkle/Sparkle/SUDiskImageUnarchiver.m sparkle-fixed/Sparkle/SUDiskImageUnarchiver.m
--- sparkle/Sparkle/SUDiskImageUnarchiver.m	2013-12-09 15:23:42.000000000 +0100
+++ sparkle-fixed/Sparkle/SUDiskImageUnarchiver.m	2014-01-31 13:25:58.000000000 +0100
@@ -12,6 +12,13 @@
 #import "SULog.h"
 #import <CoreServices/CoreServices.h>
 
+@interface SUDiskImageUnarchiver ()
+- (void)extractDMGWithPassword:(NSString *)password;
+- (BOOL)isEncrypted:(NSData*)resultData;
+- (void)continueWithPassword:(NSString *)password;
+- (void)requestPasswordFromDelegate;
+@end
+
 @implementation SUDiskImageUnarchiver
 
 + (BOOL)canUnarchivePath:(NSString *)path
