diff -ru lua/Makefile lua-f/Makefile
--- lua/Makefile	2008-08-12 02:40:48.000000000 +0200
+++ lua-f/Makefile	2012-03-21 13:58:30.000000000 +0100
@@ -38,7 +38,7 @@
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
 # Convenience platforms targets.
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= aix ansi bsd freebsd generic linux ios macosx mingw posix solaris
 
 # What to install.
 TO_BIN= lua luac
diff -ru lua/src/Makefile lua-f/src/Makefile
--- lua/src/Makefile	2012-03-21 13:58:59.000000000 +0100
+++ lua-f/src/Makefile	2012-03-21 13:57:08.000000000 +0100
@@ -98,6 +98,9 @@
 linux:
 	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl"
 
+ios:
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX
+
 macosx:
 	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline"
 # use this on Mac OS X 10.3-
