LUA: do not use readline on Linux
--- lua-5.3/src/Makefile    2016-03-16 11:53:27.008785835 +0100
+++ lua/src/Makefile    2016-03-16 11:53:38.825752456 +0100
@@ -107,7 +107,7 @@
 generic: $(ALL)
 
 linux:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl"
 
 macosx:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
--- lua-5.3/src/luaconf.h.orig	2016-03-16 12:29:35.600387048 +0100
+++ lua/src/luaconf.h	2016-03-16 12:29:51.140780084 +0100
@@ -61,7 +61,7 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
+//#define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
 
