--- lua/src/lmathlib.c	2016-03-16 16:13:27.593329107 +0100
+++ lua/src/lmathlib.c.new	2016-03-16 16:13:05.285328491 +0100
@@ -183,7 +183,7 @@
     res = l_mathop(log)(x);
   else {
     lua_Number base = luaL_checknumber(L, 2);
-#if !defined(LUA_USE_C89)
+#if !defined(LUA_USE_C89) && !defined(__ANDROID__)
     if (base == 2.0) res = l_mathop(log2)(x); else
 #endif
     if (base == 10.0) res = l_mathop(log10)(x);
