--- lua/src/luaconf.h.orig	2016-03-16 15:53:05.433295367 +0100
+++ lua/src/luaconf.h	2016-03-16 15:53:21.497295811 +0100
@@ -654,7 +654,8 @@
 ** macro must include header 'locale.h'.)
 */
 #if !defined(lua_getlocaledecpoint)
-#define lua_getlocaledecpoint()		(localeconv()->decimal_point[0])
+//#define lua_getlocaledecpoint()		(localeconv()->decimal_point[0])
+#define lua_getlocaledecpoint()		'.'
 #endif
 
 /* }================================================================== */
