--- lua/src/llex.c.orig	2016-03-16 12:21:12.567664718 +0100
+++ lua/src/llex.c	2016-03-16 12:21:39.361675706 +0100
@@ -226,7 +226,7 @@
 */
 static void trydecpoint (LexState *ls, TValue *o) {
   char old = ls->decpoint;
-  ls->decpoint = lua_getlocaledecpoint();
+  //ls->decpoint = lua_getlocaledecpoint();
   buffreplace(ls, old, ls->decpoint);  /* try new decimal separator */
   if (luaO_str2num(luaZ_buffer(ls->buff), o) == 0) {
     /* format error with correct decimal point: no more options */
