From 54a245738b12ad8985935de0ffea24980ce48db9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Thu, 20 Apr 2017 17:26:02 +0200
Subject: [PATCH 2/5] Fix pthread detection

---
 configure.ac | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6ea38f53..781f3163 100644
--- a/configure.ac
+++ b/configure.ac
@@ -239,7 +239,6 @@ LIBGCRYPT_THREAD_MODULES=""
 # Other definitions.
 have_w32_system=no
 have_w32ce_system=no
-have_pthread=no
 
 
 # Setup some stuff depending on host.
@@ -828,10 +827,9 @@ AC_SUBST(PTH_LIBS)
 # Check whether pthreads is available
 #
 if test "$have_w32_system" != yes; then
-  AC_CHECK_LIB(pthread,pthread_create,have_pthread=yes)
-  if test "$have_pthread" = yes; then
-    AC_DEFINE(HAVE_PTHREAD, 1 ,[Define if we have pthread.])
-  fi
+  AC_SEARCH_LIBS(pthread_create, pthread,[
+      AC_DEFINE(HAVE_PTHREAD, 1 ,[Define if we have pthread.])
+  ])
 fi
 
 
-- 
2.37.3.windows.1

