--- libgcrypt/Makefile.am.orig	2012-07-23 16:41:58.927873202 +0200
+++ libgcrypt/Makefile.am	2012-07-23 16:42:05.723873200 +0200
@@ -23,7 +23,7 @@
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
 DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS =         compat mpi cipher random src doc tests
+SUBDIRS =         compat mpi cipher random src doc
 EXTRA_DIST = autogen.sh README.SVN
 DISTCLEANFILES =
 
