--- automake/lib/depcomp.clangmsvc	2014-12-31 14:26:32.000000000 +0100
+++ automake/lib/depcomp	2019-03-01 11:33:02.672713200 +0100
@@ -563,6 +563,36 @@ msvc7msys)
   exit 1
   ;;
 
+gccwindows)
+## clang producing windows pathes
+  for arg
+  do
+    case $arg in
+    -c) set fnord "$@" -MT "$object" -MD -MP -MF "$tmpdepfile" "$arg" ;;
+    *)  set fnord "$@" "$arg" ;;
+    esac
+    shift # fnord
+    shift # $arg
+  done
+  "$@"
+  stat=$?
+  if test $stat -ne 0; then
+    rm -f "$tmpdepfile"
+    exit $stat
+  fi
+  rm -f "$depfile"
+
+  # Extracts the file names and escapes backslashes for cygpath.
+  sed < "$tmpdepfile" '
+/^*\(.*\)/ {
+  s//\1/
+  s/\\/\\\\/g
+  p
+}' | $cygpath_u | sed "s, /, \\\\,g" >> "$depfile"
+  echo >> "$depfile" # make sure the fragment doesn't end with a backslash
+  rm -f "$tmpdepfile"
+  ;;
+
 #nosideeffect)
   # This comment above is used by automake to tell side-effect
   # dependency tracking mechanisms from slower ones.
