--- automake/lib/depcomp	2019-03-01 16:29:06.073143400 +0100
+++ automake/lib/depcomp.gccwindows	2019-03-01 16:29:30.646075800 +0100
@@ -563,6 +563,36 @@ msvc7msys)
   exit 1
   ;;
 
+gccwindows)
+## clang producing windows pathes
+  for arg
+  do
+    case $arg in
+    -c) set fnord "$@" -MT "$object" -MD -MP -MF "$tmpdepfile" "$arg" ;;
+    *)  set fnord "$@" "$arg" ;;
+    esac
+    shift # fnord
+    shift # $arg
+  done
+  "$@"
+  stat=$?
+  if test $stat -ne 0; then
+    rm -f "$tmpdepfile"
+    exit $stat
+  fi
+  rm -f "$depfile"
+
+  # Extracts the file names and escapes backslashes for cygpath.
+  sed < "$tmpdepfile" '
+/^*\(.*\)/ {
+  s//\1/
+  s/\\/\\\\/g
+  p
+}' | $cygpath_u | sed "s, /, \\\\,g" >> "$depfile"
+  echo >> "$depfile" # make sure the fragment doesn't end with a backslash
+  rm -f "$tmpdepfile"
+  ;;
+
 #nosideeffect)
   # This comment above is used by automake to tell side-effect
   # dependency tracking mechanisms from slower ones.
--- automake/m4/depend.m4	2019-03-01 16:22:29.083243300 +0100
+++ automake/m4/depend.m4.gccwindows	2019-03-01 16:22:36.408049300 +0100
@@ -152,7 +152,8 @@ fi
 AC_SUBST([$1DEPMODE], [depmode=$am_cv_$1_dependencies_compiler_type])
 AM_CONDITIONAL([am__fastdep$1], [
   test "x$enable_dependency_tracking" != xno \
-  && test "$am_cv_$1_dependencies_compiler_type" = gcc3])
+  && ( test "$am_cv_$1_dependencies_compiler_type" = gcc3 ||
+       test "$am_cv_$1_dependencies_compiler_type" = gccwindows )])
 ])
 
 
