--- cmake/Modules/FindPkgConfig.cmake.msys_pkg	2018-10-01 16:51:37.332632500 +0200
+++ cmake/Modules/FindPkgConfig.cmake	2018-10-01 16:51:46.815041400 +0200
@@ -344,6 +344,20 @@ macro(_pkg_check_modules_internal _is_re
       unset(_lib_dirs)
       unset(_pkgconfig_path)
     endif()
+    
+    if (CYGWIN OR MSYS)
+        # fix the PKG_CONFIG_PATH transformed into a cmake path with incorrect msys pathes
+        file(TO_CMAKE_PATH "$ENV{PKG_CONFIG_PATH}" _pkgconfig_path_cmake)
+        execute_process(
+            COMMAND sh.exe "-c" "cygpath -up '${_pkgconfig_path_cmake}'"
+            RESULT_VARIABLE _cygpath_retval
+            OUTPUT_VARIABLE _cygpath_outval
+            ERROR_VARIABLE _cygpath_error
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            ERROR_STRIP_TRAILING_WHITESPACE)
+        set(ENV{PKG_CONFIG_PATH} ${_cygpath_outval})
+        #message(STATUS "new ENV $ENV{PKG_CONFIG_PATH}")
+    endif()
 
     # iterate through module list and check whether they exist and match the required version
     foreach (_pkg_check_modules_pkg ${_pkg_check_modules_list})
