From 1d28edfb0574c77591403f133d42e0edf7d037e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Wed, 19 Jun 2024 13:59:41 +0200
Subject: [PATCH] config: add support for arm64_32

---
 config.guess | 3 +++
 1 files changed, 3 insertions(+), 0 deletion(-)

diff --git a/config.guess b/config.guess
index f6d217a4..84bd5401 100755
--- a/config.guess
+++ b/config.guess
@@ -1455,6 +1455,9 @@ EOF
     arm64:Darwin:*:*)
 	GUESS=aarch64-apple-darwin$UNAME_RELEASE
 	;;
+    arm64_32:Darwin:*:*)
+	GUESS=aarch64_32-apple-darwin$UNAME_RELEASE
+	;;
     *:Darwin:*:*)
 	UNAME_PROCESSOR=`uname -p`
 	case $UNAME_PROCESSOR in
-- 
2.32.1 (Apple Git-133)

