--- meson/mesonbuild/linkers.py.shlib	2020-01-31 11:54:36.928015800 +0100
+++ meson/mesonbuild/linkers.py	2020-01-31 11:54:55.088659900 +0100
@@ -659,8 +659,17 @@ class LLVMDynamicLinker(GnuLikeDynamicLi
     This is only the posix-like linker.
     """
 
-    pass
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
+        # Some targets don't seem to support this argument (windows, wasm, ...)
+        _, _, e = mesonlib.Popen_safe(self.exelist + self._apply_prefix('--allow-shlib-undefined'))
+        self.has_allow_shlib_undefined = not ('unknown argument: --allow-shlib-undefined' in e)
+
+    def get_allow_undefined_args(self) -> T.List[str]:
+        if self.has_allow_shlib_undefined:
+            return self._apply_prefix('--allow-shlib-undefined')
+        return []
 
 class CcrxDynamicLinker(DynamicLinker):
 
