From 3d6ce13a048b7f7b328ac70313c6ff9ad6a59752 Mon Sep 17 00:00:00 2001
From: Sean McGovern <gseanmcg@gmail.com>
Date: Sat, 4 Sep 2021 18:34:09 +0200
Subject: [PATCH 07/11] fix __va_copy() usage

---
 src/misc.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/misc.h b/src/misc.h
index d0e965e..f7b0094 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -411,7 +411,9 @@ _vbi_time_max			(void)
 #endif
 
 /* __va_copy is a GNU extension. */
-#ifndef __va_copy
+#ifdef va_copy && !defined(__va_copy)
+#  define __va_copy va_copy
+#else
 #  define __va_copy(ap1, ap2) do { ap1 = ap2; } while (0)
 #endif
 
-- 
2.52.0.windows.1

