From a91bc96aed18e132fca9ac827b7ac23d44d8c21f Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Wed, 10 Dec 2014 22:13:22 +0100
Subject: [PATCH 05/11] fix zvbi compilation for Android

---
 src/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 3edf837..f2ba028 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -94,7 +94,6 @@ libzvbi_la_SOURCES = \
 	page_table.c page_table.h \
 	pdc.c pdc.h \
 	pfc_demux.c pfc_demux.h \
-	proxy-client.c proxy-client.h \
 	raw_decoder.c raw_decoder.h \
 	sampling_par.c sampling_par.h \
 	search.c search.h ure.c ure.h \
-- 
2.52.0.windows.1

