--- libupnp-1.6.21/upnp/src/api/upnpapi.c.orig	2017-04-09 19:05:45.803495288 +0200
+++ libupnp-1.6.21/upnp/src/api/upnpapi.c	2017-04-09 19:07:08.719491594 +0200
@@ -3260,7 +3260,8 @@
 		ifname_found = 1;
 	}
 	for (adapts_item = adapts; adapts_item != NULL; adapts_item = adapts_item->Next) {
-		if (adapts_item->Flags & IP_ADAPTER_NO_MULTICAST) {
+		if (adapts_item->Flags & IP_ADAPTER_NO_MULTICAST ||
+			adapts_item->OperStatus != IfOperStatusUp) {
 			adapts_item = adapts_item->Next;
 			continue;
 		}
