From 9ec9967c55e51c387f7d517e72adb6ed278d153f Mon Sep 17 00:00:00 2001
From: Alaric Senat <dev.asenat@posteo.net>
Date: Wed, 10 Mar 2021 15:25:20 +0100
Subject: win64

---
 upnp/inc/upnp.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/upnp/inc/upnp.h b/upnp/inc/upnp.h
index 0a51b34..4711e11 100644
--- a/upnp/inc/upnp.h
+++ b/upnp/inc/upnp.h
@@ -61,6 +61,19 @@
 	/* Other systems ??? */
 #endif
 
+#if defined(__MINGW32__)
+	#if !defined(_OFF_T_)
+   typedef long long _off_t;
+   typedef _off_t off_t;
+		#define _OFF_T_
+	#else
+		#ifdef off_t
+			#undef off_t
+		#endif
+		#define off_t long long
+	#endif
+#endif
+
 #ifdef UPNP_ENABLE_OPEN_SSL
 	#include <openssl/ssl.h>
 #endif
-- 
2.29.2

