diff -Naur libupnp/configure.ac libupnp.new/configure.ac
--- libupnp/configure.ac	2013-11-15 17:18:45.000000000 +0100
+++ libupnp.new/configure.ac	2014-12-15 19:36:30.625350753 +0100
@@ -557,6 +557,7 @@
 echo "-------------------------------------------------------------------------------"
 
 
+AC_DEFINE([_WIN32_WINNT], 0x0501, [Define to '0x0500' for Windows 2000 APIs.])
 #
 # Lot's of stuff to ensure large file support
 #
diff -Naur libupnp/libupnp.pc.in libupnp.new/libupnp.pc.in
--- libupnp/libupnp.pc.in	2010-12-23 21:24:05.000000000 +0100
+++ libupnp.new/libupnp.pc.in	2014-12-15 19:36:30.625350753 +0100
@@ -6,6 +6,6 @@
 Name: libupnp
 Description: Linux SDK for UPnP Devices
 Version: @VERSION@
-Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp -lthreadutil -lixml 
+Libs: @PTHREAD_CFLAGS@ -L${libdir} -lupnp -lthreadutil -lixml -liphlpapi @PTHREAD_LIBS@ 
 Cflags: @PTHREAD_CFLAGS@ -I${includedir}/upnp
 
diff -Naur libupnp/upnp/inc/UpnpInet.h libupnp.new/upnp/inc/UpnpInet.h
--- libupnp/upnp/inc/UpnpInet.h	2012-03-26 18:49:42.000000000 +0200
+++ libupnp.new/upnp/inc/UpnpInet.h	2014-12-15 19:36:48.353041056 +0100
@@ -15,13 +15,7 @@
 
 #ifdef WIN32
 	#include <stdarg.h>
-	#ifndef UPNP_USE_MSVCPP
-		/* Removed: not required (and cause compilation issues) */
-		#include <winbase.h>
-		#include <windef.h>
-	#endif
 	#include <winsock2.h>
-	#include <iphlpapi.h>
 	#include <ws2tcpip.h>
 
 	#define UpnpCloseSocket closesocket
diff -Naur libupnp/upnp/src/inc/upnputil.h libupnp.new/upnp/src/inc/upnputil.h
--- libupnp/upnp/src/inc/upnputil.h	2012-04-29 00:32:06.000000000 +0200
+++ libupnp.new/upnp/src/inc/upnputil.h	2014-12-15 19:36:30.625350753 +0100
@@ -130,7 +130,7 @@
 		#define strncasecmp		strnicmp
 		#define sleep(a)		Sleep((a)*1000)
 		#define usleep(a)		Sleep((a)/1000)
-		#define strerror_r(a,b,c)	(strerror_s((b),(c),(a)))
+		#define strerror_r(a,b,c)       strncpy( b, strerror(a), c)
 	#else
 		#define max(a, b)   (((a)>(b))? (a):(b))
 		#define min(a, b)   (((a)<(b))? (a):(b))
