--- upnp/configure.ac.pthread-w32	2018-05-17 11:51:47.115502500 +0200
+++ upnp/configure.ac	2018-05-17 12:49:30.371786100 +0200
@@ -667,6 +667,7 @@ ACX_PTHREAD(
 #
 # Update environment variables for pthreads
 #
+PTHREAD_LIBS="-lpthreadGC2"
 CC="$PTHREAD_CC"
 CFLAGS="$PTHREAD_CFLAGS $CFLAGS"
 LIBS="$PTHREAD_LIBS $LIBS"
@@ -676,7 +677,7 @@ PC_CFLAGS="$PTHREAD_CFLAGS"
 # WIN32 specific
 if test "$ac_cv_win32" = "yes"; then
 WIN32_LIBS="-liphlpapi -lws2_32"
-PC_CFLAGS="$PC_CFLAGS -DUPNP_STATIC_LIB"
+PC_CFLAGS="$PC_CFLAGS -DUPNP_STATIC_LIB -DPTW32_STATIC_LIB"
 fi
 AC_SUBST(PC_CFLAGS)
 AC_SUBST(WIN32_LIBS)
