From ef877d8c4991adf4d0615892fee64f6212f850b1 Mon Sep 17 00:00:00 2001
From: Alaric Senat <dev.asenat@posteo.net>
Date: Wed, 10 Mar 2021 14:54:00 +0100
Subject: no-debugfile

---
 upnp/src/inc/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/upnp/src/inc/config.h b/upnp/src/inc/config.h
index 8dc8cdb..6f2e17f 100644
--- a/upnp/src/inc/config.h
+++ b/upnp/src/inc/config.h
@@ -409,7 +409,7 @@
  * see <upnp/inc/upnpdebug.h>
  */
 
-#define DEBUG_ALL		1
+#define DEBUG_ALL		0
 #define DEBUG_SSDP		0
 #define DEBUG_SOAP		0
 #define DEBUG_GENA		0
-- 
2.29.2

