--- upnp/upnp/src/threadutil/ThreadPool.h.orig	2022-04-21 15:07:27.194156972 +0300
+++ upnp/upnp/src/threadutil/ThreadPool.h	2022-04-21 15:07:31.026076222 +0300
@@ -54,8 +54,8 @@
 			int  tz_minuteswest; /* minutes W of Greenwich */
 			int  tz_dsttime;     /* type of dst correction */
 		};
-		int gettimeofday(struct timeval *tv, struct timezone *tz);
 	#endif
+	int gettimeofday(struct timeval *tv, struct timezone *tz);
 #else /* _WIN32 */
 	#include <sys/param.h>
 	#include <sys/time.h> /* for gettimeofday() */
