--- lame/configure.in.autotools	2012-02-28 19:50:27.000000000 +0100
+++ lame/configure.in	2018-05-24 08:17:53.119940000 +0200
@@ -78,7 +78,7 @@ if test "${GCC}" = "yes"; then
 fi
 
 dnl more automake stuff
-AM_C_PROTOTYPES
+dnl AM_C_PROTOTYPES
 
 AC_CHECK_HEADER(dmalloc.h)
 if test "${ac_cv_header_dmalloc_h}" = "yes"; then
@@ -393,7 +393,7 @@ CONFIG_MATH_LIB="${USE_LIBM}"
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+dnl AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
--- lame/doc/html/Makefile.am.autotools	2018-05-24 08:42:16.439442600 +0200
+++ lame/doc/html/Makefile.am	2018-05-24 08:19:44.596125100 +0200
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.7 2010/09/30 20:58:40 jaz001 Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 docdir = $(datadir)/doc
 pkgdocdir = $(docdir)/$(PACKAGE)
--- lame/doc/man/Makefile.am.autotools	2018-05-24 08:42:11.199809700 +0200
+++ lame/doc/man/Makefile.am	2018-05-24 08:19:48.855510100 +0200
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.1 2000/10/22 11:39:44 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 man_MANS = lame.1
 EXTRA_DIST = ${man_MANS}
--- lame/libmp3lame/i386/Makefile.am.autotools	2018-05-24 08:42:07.015600500 +0200
+++ lame/libmp3lame/i386/Makefile.am	2018-05-24 08:19:54.559671800 +0200
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.26 2011/04/04 09:42:34 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign $(top_srcdir)/ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 DEFS = @DEFS@ @CONFIG_DEFS@
 
