--- harfbuzz/CMakeLists.txt.old	2018-02-28 22:59:53.000000000 +0100
+++ harfbuzz/CMakeLists.txt	2018-05-18 18:00:47.589767685 +0200
@@ -272,7 +272,7 @@
     message(FATAL_ERROR "HB_HAVE_FREETYPE was set, but we failed to find it. Maybe add a CMAKE_PREFIX_PATH= to your Freetype2 install prefix")
   endif ()
 
-  list(APPEND THIRD_PARTY_LIBS ${FREETYPE_LIBRARIES})
+  list(APPEND THIRD_PARTY_LIBS ${FREETYPE_LIBRARIES} -lz)
   include_directories(AFTER ${FREETYPE_INCLUDE_DIRS})
   add_definitions(-DHAVE_FREETYPE=1)
 
