--- harfbuzz-1.0.6/configure.ac.orig	2015-10-26 00:06:30.060254321 +0100
+++ harfbuzz-1.0.6/configure.ac	2015-10-26 00:07:17.364140612 +0100
@@ -93,14 +93,6 @@
 		;;
 	esac
 
-	case "$host" in
-		arm-*-*)
-			if test "x$ac_cv_alignof_struct_char__" != x1; then
-				# Request byte alignment
-				CXXFLAGS="$CXXFLAGS -mstructure-size-boundary=8"
-			fi
-		;;
-	esac
 fi
 
 AM_CONDITIONAL(HAVE_GCC, test "x$GCC" = "xyes")
