diff -urN harfbuzz.bak/CMakeLists.txt harfbuzz/CMakeLists.txt
--- harfbuzz.bak/CMakeLists.txt	2020-04-01 22:11:59.430060242 +0300
+++ harfbuzz/CMakeLists.txt	2020-04-01 22:34:27.074188908 +0300
@@ -303,32 +303,26 @@
 
   list(APPEND project_headers ${PROJECT_SOURCE_DIR}/src/hb-coretext.h)
 
-  if (HB_IOS)
-    find_library(COREFOUNDATION CoreFoundation)
-    if (COREFOUNDATION)
-      list(APPEND THIRD_PARTY_LIBS ${COREFOUNDATION})
-    endif ()
-    mark_as_advanced(COREFOUNDATION)
-
-    find_library(CORETEXT CoreText)
-    if (CORETEXT)
-      list(APPEND THIRD_PARTY_LIBS ${CORETEXT})
-    endif ()
-    mark_as_advanced(CORETEXT)
+  find_library(COREFOUNDATION CoreFoundation)
+  if (COREFOUNDATION)
+    list(APPEND THIRD_PARTY_LIBS ${COREFOUNDATION})
+    set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreText")
+  endif ()
+  mark_as_advanced(COREFOUNDATION)
 
-    find_library(COREGRAPHICS CoreGraphics)
-    if (COREGRAPHICS)
-      list(APPEND THIRD_PARTY_LIBS ${COREGRAPHICS})
-    endif ()
-    mark_as_advanced(COREGRAPHICS)
-  else ()
-    find_library(APPLICATION_SERVICES_FRAMEWORK ApplicationServices)
-    if (APPLICATION_SERVICES_FRAMEWORK)
-      list(APPEND THIRD_PARTY_LIBS ${APPLICATION_SERVICES_FRAMEWORK})
-    endif ()
+  find_library(CORETEXT CoreText)
+  if (CORETEXT)
+    list(APPEND THIRD_PARTY_LIBS ${CORETEXT})
+    set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreFoundation")
+  endif ()
+  mark_as_advanced(CORETEXT)
 
-    mark_as_advanced(APPLICATION_SERVICES_FRAMEWORK)
+  find_library(COREGRAPHICS CoreGraphics)
+  if (COREGRAPHICS)
+    list(APPEND THIRD_PARTY_LIBS ${COREGRAPHICS})
+    set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreGraphics")
   endif ()
+  mark_as_advanced(COREGRAPHICS)
 endif ()
 
 if (WIN32 AND HB_HAVE_GDI)
@@ -489,6 +483,18 @@
 add_library(harfbuzz ${project_sources} ${project_extra_sources} ${project_headers})
 target_link_libraries(harfbuzz ${THIRD_PARTY_LIBS})
 
+## harfbuzz pc file
+foreach(LIB ${THIRD_PARTY_LIBS})
+  if(NOT IS_ABSOLUTE ${LIB})
+    set(PRIVATE_LIBS "${PRIVATE_LIBS} -l${LIB}")
+  endif()
+endforeach()
+
+set(PRIVATE_LIBS "${PRIVATE_LIBS} ${FRAMEWORK_FLAGS}")
+configure_file("harfbuzz-cmake.pc.in" "harfbuzz.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/harfbuzz.pc"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+
 ## Define harfbuzz-subset library
 if (HB_BUILD_SUBSET)
   add_library(harfbuzz-subset ${subset_project_sources} ${subset_project_headers})
diff -urN harfbuzz.bak/harfbuzz-cmake.pc.in harfbuzz/harfbuzz-cmake.pc.in
--- harfbuzz.bak/harfbuzz-cmake.pc.in	1970-01-01 03:00:00.000000000 +0300
+++ harfbuzz/harfbuzz-cmake.pc.in	2020-04-01 22:12:10.638058586 +0300
@@ -0,0 +1,12 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: harfbuzz
+Description: HarfBuzz text shaping library
+Version: @HB_VERSION@
+
+Libs: -L${libdir} -lharfbuzz -lm @PRIVATE_LIBS@
+Requires.private:
+Cflags: -I${includedir}/harfbuzz
