--- harfbuzz-2.0.0/CMakeLists.txt.orig	2020-03-29 16:57:36.480432086 +0200
+++ harfbuzz-2.0.0/CMakeLists.txt	2020-03-29 16:59:29.568430730 +0200
@@ -396,18 +396,21 @@
     find_library(COREFOUNDATION CoreFoundation)
     if (COREFOUNDATION)
       list(APPEND THIRD_PARTY_LIBS ${COREFOUNDATION})
+      set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreText")
     endif ()
     mark_as_advanced(COREFOUNDATION)
 
     find_library(CORETEXT CoreText)
     if (CORETEXT)
       list(APPEND THIRD_PARTY_LIBS ${CORETEXT})
+      set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreFoundation")
     endif ()
     mark_as_advanced(CORETEXT)
 
     find_library(COREGRAPHICS CoreGraphics)
     if (COREGRAPHICS)
       list(APPEND THIRD_PARTY_LIBS ${COREGRAPHICS})
+      set(FRAMEWORK_FLAGS "${FRAMEWORK_FLAGS} -framework CoreGraphics")
     endif ()
     mark_as_advanced(COREGRAPHICS)
   else ()
@@ -579,6 +582,18 @@
 add_library(harfbuzz ${project_sources} ${project_extra_sources} ${project_headers})
 target_link_libraries(harfbuzz ${THIRD_PARTY_LIBS})
 
+## harfbuzz pc file
+foreach(LIB ${THIRD_PARTY_LIBS})
+  if(NOT IS_ABSOLUTE ${LIB})
+    set(PRIVATE_LIBS "${PRIVATE_LIBS} -l${LIB}")
+  endif()
+endforeach()
+
+set(PRIVATE_LIBS "${PRIVATE_LIBS} ${FRAMEWORK_FLAGS}")
+configure_file("src/harfbuzz.pc.in" "src/harfbuzz.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/src/harfbuzz.pc"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+
 ## Define harfbuzz-subset library
 if (NOT HB_DISABLE_SUBSET)
   add_library(harfbuzz-subset ${subset_project_sources} ${subset_project_headers})
