--- harfbuzz-2.6.0/CMakeLists.txt.orig	2019-07-26 23:23:47.000000000 +0200
+++ harfbuzz-2.6.0/CMakeLists.txt	2020-03-29 18:32:04.608364093 +0200
@@ -201,6 +201,14 @@
 
 ## Find and include needed header folders and libraries
 if (HB_HAVE_FREETYPE)
+  find_package(PkgConfig)
+  pkg_check_modules(PC_FREETYPE2 QUIET freetype2)
+
+  if (PC_FREETYPE2_FOUND)
+    include_directories(AFTER ${PC_FREETYPE2_STATIC_INCLUDE_DIRS})
+    list(APPEND THIRD_PARTY_LIBS ${PC_FREETYPE2_STATIC_LIBRARIES})
+    add_definitions(${PC_FREETYPE2_STATIC_CFLAGS})
+  else ()
   include (FindFreetype)
   if (NOT FREETYPE_FOUND)
     message(FATAL_ERROR "HB_HAVE_FREETYPE was set, but we failed to find it. Maybe add a CMAKE_PREFIX_PATH= to your Freetype2 install prefix")
@@ -208,13 +216,15 @@
 
   list(APPEND THIRD_PARTY_LIBS ${FREETYPE_LIBRARIES})
   include_directories(AFTER ${FREETYPE_INCLUDE_DIRS})
-  add_definitions(-DHAVE_FREETYPE=1)
-
-  list(APPEND project_headers ${PROJECT_SOURCE_DIR}/src/hb-ft.h)
 
   # So check_funcs can find its headers
   set (CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${FREETYPE_INCLUDE_DIRS})
   set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${FREETYPE_LIBRARIES})
+  endif ()
+
+  add_definitions(-DHAVE_FREETYPE=1)
+
+  list(APPEND project_headers ${PROJECT_SOURCE_DIR}/src/hb-ft.h)
 
   check_funcs(FT_Get_Var_Blend_Coordinates FT_Set_Var_Blend_Coordinates FT_Done_MM_Var)
 endif ()
