--- harfbuzz-2.6.0/CMakeLists.txt.orig	2020-03-29 18:34:37.596362257 +0200
+++ harfbuzz-2.6.0/CMakeLists.txt	2020-03-29 18:34:53.984362061 +0200
@@ -287,6 +287,20 @@
   # Apple Advanced Typography
   add_definitions(-DHAVE_CORETEXT)
 
+  set(CMAKE_FIND_ROOT_PATH ${CMAKE_IOS_DEVELOPER_ROOT} ${CMAKE_OSX_SYSROOT}
+    ${CMAKE_PREFIX_PATH} CACHE string  "appleOS find search path root" FORCE)
+  # Default to searching for frameworks first.
+  set(CMAKE_FIND_FRAMEWORK FIRST)
+  # Set up the default search directories for frameworks.
+  set(CMAKE_SYSTEM_FRAMEWORK_PATH
+    ${CMAKE_OSX_SYSROOT}/System/Library/Frameworks
+    ${CMAKE_OSX_SYSROOT}/System/Library/PrivateFrameworks
+    ${CMAKE_OSX_SYSROOT}/Developer/Library/Frameworks)
+  # Only search the specified iOS SDK, not the remainder of the host filesystem.
+  set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM ONLY)
+  set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+  set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
   list(APPEND project_headers ${PROJECT_SOURCE_DIR}/src/hb-coretext.h)
 
   if (HB_IOS)
