diff -urN harfbuzz-orig/src/hb-private.hh harfbuzz/src/hb-private.hh
--- harfbuzz-orig/src/hb-private.hh	2017-01-10 06:18:28.000000000 +0000
+++ harfbuzz/src/hb-private.hh	2017-12-21 21:46:12.254147419 +0000
@@ -684,7 +684,7 @@
   fprintf (stderr, "%-10s", what ? what : "");
 
   if (obj)
-    fprintf (stderr, "(%0*lx) ", (unsigned int) (2 * sizeof (void *)), (unsigned long) obj);
+    fprintf (stderr, "(%0*p) ", (unsigned int) (2 * sizeof (void *)), obj);
   else
     fprintf (stderr, " %*s  ", (unsigned int) (2 * sizeof (void *)), "");
 
