diff --git a/meson.build b/meson.build
index a77780667dfb07d700d39c6c3353c0ba4906b178..0ed820152476e0caed3ba6d203ba236083fb8438 100755
--- a/meson.build
+++ b/meson.build
@@ -156,7 +156,7 @@ endif
 
 #On ubuntu cjson does not come with pkgconfig files, hence the extended checking.
 if not builtin_cjson
-	cjson_lib = dependency('cJSON', required: false)
+	cjson_lib = dependency('libcjson', required: false)
 	if not cjson_lib.found()
 		cjson_lib = cc.find_library('cjson', required: required_library, has_headers: ['cjson/cJSON.h'])
 		if not cjson_lib.found()
