--- libvorbis/lib/os.h	2014-01-22 10:53:32.000000000 +0100
+++ /home/vlc/vlc/contrib/windowsphone/libvorbis/lib/os.h	2014-12-15 16:54:46.588645600 +0100
@@ -120,7 +120,7 @@

 /* MSVC inline assembly. 32 bit only; inline ASM isn't implemented in the
  * 64 bit compiler */
-#if defined(_MSC_VER) && !defined(_WIN64) && !defined(_WIN32_WCE)
+#if defined(_MSC_VER) && !defined(_WIN64) && !defined(_WIN32_WCE) && !defined(_M_ARM)
 #  define VORBIS_FPU_CONTROL

 typedef ogg_int16_t vorbis_fpu_control;
