diff -ru libvorbis/configure.ac libvorbis-fixed/configure.ac
--- libvorbis/configure.ac	2015-01-12 21:00:37.000000000 +0100
+++ libvorbis-fixed/configure.ac	2015-09-30 22:25:19.000000000 +0200
@@ -197,9 +197,9 @@
 		CFLAGS="-O3 -Wall -Wextra -ffast-math -D__NO_MATH_INLINES -fsigned-char $sparc_cpu"
 		PROFILE="-pg -g -O3 -D__NO_MATH_INLINES -fsigned-char $sparc_cpu" ;;
 	*-*-darwin*)
-		DEBUG="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -O0 -fsigned-char"
-		CFLAGS="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -O3 -ffast-math -fsigned-char"
-		PROFILE="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -pg -O3 -ffast-math -fsigned-char";;
+		DEBUG="-DDARWIN -Wall -g -O0 -fsigned-char"
+		CFLAGS="-DDARWIN -Wall -g -O3 -ffast-math -fsigned-char"
+		PROFILE="-DDARWIN -Wall -g -pg -O3 -ffast-math -fsigned-char";;
 	*-*-os2*)
 		# Use -W instead of -Wextra because gcc on OS/2 is an old version.
 		DEBUG="-g -Wall -W -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
