--- orc/orc/orc.h	2011-04-14 21:07:08.000000000 +0200
+++ orc.new/orc/orc.h	2011-05-08 19:03:51.294908858 +0200
@@ -2,6 +2,7 @@
 #ifndef _ORC_ORC_H_
 #define _ORC_ORC_H_
 
+#include <orc/orcutils.h>
 #include <orc/orcprogram.h>
 #include <orc/orc-stdint.h>
 #include <orc/orcarm.h>
@@ -12,7 +13,6 @@
 #include <orc/orcparse.h>
 #include <orc/orcprogram.h>
 #include <orc/orcsse.h>
-#include <orc/orcutils.h>
 #include <orc/orcx86.h>
 #include <orc/orcpowerpc.h>
 
--- orc/orc/orcutils.h	2011-04-28 00:29:24.000000000 +0200
+++ orc.new/orc/orcutils.h	2011-05-08 19:10:13.380803529 +0200
@@ -30,8 +30,8 @@
 
 #ifndef _ORC_INTEGER_TYPEDEFS_
 #define _ORC_INTEGER_TYPEDEFS_
-#if defined(__STDC__) && __STDC__ && __STDC_VERSION__ >= 199901L
 #include <stdint.h>
+#if defined(__STDC__) && __STDC__ && __STDC_VERSION__ >= 199901L
 typedef int8_t orc_int8;
 typedef int16_t orc_int16;
 typedef int32_t orc_int32;
