Index: dvdread/src/dvdread/ifo_types.h
===================================================================
--- dvdread/src/dvdread/ifo_types.h	(révision 1188)
+++ dvdread/src/dvdread/ifo_types.h	(copie de travail)
@@ -32,7 +32,7 @@
 
 #if defined(__GNUC__)
 #if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 95)
-#define ATTRIBUTE_PACKED __attribute__ ((packed))
+#define ATTRIBUTE_PACKED __attribute__ ((packed,gcc_struct))
 #define PRAGMA_PACK 0
 #endif
 #endif
