--- dvdread/configure.ac.orig	2011-11-23 00:23:45.017580484 -0500
+++ dvdread/configure.ac	2011-11-23 00:24:23.381770721 -0500
@@ -170,9 +170,10 @@
     saved_CFLAGS=$CFLAGS
     CFLAGS="$CFLAGS -L$dvdcss_path/lib"
     AC_CHECK_LIB(dvdcss, dvdcss_interface_2,
-      [ CSS_LIBS="-L$dvdcss_path/lib -R$dvdcss_path/lib -ldvdcss"
+      [ CSS_LIBS="-L$dvdcss_path/lib -R$dvdcss_path/lib -ldvdcss -Wl,-framework,CoreFoundation -Wl,-framework,IOKit"
         CSS_CFLAGS=-I$dvdcss_path/include ],
-      AC_MSG_ERROR(You need libdvdcss.so.2 or later)
+      AC_MSG_ERROR(You need libdvdcss.so.2 or later),
+      [-Wl,-framework,CoreFoundation -Wl,-framework,IOKit]
     )
     CFLAGS=$saved_CFLAGS
     dnl -w added to shutup GCC3.1's cpp warning about -I/usr/local
