From 7d9b5ac491659360cce4ba912f31b7fa2a020c20 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C3=ABl=20Carr=C3=A9?= <funman@videolan.org>
Date: Tue, 19 Feb 2013 15:02:41 +0100
Subject: [PATCH] Use the correct libraries for linker test

---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 30cab8d..7a2917b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -186,7 +186,7 @@ dnl link with libdvdcss
     *darwin*) CSS_LIBS="${CSS_LIBS} -Wl,-framework,CoreFoundation -Wl,-framework,IOKit"
     ;;
   esac
-  AC_CHECK_LIB(dvdcss, dvdcss_interface_2,, AC_MSG_ERROR(You need libdvdcss.so.2 or later))
+  AC_CHECK_LIB(dvdcss, dvdcss_interface_2,, AC_MSG_ERROR(You need libdvdcss.so.2 or later), ${CSS_LIBS})
   AC_CHECK_HEADERS(dvdcss/dvdcss.h,, AC_MSG_ERROR(You need libdvdcss (dvdcss.h)))
 fi
 
-- 
1.7.10.4

