--- a/Framework/Source/GrowlApplicationBridge.m	2018-02-28 01:49:46.000000000 +0100
+++ b/Framework/Source/GrowlApplicationBridge.m	2018-02-28 01:52:48.000000000 +0100
@@ -730,7 +730,7 @@
 		regDict = [self bestRegistrationDictionary];
 	
 	if(!regDict){
-		NSLog(@"Cannot register without a registration dictionary!");
+		//NSLog(@"Cannot register without a registration dictionary!");
 		return NO;
 	}
 
@@ -815,8 +815,8 @@
 	NSDictionary *registrationDictionary = [self registrationDictionaryFromDelegate];
 	if (!registrationDictionary) {
 		registrationDictionary = [self registrationDictionaryFromBundle:nil];
-		if (!registrationDictionary)
-			NSLog(@"GrowlApplicationBridge: The Growl delegate did not supply a registration dictionary, and the app bundle at %@ does not have one. Please tell this application's developer.", [[NSBundle mainBundle] bundlePath]);
+		//if (!registrationDictionary)
+			//NSLog(@"GrowlApplicationBridge: The Growl delegate did not supply a registration dictionary, and the app bundle at %@ does not have one. Please tell this application's developer.", [[NSBundle mainBundle] bundlePath]);
 	}
 
 	return [self registrationDictionaryByFillingInDictionary:registrationDictionary];
