    
--- gettext/gettext-tools/libgrep/regex_internal.h.orig	2012-01-27 16:56:14.000000000 +0100
+++ gettext/gettext-tools/libgrep/regex_internal.h	2012-01-27 16:53:34.000000000 +0100
@@ -455,7 +455,9 @@
 #define re_string_skip_bytes(pstr,idx) ((pstr)->cur_idx += (idx))
 #define re_string_set_index(pstr,idx) ((pstr)->cur_idx = (idx))
 
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 
 #ifndef _LIBC
 # if HAVE_ALLOCA
