diff -urN gettext-orig/gettext-tools/gnulib-lib/localtime.c gettext/gettext-tools/gnulib-lib/localtime.c
--- gettext-orig/gettext-tools/gnulib-lib/localtime.c	2024-02-21 12:44:25.000000000 +0200
+++ gettext/gettext-tools/gnulib-lib/localtime.c	2025-09-16 13:24:27.574053984 +0300
@@ -63,13 +63,19 @@
       char **env = _environ;
       wchar_t **wenv = _wenviron;
       if (env != NULL)
-        for (char *s = env; *s != NULL; s++)
-          if (s[0] == 'T' && s[1] == 'Z' && s[2] == '=')
-            s[0] = '$';
+        for (char **ep = env; *ep != NULL; ep++)
+          {
+            char *s = *ep;
+            if (s[0] == 'T' && s[1] == 'Z' && s[2] == '=')
+              s[0] = '$';
+          }
       if (wenv != NULL)
-        for (wchar_t *ws = wenv; *ws != NULL; ws++)
-          if (ws[0] == L'T' && ws[1] == L'Z' && ws[2] == L'=')
-            ws[0] = L'$';
+        for (wchar_t **wep = wenv; *wep != NULL; wep++)
+          {
+            wchar_t *ws = *wep;
+            if (ws[0] == L'T' && ws[1] == L'Z' && ws[2] == L'=')
+              ws[0] = L'$';
+          }
     }
 #endif
 
