diff -urN gettext-0.22.5-orig/gettext-tools/gnulib-lib/unistd.in.h gettext-0.22.5/gettext-tools/gnulib-lib/unistd.in.h
--- gettext-0.22.5-orig/gettext-tools/gnulib-lib/unistd.in.h	2024-02-21 12:44:25
+++ gettext-0.22.5/gettext-tools/gnulib-lib/unistd.in.h	2025-08-15 14:19:45
@@ -52,7 +52,9 @@
 # endif
 # if (!defined MAC_OS_X_VERSION_MIN_REQUIRED \
       || MAC_OS_X_VERSION_MIN_REQUIRED < 99990000)
+#define openat real_openat
 #  include <sys/fcntl.h> /* It also defines the two macros.  */
+#undef openat
 #  undef SEEK_DATA
 #  undef SEEK_HOLE
 # endif
