--- libdca/libdca/Makefile.am.ln-s	2018-05-15 15:46:09.000000000 +0200
+++ libdca/libdca/Makefile.am	2018-05-22 15:01:28.117715200 +0200
@@ -12,7 +12,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libdca.pc libdts.pc
 
 install-exec-hook:
-	ln -sf "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
+	$(LN_S)f "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
 	unlink "$(DESTDIR)$(libdir)/libdts.a"
