--- libdca/libdca/Makefile.am.dtsln	2018-05-25 15:28:10.003436800 +0200
+++ libdca/libdca/Makefile.am	2018-05-25 15:28:15.347215500 +0200
@@ -12,7 +12,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libdca.pc libdts.pc
 
 install-exec-hook:
-	$(LN_S)f "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
+	$(LN_S)f "$(DESTDIR)$(libdir)/libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
 	unlink "$(DESTDIR)$(libdir)/libdts.a"
