--- asdcplib/src/KM_fileio.h.orig	2017-02-14 17:17:22.745776193 +0100
+++ asdcplib/src/KM_fileio.h	2017-02-14 17:17:29.713782420 +0100
@@ -37,7 +37,6 @@
 
 #ifdef KM_WIN32
 # include <io.h>
-# include "dirent_win.h"
 #else
 # include <dirent.h>
 # include <unistd.h>
@@ -46,6 +45,7 @@
 #include <regex.h>
 #endif
 
+#include <dirent.h>
 #include <sys/stat.h>
 
 
--- asdcplib/src/KM_fileio.cpp.orig	2017-02-14 17:16:18.297717750 +0100
+++ asdcplib/src/KM_fileio.cpp	2017-02-14 17:17:48.833799420 +0100
@@ -1478,7 +1478,7 @@
   if ( m_Handle == 0 )
     return RESULT_FILEOPEN;
 
-#if defined(__sun) && defined(__SVR4)
+#if (defined(__sun) && defined(__SVR4)) || defined(_WIN32)
   struct stat s;
 #endif
   struct dirent* entry;
@@ -1493,7 +1493,7 @@
 
   next_item_name.assign(entry->d_name, strlen(entry->d_name));
 
-#if defined(__sun) && defined(__SVR4)
+#if (defined(__sun) && defined(__SVR4)) || defined(_WIN32)
 
   stat(entry->d_name, &s);
 
@@ -1506,10 +1506,11 @@
     case S_IFREG:
       next_item_type = DET_FILE;
       break;
-
+#ifndef _WIN32
     case S_IFLNK:
       next_item_type = DET_LINK;
       break;
+#endif
 
     default:
       next_item_type = DET_DEV;
