From f88ec35dfd616ca94217658523ca5b8de0a2cb73 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Thu, 1 Mar 2018 11:12:10 +0100
Subject: [PATCH 2/2] Fix potential free of uninitialized pointer

Signed-off-by: Jean-Baptiste Kempf <jb@videolan.org>
---
 src/libdvdcss.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libdvdcss.c b/src/libdvdcss.c
index a659bf3..8f4c421 100644
--- a/src/libdvdcss.c
+++ b/src/libdvdcss.c
@@ -522,6 +522,7 @@ static dvdcss_t dvdcss_open_common ( const char *psz_target, void *p_stream,
     if( psz_target == NULL &&
       ( p_stream == NULL || p_stream_cb == NULL ) )
     {
+        dvdcss->psz_device = NULL;
         goto error;
     }
 
-- 
2.11.0

