--- gnutls-3.1.14/configure.ac.orig	2013-09-17 18:17:09.840217108 +0200
+++ gnutls-3.1.14/configure.ac	2013-09-17 18:19:36.609535012 +0200
@@ -26,7 +26,7 @@
 AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_HOST
 
-AM_INIT_AUTOMAKE([1.12.2 no-dist-gzip dist-xz dist-lzip -Wall -Wno-override])
+AM_INIT_AUTOMAKE([1.11.1 no-dist-gzip dist-xz -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_CONFIG_HEADERS([config.h])
 
