diff -ru gnutls.orig//lib/nettle/Makefile.am gnutls//lib/nettle/Makefile.am
--- gnutls.orig//lib/nettle/Makefile.am	2011-07-19 20:25:09.000000000 +0300
+++ gnutls//lib/nettle/Makefile.am	2011-07-19 20:25:39.000000000 +0300
@@ -35,4 +35,4 @@
 
 noinst_LTLIBRARIES = libcrypto.la
 
-libcrypto_la_SOURCES = pk.c mpi.c mac.c cipher.c rnd.c init.c egd.c egd.h
+libcrypto_la_SOURCES = pk.c mpi.c mac.c cipher.c rnd.c init.c
--- gnutls/lib/nettle/Makefile.in.orig	2011-11-22 23:27:22.099568057 -0500
+++ gnutls/lib/nettle/Makefile.in	2011-11-22 23:27:45.711685158 -0500
@@ -117,8 +117,7 @@
 CONFIG_CLEAN_VPATH_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
 libcrypto_la_LIBADD =
-am_libcrypto_la_OBJECTS = pk.lo mpi.lo mac.lo cipher.lo rnd.lo init.lo \
-	egd.lo
+am_libcrypto_la_OBJECTS = pk.lo mpi.lo mac.lo cipher.lo rnd.lo init.lo
 libcrypto_la_OBJECTS = $(am_libcrypto_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
@@ -829,7 +828,7 @@
 	-I$(srcdir)/../includes -I$(builddir)/../includes \
 	-I$(srcdir)/.. $(am__append_1)
 noinst_LTLIBRARIES = libcrypto.la
-libcrypto_la_SOURCES = pk.c mpi.c mac.c cipher.c rnd.c init.c egd.c egd.h
+libcrypto_la_SOURCES = pk.c mpi.c mac.c cipher.c rnd.c init.c
 all: all-am
 
 .SUFFIXES:
@@ -883,7 +882,6 @@
 	-rm -f *.tab.c
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cipher.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/egd.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/init.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/mac.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/mpi.Plo@am__quote@
diff -ru gnutls.orig//lib/nettle/rnd.c gnutls//lib/nettle/rnd.c
--- gnutls.orig//lib/nettle/rnd.c	2011-07-19 20:25:09.000000000 +0300
+++ gnutls//lib/nettle/rnd.c	2011-07-19 20:26:47.000000000 +0300
@@ -159,7 +159,7 @@
 #ifdef HAVE_GETRUSAGE
 #include <sys/resource.h>
 #endif
-#include "egd.h"
+//#include "egd.h"
 
 #define DEVICE_READ_SIZE 16
 #define DEVICE_READ_SIZE_MAX 32
@@ -288,6 +288,7 @@
   return 0;
 }
 
+#if 0
 static int
 do_device_source_egd (int init)
 {
@@ -342,6 +343,7 @@
     }
   return 0;
 }
+#endif
 
 static int
 do_device_source (int init)
@@ -359,11 +361,13 @@
 
       do_source = do_device_source_urandom;
       ret = do_source (init);
+#if 0
       if (ret < 0)
         {
           do_source = do_device_source_egd;
           ret = do_source (init);
         }
+#endif
 
       if (ret < 0)
         {
