--- gnutls-2.12.20/gl/gai_strerror.c.orig	2012-03-01 16:45:12.000000000 +0100
+++ gnutls-2.12.20/gl/gai_strerror.c	2012-09-27 14:39:30.273584236 +0200
@@ -75,7 +75,7 @@
     { EAI_IDN_ENCODE, N_("Parameter string not correctly encoded") }
 #endif
   };
-
+#ifndef _WIN32
 const char *
 gai_strerror (int code)
 {
@@ -89,4 +89,5 @@
 # ifdef _LIBC
 libc_hidden_def (gai_strerror)
 # endif
+#endif
 #endif /* !HAVE_DECL_GAI_STRERROR */
--- gnutls-3.1.10/lib/gnutls.pc.in.orig	2013-03-25 14:41:04.248360118 +0100
+++ gnutls-3.1.10/lib/gnutls.pc.in	2013-03-25 14:41:03.355020008 +0100
@@ -18,7 +18,7 @@
 Description: Transport Security Layer implementation for the GNU system
 URL: http://www.gnutls.org/
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls
+Libs: -L${libdir} -lgnutls -lws2_32 -lcrypt32 @LTLIBINTL@
 Libs.private: @LTLIBNETTLE@ @LTLIBZ@ @LTLIBINTL@ @LIBSOCKET@ @LTLIBPTHREAD@ @LTLIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @LIB_CLOCK_GETTIME@
 @GNUTLS_REQUIRES_PRIVATE@
 Cflags: -I${includedir}
--- gnutls/gl/msvc-inval.c.orig	2013-07-02 17:57:58.000000000 +0200
+++ gnutls/gl/msvc-inval.c	2013-07-02 17:58:05.000000000 +0200
@@ -28,7 +28,7 @@
 
 # if MSVC_INVALID_PARAMETER_HANDLING == DEFAULT_HANDLING
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
