--- gnutls-2.12.20/gl/gai_strerror.c.orig	2012-03-01 16:45:12.000000000 +0100
+++ gnutls-2.12.20/gl/gai_strerror.c	2012-09-27 14:39:30.273584236 +0200
@@ -75,7 +75,7 @@
     { EAI_IDN_ENCODE, N_("Parameter string not correctly encoded") }
 #endif
   };
-
+#ifndef _WIN32
 const char *
 gai_strerror (int code)
 {
@@ -89,4 +89,5 @@
 # ifdef _LIBC
 libc_hidden_def (gai_strerror)
 # endif
+#endif
 #endif /* !HAVE_DECL_GAI_STRERROR */
