--- gnutls-2.2.5/gl/gai_strerror.c	2008-02-21 08:58:18.000000000 +0100
+++ gnutls/gl/gai_strerror.c	2008-05-22 08:20:56.000000000 +0200
@@ -62,7 +62,7 @@
     { EAI_IDN_ENCODE, N_("Parameter string not correctly encoded") }
 #endif
   };
-
+#ifndef _WIN32
 const char *
 gai_strerror (int code)
 {
@@ -73,6 +73,7 @@
 
   return _("Unknown error");
 }
+#endif
 #ifdef _LIBC
 libc_hidden_def (gai_strerror)
 #endif

