--- gnutls-2.12.20/gl/gai_strerror.c.orig	2012-03-01 16:45:12.000000000 +0100
+++ gnutls-2.12.20/gl/gai_strerror.c	2012-09-27 14:39:30.273584236 +0200
@@ -75,7 +75,7 @@
     { EAI_IDN_ENCODE, N_("Parameter string not correctly encoded") }
 #endif
   };
-
+#ifndef _WIN32
 const char *
 gai_strerror (int code)
 {
@@ -89,4 +89,5 @@
 # ifdef _LIBC
 libc_hidden_def (gai_strerror)
 # endif
+#endif
 #endif /* !HAVE_DECL_GAI_STRERROR */
--- gnutls-3.1.14/lib/gnutls.pc.in.orig	2013-09-17 18:14:16.270374773 +0200
+++ gnutls-3.1.14/lib/gnutls.pc.in	2013-09-17 18:16:10.232464936 +0200
@@ -18,7 +18,7 @@
 Description: Transport Security Layer implementation for the GNU system
 URL: http://www.gnutls.org/
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls
+Libs: -L${libdir} -lgnutls -lws2_32 -lcrypt32 @LTLIBINTL@
 Libs.private: @LTLIBNETTLE@ @LTLIBZ@ @LTLIBINTL@ @LIBSOCKET@ @LTLIBPTHREAD@ @LTLIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @LIB_CLOCK_GETTIME@ @GMP_LIBS@
 @GNUTLS_REQUIRES_PRIVATE@
 Cflags: -I${includedir}
