--- gnutls-3.5.16/lib/system/keys-win.c	2017-12-26 16:29:00.221119300 +0100
+++ gnutls-3.5.16/lib/system/keys-win.c.new	2017-12-26 16:33:02.501126487 +0100
@@ -44,6 +44,18 @@
 #include <wincrypt.h>
 #include <winbase.h>
 
+#include <winapifamily.h>
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+# if defined(_WIN32_WINNT) && _WIN32_WINNT >= 0x0A00 /* Univeral Winstore */
+# undef CertOpenSystemStore
+HCERTSTORE CertOpenSystemStore( HCRYPTPROV_LEGACY hprov, LPCSTR szSubsystemProtocol )
+{
+    return CertOpenStore( CERT_STORE_PROV_SYSTEM_A, X509_ASN_ENCODING, 0,
+        CERT_SYSTEM_STORE_CURRENT_USER, szSubsystemProtocol );
+}
+# endif /* _WIN32_WINNT */
+#endif /* WINAPI_FAMILY */
+
 #define DYN_NCRYPT
 
 #ifdef __MINGW32__
--- gnutls/lib/system/keys-win.c.old	2017-12-28 11:04:45.599954683 +0100
+++ gnutls/lib/system/keys-win.c	2017-12-28 11:07:37.999947957 +0100
@@ -624,6 +624,9 @@
  -*/
 int _gnutls_privkey_import_system_url(gnutls_privkey_t pkey, const char *url)
 {
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+    return GNUTLS_E_UNIMPLEMENTED_FEATURE;
+#else
 	uint8_t id[MAX_WID_SIZE];
 	HCERTSTORE store = NULL;
 	size_t id_size;
@@ -873,6 +876,7 @@
 
 	CertCloseStore(store, 0);
 	return ret;
+#endif
 }
 
 int _gnutls_x509_crt_import_system_url(gnutls_x509_crt_t crt, const char *url)
