diff -ru gnutls/libdane/gnutls-dane.pc.in gnutls-fixed/libdane/gnutls-dane.pc.in
--- gnutls/libdane/gnutls-dane.pc.in	2015-03-23 08:19:23.000000000 +0100
+++ gnutls-fixed/libdane/gnutls-dane.pc.in	2016-06-04 14:25:53.000000000 +0200
@@ -18,7 +18,7 @@
 Description: DANE security library for the GNU system
 URL: http://www.gnu.org/software/gnutls/
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls-dane
+Libs: -L${libdir} -lgnutls-dane @INTL_MACOSX_LIBS@
 Libs.private: @UNBOUND_LIBS@
 Requires.private: gnutls
 Cflags: -I${includedir}
diff -ru gnutls/m4/intlmacosx.m4 gnutls-fixed/m4/intlmacosx.m4
--- gnutls/m4/intlmacosx.m4	2015-03-23 19:23:36.000000000 +0100
+++ gnutls-fixed/m4/intlmacosx.m4	2016-06-04 14:25:53.000000000 +0200
@@ -43,9 +43,25 @@
     AC_DEFINE([HAVE_CFLOCALECOPYCURRENT], [1],
       [Define to 1 if you have the MacOS X function CFLocaleCopyCurrent in the CoreFoundation framework.])
   fi
+  AC_CACHE_CHECK([for SecTrustCopyAnchorCertificates],
+    [SecTrustCopyAnchorCertificates],
+    [gt_save_LIBS="$LIBS"
+     LIBS="$LIBS -Wl,-framework -Wl,Security"
+     AC_TRY_LINK([#include <Security/SecTrust.h>],
+       [SecTrustCopyAnchorCertificates(NULL)],
+       [gt_cv_func_SecTrustCopyAnchorCertificates=yes],
+       [gt_cv_func_SecTrustCopyAnchorCertificates=no])
+     LIBS="$gt_save_LIBS"])
+  if test $gt_cv_func_SecTrustCopyAnchorCertificates = yes; then
+    AC_DEFINE([HAVE_SecTrustCopyAnchorCertificates], [1],
+      [Define to 1 if you have the MacOS X function SecTrustCopyAnchorCertificates in the Security framework.])
+  fi
   INTL_MACOSX_LIBS=
   if test $gt_cv_func_CFPreferencesCopyAppValue = yes || test $gt_cv_func_CFLocaleCopyCurrent = yes; then
-    INTL_MACOSX_LIBS="-Wl,-framework -Wl,CoreFoundation"
+    INTL_MACOSX_LIBS+="-Wl,-framework -Wl,CoreFoundation "
+  fi
+  if test $gt_cv_func_SecTrustCopyAnchorCertificates = yes; then
+    INTL_MACOSX_LIBS+="-Wl,-framework -Wl,Security "
   fi
   AC_SUBST([INTL_MACOSX_LIBS])
 ])
