--- gnutls-3.4.13/lib/system.c.orig	2016-05-27 22:40:06.000000000 +0200
+++ gnutls-3.4.13/lib/system.c	2016-06-29 15:36:26.645190212 +0200
@@ -34,7 +34,7 @@
 #ifdef _WIN32
 # include <windows.h>
 # include <wincrypt.h>
-# if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+# if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 typedef PCCRL_CONTEXT WINAPI(*CertEnumCRLsInStoreFunc) (HCERTSTORE
 							 hCertStore,
 							 PCCRL_CONTEXT
@@ -300,7 +300,7 @@
 int gnutls_system_global_init(void)
 {
 #ifdef _WIN32
-#if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 	HMODULE crypto;
 	crypto = LoadLibraryA("Crypt32.dll");
 
@@ -325,7 +325,7 @@
 void gnutls_system_global_deinit(void)
 {
 #ifdef _WIN32
-#if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 	FreeLibrary(Crypt32_dll);
 #endif
 #endif
