--- gnutls/lib/system.c.orig	2016-07-08 14:39:45.104700999 +0200
+++ gnutls/lib/system.c	2016-07-08 14:40:28.276750643 +0200
@@ -322,7 +322,7 @@
 #ifdef _WIN32
 #if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 	HMODULE crypto;
-	crypto = LoadLibraryA("Crypt32.dll");
+	crypto = LoadLibrary(TEXT("Crypt32.dll"));
 
 	if (crypto == NULL)
 		return GNUTLS_E_CRYPTO_INIT_FAILED;
--- gnutls/lib/system-keys-win.c.orig	2016-07-08 14:39:58.336716261 +0200
+++ gnutls/lib/system-keys-win.c	2016-07-08 14:40:12.568732630 +0200
@@ -1059,7 +1059,7 @@
 	int ret;
 
 #ifdef DYN_NCRYPT
-	ncrypt_lib = LoadLibraryA("ncrypt.dll");
+	ncrypt_lib = LoadLibrary(TEXT("ncrypt.dll"));
 	if (ncrypt_lib == NULL) {
 		return gnutls_assert_val(GNUTLS_E_CRYPTO_INIT_FAILED);
 	}
