--- gnutls-3.5.16/lib/system/keys-win.c	2016-11-05 19:12:02.000000000 +0100
+++ gnutls-3.5.16/lib/system/keys-win.c.new	2017-12-26 16:29:00.221119300 +0100
@@ -1426,7 +1426,7 @@
 	int ret;
 
 #ifdef DYN_NCRYPT
-	ncrypt_lib = LoadLibraryA("ncrypt.dll");
+	ncrypt_lib = LoadLibrary(TEXT("ncrypt.dll"));
 	if (ncrypt_lib == NULL) {
 		return gnutls_assert_val(GNUTLS_E_CRYPTO_INIT_FAILED);
 	}
--- gnutls/lib/system.c.old	2017-12-28 15:24:07.479820602 +0100
+++ gnutls/lib/system.c	2017-12-28 15:25:33.587817242 +0100
@@ -50,7 +50,7 @@
 #if defined(_WIN32) && defined(NEED_CERT_ENUM_CRLS)
 	/* used in system/certs.c */
 	HMODULE crypto;
-	crypto = LoadLibraryA("Crypt32.dll");
+	crypto = LoadLibrary(TEXT("Crypt32.dll"));
 
 	if (crypto == NULL)
 		return GNUTLS_E_CRYPTO_INIT_FAILED;
