--- gnutls/lib/system.c.orig	2016-04-01 17:15:48.424481273 +0200
+++ gnutls/lib/system.c	2016-04-01 17:15:57.200493470 +0200
@@ -34,7 +34,7 @@
 #ifdef _WIN32
 # include <windows.h>
 # include <wincrypt.h>
-# if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+# if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 typedef PCCRL_CONTEXT WINAPI(*Type_CertEnumCRLsInStore) (HCERTSTORE
 							 hCertStore,
 							 PCCRL_CONTEXT
@@ -289,7 +289,7 @@
 int gnutls_system_global_init()
 {
 #ifdef _WIN32
-#if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 	HMODULE crypto;
 	crypto = LoadLibraryA("Crypt32.dll");
 
@@ -313,7 +313,7 @@
 void gnutls_system_global_deinit()
 {
 #ifdef _WIN32
-#if defined(__MINGW32__) && !defined(__MINGW64__) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) && __MINGW32_MAJOR_VERSION <= 3 && __MINGW32_MINOR_VERSION <= 20
 	FreeLibrary(Crypt32_dll);
 #endif
 #endif
