--- gnutls/gl/read-file.c.orig	2012-03-06 20:59:29.600593329 -0500
+++ gnutls/gl/read-file.c	2012-03-06 20:59:44.568593328 -0500
@@ -35,6 +35,9 @@
 /* Get errno. */
 #include <errno.h>
 
+/* Get SIZE_MAX */
+#include <limits.h>
+
 /* Read a STREAM and return a newly allocated string with the content,
    and set *LENGTH to the length of the string.  The string is
    zero-terminated, but the terminating zero byte is not counted in
