--- gnutls-3.5.19/gl/vasnprintf.c.orig	2018-12-19 10:14:55.478883083 +0100
+++ gnutls-3.5.19/gl/vasnprintf.c	2018-12-19 10:15:22.798882978 +0100
@@ -4858,7 +4858,7 @@
 #endif
                   *fbp = dp->conversion;
 #if USE_SNPRINTF
-# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
+# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__) || defined __ANDROID__ )
                 fbp[1] = '%';
                 fbp[2] = 'n';
                 fbp[3] = '\0';
