--- gnutls-3.5.16/lib/x509/verify-high2.c.orig	2016-12-19 15:10:59.000000000 +0100
+++ gnutls-3.5.16/lib/x509/verify-high2.c	2017-12-26 16:39:02.565137168 +0100
@@ -43,6 +43,9 @@
 # endif
 #endif
 
+#include <windows.h>
+#include <tchar.h>
+
 /* Convenience functions for verify-high functionality 
  */
 
@@ -368,29 +371,59 @@
 	return ret;
 }
 
+static inline char *FromWide (const wchar_t *wide)
+{
+    size_t len = WideCharToMultiByte (CP_UTF8, 0, wide, -1, NULL, 0, NULL, NULL);
+    if (len == 0)
+        return NULL;
+
+    char *out = (char *)malloc (len);
+
+    if (out)
+        WideCharToMultiByte (CP_UTF8, 0, wide, -1, out, len, NULL, NULL);
+    return out;
+}
+
+static inline wchar_t *ToWide (const char *utf8)
+{
+    int len = MultiByteToWideChar (CP_UTF8, 0, utf8, -1, NULL, 0);
+    if (len == 0)
+        return NULL;
+
+    wchar_t *out = (wchar_t *)malloc (len * sizeof (wchar_t));
+
+    if (out)
+        MultiByteToWideChar (CP_UTF8, 0, utf8, -1, out, len);
+    return out;
+}
+
 static
 int load_dir_certs(const char *dirname,
 			  gnutls_x509_trust_list_t list,
 			  unsigned int tl_flags, unsigned int tl_vflags,
 			  unsigned type, unsigned crl)
 {
-	DIR *dirp;
-	struct dirent *d;
+	_TDIR *dirp;
+	struct _tdirent *d;
 	int ret;
 	int r = 0;
 	char path[GNUTLS_PATH_MAX];
 
-	dirp = opendir(dirname);
+	TCHAR* dirnameW = ToWide(dirname);
+	dirp = _topendir(dirnameW);
+	free(dirnameW);
 	if (dirp != NULL) {
 		do {
-			d = readdir(dirp);
+			d = _treaddir(dirp);
 			if (d != NULL
 #ifdef _DIRENT_HAVE_D_TYPE
 				&& (d->d_type == DT_REG || d->d_type == DT_LNK || d->d_type == DT_UNKNOWN)
 #endif
 			) {
+                char* d_name = FromWide(d->d_name);
 				snprintf(path, sizeof(path), "%s/%s",
-					 dirname, d->d_name);
+					 dirname, d_name);
+				free(d_name);
 
 				if (crl != 0) {
 					ret =
@@ -408,7 +441,7 @@
 			}
 		}
 		while (d != NULL);
-		closedir(dirp);
+		_tclosedir(dirp);
 	}
 
 	return r;
