--- libshout/m4/xiph_net.m4.old	2018-12-03 14:41:36.427782211 +0100
+++ libshout/m4/xiph_net.m4	2018-12-03 14:41:43.235782185 +0100
@@ -7,7 +7,7 @@
 AC_CHECK_HEADERS([sys/select.h sys/uio.h])
 AC_CHECK_HEADER([winsock2.h],
   [AC_DEFINE([HAVE_WINSOCK2_H], [1], [Define if you have winsock2.h on MINGW])
-   LIBS="$LIBS -lwsock32"])
+   ])
 
 # These tests are ordered based on solaris 8 tests
 AC_SEARCH_LIBS([sethostent], [nsl],
