--- libshout/src/common/net/resolver.c.ws2tcpip	2015-01-27 19:08:54.000000000 +0100
+++ libshout/src/common/net/resolver.c	2019-03-05 12:02:46.595628700 +0100
@@ -39,6 +39,7 @@
 #include <arpa/inet.h>
 #else
 #include <winsock2.h>
+#include <ws2tcpip.h>
 #endif
 
 #ifndef NO_THREAD
--- libshout.nok/src/common/net/sock.c.ws2tcpip	2019-03-06 13:20:11.624823900 +0100
+++ libshout.nok/src/common/net/sock.c	2019-03-07 07:21:23.841338100 +0100
@@ -57,6 +57,7 @@
 #include <netdb.h>
 #else
 #include <winsock2.h>
+#include <ws2tcpip.h>
 #endif
 
 #include "sock.h"
