From 4bb53ad8e35f899f14e6d6144e60d8caa5c0651c Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Mon, 6 Sep 2021 11:49:03 +0200
Subject: [PATCH 1/1] ffmpeg: add target_os support for emscripten

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index d7a3f50..57ca005 100755
--- a/configure
+++ b/configure
@@ -5579,6 +5579,8 @@ case $target_os in
         ;;
     minix)
         ;;
+    emscripten)
+	;;
     none)
         ;;
     *)
-- 
2.33.0

