From 6c9566016d1be55ff159d884ac3498dc2d1b7857 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Mon, 6 Sep 2021 11:49:03 +0200
Subject: [PATCH 10/10] ffmpeg: add target_os support for emscripten

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index a89cfa6d95..a3ba6eeae9 100755
--- a/configure
+++ b/configure
@@ -5802,6 +5802,8 @@ case $target_os in
         ;;
     minix)
         ;;
+    emscripten)
+	;;
     none)
         ;;
     *)
-- 
2.37.3.windows.1

