Index: ffmpeg-svn/libswscale/swscale_template.c
===================================================================
--- ffmpeg-svn/libswscale/swscale_template.c	(revision 31036)
+++ ffmpeg-svn/libswscale/swscale_template.c	(working copy)
@@ -2466,6 +2466,7 @@
         }
     } else {
 #endif /* COMPILE_TEMPLATE_MMX2 */
+#if 0
         x86_reg xInc_shr16 = (x86_reg) (xInc >> 16);
         uint16_t xInc_mask = xInc & 0xffff;
         __asm__ volatile(
@@ -2501,6 +2502,7 @@
             "r" (src2)
             : "%"REG_a, "%"REG_d, "%ecx", "%"REG_D, "%esi"
         );
+#endif
 #if COMPILE_TEMPLATE_MMX2
     } //if MMX2 can't be used
 #endif
