--- ffmpeg/compat/w32dlfcn.h.old	2017-06-02 14:35:36.726230731 +0200
+++ ffmpeg/compat/w32dlfcn.h	2017-06-02 15:14:38.472865111 +0200
@@ -71,7 +71,11 @@
 #ifndef LOAD_LIBRARY_SEARCH_SYSTEM32
 #   define LOAD_LIBRARY_SEARCH_SYSTEM32        0x00000800
 #endif
+#if WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP)
     return LoadLibraryExA(name, NULL, LOAD_LIBRARY_SEARCH_APPLICATION_DIR | LOAD_LIBRARY_SEARCH_SYSTEM32);
+#else
+    return NULL;
+#endif
 }
 #define dlopen(name, flags) win32_dlopen(name)
 #define dlclose FreeLibrary
