--- ffmpeg-git/configure.orig	2011-11-24 19:07:09.080887690 -0500
+++ ffmpeg-git/configure	2011-11-24 19:07:36.597024164 -0500
@@ -3000,6 +3000,7 @@
     if check_header VideoDecodeAcceleration/VDADecoder.h; then
         enable vda
         add_ldflags -framework CoreFoundation -framework VideoDecodeAcceleration -framework QuartzCore
+        add_extralibs -framework CoreFoundation -framework VideoDecodeAcceleration -framework QuartzCore
     fi
 fi
 
