--- libpng-1.6.8/configure.ac.orig	2014-02-08 15:42:46.399091835 +0100
+++ libpng-1.6.8/configure.ac	2014-02-08 15:47:42.684534041 +0100
@@ -27,7 +27,7 @@
 # dist-xz requires automake 1.11 or later
 # 1.12.2 fixes a security issue in 1.11.2 and 1.12.1
 # 1.13 is required for parallel tests
-AM_INIT_AUTOMAKE([1.13 foreign dist-xz color-tests silent-rules subdir-objects])
+AM_INIT_AUTOMAKE([1.11 foreign dist-xz color-tests silent-rules subdir-objects])
 # The following line causes --disable-maintainer-mode to be the default to
 # configure, this is necessary because libpng distributions cannot rely on the
 # time stamps of the autotools generated files being correct
