--- libpng-1.6.10/Makefile.am.orig	2014-05-24 18:11:01.881964331 +0200
+++ libpng-1.6.10/Makefile.am	2014-05-24 18:12:02.839819773 +0200
@@ -7,10 +7,10 @@
 ACLOCAL_AMFLAGS = -I scripts
 
 # test programs - run on make check, make distcheck
-check_PROGRAMS= pngtest pngunknown pngstest pngvalid pngimage
+check_PROGRAMS= 
 
 # Utilities - installed
-bin_PROGRAMS= pngfix png-fix-itxt
+bin_PROGRAMS= 
 
 # This ensures that pnglibconf.h gets built at the start of 'make all' or
 # 'make check', but it does not add dependencies to the individual programs,
