diff -urN x265-orig/source/CMakeLists.txt x265/source/CMakeLists.txt
--- x265-orig/source/CMakeLists.txt	2017-12-14 10:04:53.932080387 +0000
+++ x265/source/CMakeLists.txt	2017-12-14 10:05:35.151167937 +0000
@@ -441,7 +441,7 @@
     ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 install(FILES x265.h "${PROJECT_BINARY_DIR}/x265_config.h" DESTINATION include)
 
-if(CMAKE_RC_COMPILER)
+if(FALSE)
     # The resource compiler does not need CFLAGS or macro defines. It
     # often breaks them
     string(REPLACE "<FLAGS>" "" CMAKE_RC_COMPILE_OBJECT "${CMAKE_RC_COMPILE_OBJECT}")
