--- x265/source/CMakeLists.txt.orig	2022-03-20 16:40:01.560434700 +0000
+++ x265/source/CMakeLists.txt	2022-03-20 16:41:20.563787600 +0000
@@ -713,7 +713,7 @@
         if(WIN32 OR NOT ENABLE_SHARED OR INTEL_CXX)
             # The CLI cannot link to the shared library on Windows, it
             # requires internal APIs not exported from the DLL
-            target_link_libraries(cli x265-static ${PLATFORM_LIBS})
+            target_link_libraries(cli x265-static ${PLATFORM_LIBS} ${CMAKE_DL_LIBS})
         else()
             target_link_libraries(cli x265-shared ${PLATFORM_LIBS})
         endif()
