*** openjpeg/libopenjpeg/opj_malloc.h.orig	2012-01-27 16:25:43.875875000 -0500
--- openjpeg/libopenjpeg/opj_malloc.h	2012-01-27 16:25:43.875875000 -0500
***************
*** 83,88 ****
--- 83,90 ----
  #else /* Not _WIN32 */
  	#if defined(__sun)
  		#define HAVE_MEMALIGN
+ 	#elif defined(__FreeBSD__)
+ 		#define HAVE_POSIX_MEMALIGN
  	/* Linux x86_64 and OSX always align allocations to 16 bytes */
  	#elif !defined(__amd64__) && !defined(__APPLE__)	
  		#define HAVE_MEMALIGN
