--- qt/src/corelib/tools/qsimd.cpp.mingw32	2020-01-24 06:54:31.000000000 +0100
+++ qt/src/corelib/tools/qsimd.cpp	2020-03-11 07:59:52.182784800 +0100
@@ -621,7 +621,7 @@ void qDumpCPUFeatures()
     puts("");
 }
 
-#if defined(Q_PROCESSOR_X86) && QT_COMPILER_SUPPORTS_HERE(RDRND)
+#if defined(Q_PROCESSOR_X86) && QT_COMPILER_SUPPORTS_HERE(RDRND) && 0 // mingw32 32-bits crash
 
 #  ifdef Q_PROCESSOR_X86_64
 #    define _rdrandXX_step _rdrand64_step
--- qt/src/corelib/tools/qsimd_p.h.mingw32	2020-01-24 06:54:31.000000000 +0100
+++ qt/src/corelib/tools/qsimd_p.h	2020-03-11 07:59:43.086627000 +0100
@@ -346,7 +346,7 @@ extern Q_CORE_EXPORT QBasicAtomicInteger
 #endif
 Q_CORE_EXPORT void qDetectCpuFeatures();
 
-#if defined(Q_PROCESSOR_X86) && QT_COMPILER_SUPPORTS_HERE(RDRND)
+#if defined(Q_PROCESSOR_X86) && QT_COMPILER_SUPPORTS_HERE(RDRND) && 0 // mingw32 32-bits crash
 Q_CORE_EXPORT qsizetype qRandomCpu(void *, qsizetype) Q_DECL_NOTHROW;
 #else
 static inline qsizetype qRandomCpu(void *, qsizetype) Q_DECL_NOTHROW
