From d05eac31981d69c4b29aaf38f9f0addb2b549fae Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Tue, 9 Jan 2024 20:59:45 +0200
Subject: [PATCH 2/8] Windows Tray Icon Set NOSOUND

---
 src/plugins/platforms/windows/qwindowssystemtrayicon.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/plugins/platforms/windows/qwindowssystemtrayicon.cpp b/src/plugins/platforms/windows/qwindowssystemtrayicon.cpp
index acad849a3d..fae00c2081 100644
--- a/src/plugins/platforms/windows/qwindowssystemtrayicon.cpp
+++ b/src/plugins/platforms/windows/qwindowssystemtrayicon.cpp
@@ -214,6 +214,9 @@ void QWindowsSystemTrayIcon::showMessage(const QString &title, const QString &me
     qStringToLimitedWCharArray(message, tnd.szInfo, 256);
     qStringToLimitedWCharArray(title, tnd.szInfoTitle, 64);
 
+    // No audio:
+    tnd.dwInfoFlags |= NIIF_NOSOUND;
+
     tnd.uID = q_uNOTIFYICONID;
 
     const auto size = icon.actualSize(QSize(256, 256));
-- 
2.43.1

