diff -ruN qt/mkspecs/win32-g++/qmake.conf qt.new/mkspecs/win32-g++/qmake.conf
--- qt/mkspecs/win32-g++/qmake.conf	2016-03-03 12:35:22.000000000 +0100
+++ qt.new/mkspecs/win32-g++/qmake.conf	2016-08-06 11:33:47.010286322 +0200
@@ -12,7 +12,7 @@
 
 MAKEFILE_GENERATOR      = MINGW
 QMAKE_PLATFORM          = win32 mingw
-CONFIG                 += debug_and_release debug_and_release_target precompile_header
+CONFIG                 += debug_and_release debug_and_release_target
 DEFINES                += UNICODE
 QMAKE_COMPILER_DEFINES += __GNUC__ WIN32
 
