--- qt/src/corelib/global/qfloat16.h.old	2022-01-18 17:46:03.821787862 +0100
+++ qt/src/corelib/global/qfloat16.h	2022-01-18 17:46:52.824906487 +0100
@@ -44,6 +44,7 @@
 #include <QtCore/qglobal.h>
 #include <QtCore/qmetatype.h>
 #include <string.h>
+#include <limits>
 
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__AVX2__) && !defined(__F16C__)
 // All processors that support AVX2 do support F16C too. That doesn't mean
--- qt/src/corelib/text/qbytearraymatcher.h.old	2022-01-18 17:47:49.667887059 +0100
+++ qt/src/corelib/text/qbytearraymatcher.h	2022-01-18 17:48:09.055540031 +0100
@@ -41,6 +41,7 @@
 #define QBYTEARRAYMATCHER_H
 
 #include <QtCore/qbytearray.h>
+#include <limits>
 
 QT_BEGIN_NAMESPACE
 
--- qt/src/3rdparty/angle/src/libANGLE/HandleAllocator.cpp.old	2022-01-18 17:55:16.527952044 +0100
+++ qt/src/3rdparty/angle/src/libANGLE/HandleAllocator.cpp	2022-01-18 17:55:28.315744081 +0100
@@ -10,6 +10,7 @@
 #include "libANGLE/HandleAllocator.h"
 
 #include <algorithm>
+#include <limits>
 
 #include "common/debug.h"
 
