--- a/qmake/generators/makefile.cpp	2020-09-02 12:15:07.000000000 +0200
+++ b/qmake/generators/makefile.cpp	2022-09-12 11:42:01.170411450 +0200
@@ -3431,11 +3431,18 @@
                                                + "," + windowsifyPath(replace.toQString()) + ",gi");
             }
         }
-        if (sedargs.isEmpty()) {
-            ret += "$(INSTALL_FILE) " + escapeFilePath(src) + ' ' + escapeFilePath(dst);
-        } else {
-            ret += "$(SED) " + sedargs + ' ' + escapeFilePath(src) + " > " + escapeFilePath(dst);
+
+        QString escaped_src = escapeFilePath(src);
+        if (!sedargs.isEmpty()) {
+            int pos = src.lastIndexOf('.');
+            QString filename = src.left(pos);
+            QString ext = src.mid(pos + 1);
+            escaped_src = escapeFilePath(filename + ".qmakereplace." + ext);
+            ret += "$(SED) " + sedargs + ' ' + escapeFilePath(src) + " > "
+                   + escaped_src + " && ";
         }
+
+        ret += "$(INSTALL_FILE) " + escaped_src + ' ' + escapeFilePath(dst);
     }
     return ret;
 }
