--- a/mkspecs/features/qml_plugin.prf	2022-09-02 14:15:09.023765377 +0200
+++ b/mkspecs/features/qml_plugin.prf	2022-09-02 14:59:06.296938907 +0200
@@ -30,9 +30,9 @@
 }
 isEmpty(TARGETPATH): TARGETPATH = $$eval(QT.$${CXX_MODULE}.name)
 
+isEmpty(VERSION): VERSION = $$MODULE_VERSION
 win32:CONFIG(shared, static|shared) {
     # Embed a VERSIONINFO resource into the plugin's DLL.
-    isEmpty(VERSION): VERSION = $$MODULE_VERSION
     CONFIG += skip_target_version_ext
 }
 
--- a/mkspecs/features/qt_plugin.prf	2022-09-02 14:15:04.593670115 +0200
+++ b/mkspecs/features/qt_plugin.prf	2022-09-02 14:59:06.296938907 +0200
@@ -17,9 +17,9 @@
 CONFIG += plugin create_pc
 DESTDIR = $$MODULE_BASE_OUTDIR/plugins/$$PLUGIN_TYPE
 
+isEmpty(VERSION): VERSION = $$QT_VERSION
 win32:CONFIG(shared, static|shared) {
     # Embed a VERSIONINFO resource into the plugin's DLL.
-    isEmpty(VERSION): VERSION = $$QT_VERSION
     CONFIG += skip_target_version_ext
 }
 
