--- libcaca/caca/caca.h 2011-07-05 00:09:51.000000000 -0700
+++ libcaca.new/caca/caca.h 2011-07-05 00:10:10.000000000 -0700
@@ -645,7 +645,7 @@
 #       define CACA_DEPRECATED
 #   endif
 
-#   if defined __GNUC__ && __GNUC__ > 3
+#   if !(defined __APPLE__ && defined __llvm__) && defined __GNUC__ && __GNUC__ > 3
 #       define CACA_ALIAS(x) __attribute__ ((weak, alias(#x)))
 #   else
 #       define CACA_ALIAS(x)
