diff --git a/builds/windows/ftsystem.c b/builds/windows/ftsystem.c
index 1ebadd49f..ac8ff1f92 100644
--- a/builds/windows/ftsystem.c
+++ b/builds/windows/ftsystem.c
@@ -28,6 +28,7 @@
   /* memory mapping and allocation includes and definitions */
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
+#include <winapifamily.h>
 
 
   /**************************************************************************
@@ -296,7 +297,11 @@
       goto Fail_Open;
     }
 
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
     fm = CreateFileMapping( file, NULL, PAGE_READONLY, 0, 0, NULL );
+#else
+    fm = CreateFileMappingFromApp( file, NULL, PAGE_READONLY, 0, NULL );
+#endif
     if ( fm == NULL )
     {
       FT_ERROR(( "FT_Stream_Open: can not map file\n" ));
@@ -308,8 +313,13 @@
     /* a size greater than LONG_MAX                                    */
     stream->size = size.LowPart;
     stream->pos  = 0;
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
     stream->base = (unsigned char *)
                      MapViewOfFile( fm, FILE_MAP_READ, 0, 0, 0 );
+#else
+    stream->base = (unsigned char *)
+                     MapViewOfFileFromApp( fm, FILE_MAP_READ, 0, 0 );
+#endif
 
     CloseHandle( fm );
 
