--- qt-everywhere-opensource-src-4.7.4/configure.orig	2011-09-28 14:31:35.039958884 -0400
+++ qt-everywhere-opensource-src-4.7.4/configure	2011-09-28 14:31:48.649958702 -0400
@@ -4568,7 +4568,7 @@
 }
 
 # build qmake
-if true; then ###[ '!' -f "$outpath/bin/qmake" ];
+if [ '!' -f "$outpath/bin/qmake" ]; then
     echo "Creating qmake. Please wait..."
 
     OLD_QCONFIG_H=
--- qt-everywhere-opensource-src-4.7.4/src/tools/tools.pro.orig	2011-09-28 14:33:24.119957315 -0400
+++ qt-everywhere-opensource-src-4.7.4/src/tools/tools.pro	2011-09-28 14:33:56.619956845 -0400
@@ -1,11 +1,6 @@
 TEMPLATE = subdirs
 
-TOOLS_SUBDIRS = src_tools_bootstrap src_tools_moc src_tools_rcc
-!contains(QT_CONFIG, no-gui): TOOLS_SUBDIRS += src_tools_uic
-!cross_compile {
-    contains(QT_CONFIG, qt3support): SRC_SUBDIRS += src_tools_uic3
-    win32:!wince*: SRC_SUBDIRS += src_tools_idc
-}
+TOOLS_SUBDIRS = src_tools_bootstrap 
 
 # Set subdir and respective target name
 src_tools_bootstrap.subdir = $$QT_SOURCE_TREE/src/tools/bootstrap
--- qt-everywhere-opensource-src-4.7.4/mkspecs/win32-g++/qmake.conf.orig	2011-09-28 14:32:02.709958498 -0400
+++ qt-everywhere-opensource-src-4.7.4/mkspecs/win32-g++/qmake.conf	2011-09-28 14:36:32.529954583 -0400
@@ -1,5 +1,5 @@
 #
-# qmake configuration for win32-g++
+# qmake configuration for win32-x86_64-w64-mingw32-g++
 #
 # Written for MinGW
 #
@@ -14,7 +14,7 @@
 QMAKE_EXT_OBJ           = .o
 QMAKE_EXT_RES           = _res.o
 
-QMAKE_CC		= gcc
+QMAKE_CC		= x86_64-w64-mingw32-gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= byacc
@@ -23,11 +23,11 @@
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2
+QMAKE_CFLAGS_RELEASE	= -O3
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= x86_64-w64-mingw32-g++
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
@@ -50,8 +50,8 @@
 QMAKE_RUN_CXX		= $(CXX) -c $(CXXFLAGS) $(INCPATH) -o $obj $src
 QMAKE_RUN_CXX_IMP	= $(CXX) -c $(CXXFLAGS) $(INCPATH) -o $@ $<
 
-QMAKE_LINK		= g++
-QMAKE_LINK_C		= gcc
+QMAKE_LINK		= x86_64-w64-mingw32-g++
+QMAKE_LINK_C		= x86_64-w64-mingw32-gcc
 QMAKE_LFLAGS		=
 QMAKE_LFLAGS_EXCEPTIONS_ON = -mthreads
 QMAKE_LFLAGS_EXCEPTIONS_OFF =
@@ -95,15 +95,15 @@
     QMAKE_CHK_DIR_EXISTS	= if not exist
 }
 
-QMAKE_MOC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}moc.exe
-QMAKE_UIC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}uic.exe
-QMAKE_IDC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}idc.exe
+QMAKE_MOC		= moc
+QMAKE_UIC		= uic
+QMAKE_IDC		= idc
 
 QMAKE_IDL		= midl
-QMAKE_LIB		= ar -ru
-QMAKE_RC		= windres
+QMAKE_LIB		= x86_64-w64-mingw32-ar -ru
+QMAKE_RC		= x86_64-w64-mingw32-windres
 QMAKE_ZIP		= zip -r -9
 
-QMAKE_STRIP		= strip
+QMAKE_STRIP		= x86_64-w64-mingw32-strip
 QMAKE_STRIPFLAGS_LIB 	+= --strip-unneeded
 load(qt_config)
