--- qt4/src/plugins/imageformats/imageformats.pro.orig	2011-11-10 18:56:04.617315921 -0500
+++ qt4/src/plugins/imageformats/imageformats.pro	2011-11-10 18:56:39.269487754 -0500
@@ -1,8 +1,8 @@
 TEMPLATE = subdirs
 
-!contains(QT_CONFIG, no-jpeg):!contains(QT_CONFIG, jpeg):SUBDIRS += jpeg
+!contains(QT_CONFIG, no-libjpeg):!contains(QT_CONFIG, libjpeg):SUBDIRS += jpeg
 !contains(QT_CONFIG, no-gif):!contains(QT_CONFIG, gif):SUBDIRS += gif
-!contains(QT_CONFIG, no-mng):!contains(QT_CONFIG, mng):SUBDIRS += mng
+!contains(QT_CONFIG, no-libmng):!contains(QT_CONFIG, libmng):SUBDIRS += mng
 contains(QT_CONFIG, svg):SUBDIRS += svg
-!contains(QT_CONFIG, no-tiff):!contains(QT_CONFIG, tiff):SUBDIRS += tiff
+!contains(QT_CONFIG, no-libtiff):!contains(QT_CONFIG, libtiff):SUBDIRS += tiff
 !contains(QT_CONFIG, no-ico):SUBDIRS += ico
