Remove me after qt4 4.8.0
--- qt-everywhere-opensource-src-4.7.4.orig/src/corelib/tools/qlocale.cpp	2012-01-19 23:34:18.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.4/src/corelib/tools/qlocale.cpp	2012-01-19 23:37:41.000000000 +0100
@@ -6614,6 +6614,14 @@
  *           calculation.
  */
 
+#if defined(Q_OS_WIN) && defined (Q_CC_GNU) && !defined(_clear87) // See QTBUG-7576
+extern "C" {
+__attribute__ ((dllimport)) unsigned int __cdecl __MINGW_NOTHROW _control87 (unsigned int unNew, unsigned int unMask);
+__attribute__ ((dllimport)) unsigned int __cdecl __MINGW_NOTHROW _clearfp (void);  /* Clear the FPU status word */
+#  define _clear87 _clearfp
+}
+#endif
+
 
 /* This actually sometimes returns a pointer to a string literal
    cast to a char*. Do NOT try to modify the return value. */
