--- qt-everywhere-opensource-src-4.7.4/configure.orig	2011-09-28 14:31:35.039958884 -0400
+++ qt-everywhere-opensource-src-4.7.4/configure	2011-09-28 14:31:48.649958702 -0400
@@ -4568,7 +4568,7 @@
 }
 
 # build qmake
-if true; then ###[ '!' -f "$outpath/bin/qmake" ];
+if [ '!' -f "$outpath/bin/qmake" ]; then
     echo "Creating qmake. Please wait..."
 
     OLD_QCONFIG_H=
--- qt-everywhere-opensource-src-4.7.4/src/tools/tools.pro.orig	2011-09-28 14:33:24.119957315 -0400
+++ qt-everywhere-opensource-src-4.7.4/src/tools/tools.pro	2011-09-28 14:33:56.619956845 -0400
@@ -1,11 +1,6 @@
 TEMPLATE = subdirs
 
-TOOLS_SUBDIRS = src_tools_bootstrap src_tools_moc src_tools_rcc
-!contains(QT_CONFIG, no-gui): TOOLS_SUBDIRS += src_tools_uic
-!cross_compile {
-    contains(QT_CONFIG, qt3support): SRC_SUBDIRS += src_tools_uic3
-    win32:!wince*: SRC_SUBDIRS += src_tools_idc
-}
+TOOLS_SUBDIRS = src_tools_bootstrap 
 
 # Set subdir and respective target name
 src_tools_bootstrap.subdir = $$QT_SOURCE_TREE/src/tools/bootstrap
