From c68baab79783c2f103d779b9b9eb4de24781cd76 Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Fri, 12 Jan 2024 20:40:24 +0200
Subject: [PATCH 1/2] Fix incorrect library inclusion

---
 src/qmlintegration/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qmlintegration/CMakeLists.txt b/src/qmlintegration/CMakeLists.txt
index 0dd54f53bd..e77232df84 100644
--- a/src/qmlintegration/CMakeLists.txt
+++ b/src/qmlintegration/CMakeLists.txt
@@ -5,6 +5,6 @@ qt_internal_add_module(QmlIntegration
     HEADER_MODULE
     SOURCES
         qqmlintegration.h
-    LIBRARIES
+    PRIVATE_MODULE_INTERFACE
         Qt::Core
 )
-- 
2.44.0

