--- speexdsp/include/speex/Makefile.am.orig	2013-12-08 11:50:39.859569265 +0100
+++ speexdsp/include/speex/Makefile.am	2013-12-08 11:50:48.535568900 +0100
@@ -2,6 +2,7 @@
 #AUTOMAKE_OPTIONS = no-dependencies
 
 nodist_pkginclude_HEADERS = speexdsp_config_types.h
+pkgincludedir=$(includedir)/speex
 
 pkginclude_HEADERS = speex_echo.h speex_jitter.h speex_preprocess.h speex_resampler.h \
 	speexdsp_types.h
