diff -ruN goom.orig/src/xmmx.c goom/src/xmmx.c
--- goom.orig/src/xmmx.c	2006-10-01 22:10:15.000000000 +0200
+++ goom/src/xmmx.c	2006-10-01 23:45:59.000000000 +0200
@@ -69,8 +69,8 @@
 		 */
 
 		asm volatile
-      ("#1 \n\t movq %[brutS], %%mm0"
-       "#1 \n\t movq %[brutD], %%mm1"
+      ("#1 \n\t movq (%[brutS]), %%mm0"
+       "#1 \n\t movq (%[brutD]), %%mm1"
        "#1 \n\t psubd   %%mm0, %%mm1" /* mm1 = D - S */
        "#1 \n\t movq    %%mm1, %%mm2" /* mm2 = D - S */
        "#1 \n\t pslld     $16, %%mm1"
