--- ebml/CMakeLists.txt.uwp	2018-05-22 16:23:07.388136700 +0200
+++ ebml/CMakeLists.txt	2018-05-22 16:23:21.243621600 +0200
@@ -60,7 +60,7 @@ set(libebml_PUBLIC_HEADERS
   ebml/MemReadIOCallback.h
   ebml/SafeReadIOCallback.h
   ebml/StdIOCallback.h)
-if(WIN32)
+if(WIN32 AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "WindowsStore")
   list(APPEND libebml_SOURCES
     src/platform/win32/WinIOCallback.cpp)
   list(APPEND libebml_PUBLIC_HEADERS
