--- libebml-1.3.0/make/linux/Makefile.orig	2013-07-04 19:08:40.474927603 +0200
+++ libebml-1.3.0/make/linux/Makefile	2013-07-04 19:09:09.568664476 +0200
@@ -61,7 +61,7 @@
 objects:=$(patsubst %$(EXTENSION),%.o,$(sources))
 objects_so:=$(objects)
 
-WARNINGFLAGS=-Wall -Wextra -Wno-unknown-pragmas -ansi -fno-gnu-keywords -Wshadow
+WARNINGFLAGS=-Wall -Wextra -Wno-unknown-pragmas -fno-gnu-keywords -Wshadow
 COMPILEFLAGS=$(WARNINGFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(DEBUGFLAGS) $(INCLUDE)
 DEPENDFLAGS  = $(CXXFLAGS) $(INCLUDE)
 
