--- zlib/configure	2016-02-25 16:35:50.228444200 +0100
+++ zlib/configure.mingw	2016-02-25 16:34:19.725510100 +0100
@@ -194,7 +194,7 @@ if test "$gcc" -eq 1 && ($cc -c $test.c)
 # temporary bypass
         rm -f $test.[co] $test $test$shared_ext
         echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
-        leave 1
+#        leave 1
         LDSHARED=${LDSHARED-"$cc -shared"}
         LDSHAREDLIBC=""
         EXE='.exe' ;;
