--- libdvbpsi/Makefile.in	2013-11-10 14:17:53.000000000 +0200
+++ dvbpsi.fix/Makefile.in	2014-02-15 17:25:31.184012648 +0200
@@ -349,7 +349,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src examples misc
+SUBDIRS = src
 DIST_SUBDIRS = $(SUBDIRS) doc wince
 EXTRA_DIST = libdvbpsi.spec libdvbpsi.spec.in libdvbpsi.pc.in bootstrap
 pkgconfigdir = $(libdir)/pkgconfig
