--- libass-0.15.2/libass/ass_fontselect.c.orig	2021-12-29 16:11:04.588615131 +0100
+++ libass-0.15.2/libass/ass_fontselect.c	2021-12-29 16:13:00.276849748 +0100
@@ -48,7 +48,6 @@
 #include "ass_string.h"
 
 #include <windows.h>
-#include <tchar.h>
 
 #define ABS(x) ((x) < 0 ? -(x) : (x))
 #define MAX_FULLNAME 100
@@ -201,7 +200,7 @@
 {
     DIR *d = opendir(dir);
     wchar_t *dirw = ToWide(dir);
-    _TDIR *d = _topendir(dirw);
+    _WDIR *d = _wopendir(dirw);
     free(dirw);
     if (!d)
         return;
@@ -209,7 +208,7 @@
     size_t namemax = 0;
     char *namebuf = NULL;
     while (1) {
-        struct _tdirent *entry = _treaddir(d);
+        struct _wdirent *entry = _wreaddir(d);
         if (!entry)
             break;
         char* d_name = FromWide(entry->d_name);
@@ -239,7 +238,7 @@
         free(d_name);
     }
     free(namebuf);
-    _tclosedir(d);
+    _wclosedir(d);
 }
 
 /**
