--- schroedinger/schroedinger/schroorc-dist.c.orig	2012-01-23 01:44:05.000000000 +0100
+++ schroedinger/schroedinger/schroorc-dist.c	2012-02-25 16:21:34.387026910 +0100
@@ -52,7 +52,7 @@
 typedef union { orc_int64 i; double f; orc_int32 x2[2]; float x2f[2]; orc_int16 x4[4]; } orc_union64;
 #endif
 #ifndef ORC_RESTRICT
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L && !defined(__ANDROID__)
 #define ORC_RESTRICT restrict
 #elif defined(__GNUC__) && __GNUC__ >= 4
 #define ORC_RESTRICT __restrict__
