--- schroedinger/Makefile.am	2009-09-29 07:08:51.000000000 +0200
+++ schroedinger.new/Makefile.am	2010-01-09 02:18:41.000000000 +0100
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = schroedinger doc tools testsuite
+SUBDIRS = schroedinger doc tools 
 
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
--- schroedinger/tools/Makefile.am	2010-01-04 03:52:01.000000000 +0100
+++ schroedinger.new/tools/Makefile.am	2010-05-08 20:50:30.000000000 +0200
@@ -5,7 +5,7 @@
 
 
 AM_CFLAGS = $(SCHRO_CFLAGS) $(ERROR_CFLAGS) \
-	$(ORC_CFLAGS) \
+	-L$(prefix)/lib $(ORC_CFLAGS) \
 	-DSCHRO_ENABLE_UNSTABLE_API
 LDADD = $(SCHRO_LIBS)
 
