diff --git a/build/cmake/pkg_config.cmake b/build/cmake/pkg_config.cmake
index c3914d79e..1eb36f7ec 100644
--- a/build/cmake/pkg_config.cmake
+++ b/build/cmake/pkg_config.cmake
@@ -53,10 +53,6 @@ file(
 file(APPEND "${pkgconfig_file}" "Version: ${package_version}\n")
 file(APPEND "${pkgconfig_file}" "Requires:\n")
 file(APPEND "${pkgconfig_file}" "Conflicts:\n")
-file(APPEND "${pkgconfig_file}" "Libs: -L\${libdir} -l${pkg_name}\n")
-if(CONFIG_MULTITHREAD AND HAVE_PTHREAD_H)
-  file(APPEND "${pkgconfig_file}" "Libs.private: -lm -lpthread\n")
-else()
-  file(APPEND "${pkgconfig_file}" "Libs.private: -lm\n")
-endif()
+file(APPEND "${pkgconfig_file}" "Libs: -L\${prefix}/lib -l${pkg_name} -lm\n")
+file(APPEND "${pkgconfig_file}" "Libs.private: -lm\n")
 file(APPEND "${pkgconfig_file}" "Cflags: -I\${includedir}\n")
