--- libmodplug/configure.ac.old	2019-08-14 09:18:02.000000000 +0200
+++ libmodplug/configure.ac	2019-08-14 09:18:28.000000000 +0200
@@ -44,12 +44,6 @@
 esac
 AC_SUBST(LT_LDFLAGS)
 
-# require 10.5+ for osx/x86_64 builds
-case "$host" in
-	x86_64-*-darwin*)
-		CXXFLAGS="$CXXFLAGS -mmacosx-version-min=10.5"
-		LDFLAGS="$LDFLAGS -mmacosx-version-min=10.5" ;;
-esac
 
 # symbol visibility
 ac_save_CXXFLAGS="$CXXFLAGS"
