diff -ruN fontconfig/src/fcxml.c fontconfig.new/src/fcxml.c
--- fontconfig/src/fcxml.c	2014-07-06 00:38:32.773443543 +0200
+++ fontconfig.new/src/fcxml.c	2014-07-05 21:11:51.171349523 +0200
@@ -23,6 +23,8 @@
  */
 
 #include "fcint.h"
+
+#if 0
 #include <fcntl.h>
 #include <stdarg.h>
 #include <dirent.h>
@@ -56,6 +58,8 @@
 #include <mbstring.h>
 #endif
 
+#endif
+
 static void
 FcExprDestroy (FcExpr *e);
 
@@ -86,6 +90,7 @@
 	FcRuleDestroy (n);
 }
 
+#if 0
 static FcExpr *
 FcExprCreateInteger (FcConfig *config, int i)
 {
@@ -132,6 +137,7 @@
   }
   return m;
 }
+#endif 
 
 static void
 FcExprMatrixFreeShallow (FcExprMatrix *m)
@@ -156,6 +162,7 @@
   free (m);
 }
 
+#if 0
 static FcExpr *
 FcExprCreateMatrix (FcConfig *config, const FcExprMatrix *matrix)
 {
@@ -240,7 +247,7 @@
     }
     return e;
 }
-
+#endif
 static void
 FcExprDestroy (FcExpr *e)
 {
@@ -323,6 +330,7 @@
     free (e);
 }
 
+#if 0
 typedef enum _FcElement {
     FcElementNone,
     FcElementFontconfig,
@@ -2973,6 +2981,7 @@
     const FcChar8    *bs = *((FcChar8 **) b);
     return FcStrCmp (as, bs);
 }
+#endif
 
 static FcBool
 FcConfigParseAndLoadDir (FcConfig	*config,
@@ -2980,6 +2989,9 @@
 			 const FcChar8	*dir,
 			 FcBool		complain)
 {
+    return FcFalse;
+}
+#if 0
     DIR		    *d;
     struct dirent   *e;
     FcBool	    ret = FcTrue;
@@ -3056,6 +3068,7 @@
 bail0:
     return ret || !complain;
 }
+#endif
 
 #ifdef _WIN32
 pfnGetSystemWindowsDirectory pGetSystemWindowsDirectory = NULL;
@@ -3067,7 +3080,10 @@
 		      const FcChar8 *name,
 		      FcBool	    complain)
 {
+    return FcFalse;
+}
 
+#if 0
     XML_Parser	    p;
     FcChar8	    *filename;
     int		    fd;
@@ -3210,3 +3226,4 @@
 #define __fcxml__
 #include "fcaliastail.h"
 #undef __fcxml__
+#endif
--- fontconfig-2.12.1/src/fcinit.c.orig	2017-04-08 11:38:02.488354178 +0200
+++ fontconfig-2.12.1/src/fcinit.c	2017-04-08 11:38:52.072351969 +0200
@@ -45,11 +45,16 @@
     if (!config)
 	goto bail0;
     FcConfigSetSysRoot (config, sysroot);
-    if (!FcConfigAddDir (config, (FcChar8 *) FC_DEFAULT_FONTS))
+    char path[MAX_PATH + 8];
+    GetWindowsDirectory( path, MAX_PATH + 1 );
+    strcat( path, "\\fonts" );
+//    printf( "Fonts dir: %s\n", path );
+    if (!FcConfigAddDir (config, (FcChar8 *) path))
 	goto bail1;
     char dir[MAX_PATH+5];
     SHGetFolderPath( NULL, CSIDL_APPDATA, NULL, 0, dir );
     strncat( dir, "\\vlc", 4 );
+//    printf( "Cache dir: %s\n", dir );
     if (!FcConfigAddCacheDir (config, (FcChar8 *)dir))
 	goto bail1;
     return config;
@@ -193,7 +202,7 @@
     FcCacheFini ();
     FcDefaultFini ();
     FcObjectFini ();
-    FcConfigPathFini ();
+//    FcConfigPathFini ();
 }
 
 /*
