--- fontconfig/src/fcxml.c	2009-08-13 16:07:13.000000000 +0200
+++ fontconfig.new/src/fcxml.c	2009-08-13 16:05:41.000000000 +0200
@@ -23,6 +23,8 @@
  */
 
 #include "fcint.h"
+
+#if 0
 #include <fcntl.h>
 #include <stdarg.h>
 #include <dirent.h>
@@ -58,6 +60,8 @@
 #undef STRICT
 #endif
 
+#endif
+
 static void
 FcExprDestroy (FcExpr *e);
 
@@ -71,6 +75,7 @@
     free (test);
 }
 
+#if 0
 static FcExpr *
 FcExprCreateInteger (FcConfig *config, int i)
 {
@@ -179,7 +184,7 @@
     }
     return e;
 }
-
+#endif
 static void
 FcExprDestroy (FcExpr *e)
 {
@@ -255,6 +260,7 @@
     free (e);
 }
 
+#if 0
 typedef enum _FcElement {
     FcElementNone,
     FcElementFontconfig,
@@ -2373,6 +2379,7 @@
     const FcChar8    *bs = *((FcChar8 **) b);
     return FcStrCmp (as, bs);
 }
+#endif
 
 static FcBool
 FcConfigParseAndLoadDir (FcConfig	*config,
@@ -2380,6 +2387,9 @@
 			 const FcChar8	*dir,
 			 FcBool		complain)
 {
+    return FcFalse;
+}
+#if 0
     DIR		    *d;
     struct dirent   *e;
     FcBool	    ret = FcTrue;
@@ -2589,3 +2599,4 @@
 #define __fcxml__
 #include "fcaliastail.h"
 #undef __fcxml__
+#endif
diff -ruN fontconfig/src/fcxml.c fontconfig.new/src/fcxml.c
--- fontconfig/src/fcxml.c	2009-08-13 16:08:49.000000000 +0200
+++ fontconfig.new/src/fcxml.c	2009-08-19 00:03:35.000000000 +0200
@@ -2850,6 +2850,7 @@
 bail0:
     return ret || !complain;
 }
+#endif
 
 #ifdef _WIN32
 pfnGetSystemWindowsDirectory pGetSystemWindowsDirectory = NULL;
@@ -2861,7 +2862,10 @@
 		      const FcChar8 *name,
 		      FcBool	    complain)
 {
+    return FcFalse;
+}
 
+#if 0
     XML_Parser	    p;
     FcChar8	    *filename;
     int		    fd;
--- fontconfig/src/fcinit.c	2010-04-27 15:31:41.000000000 +0200
+++ fontconfig.new/src/fcinit.c	2010-04-27 15:35:42.000000000 +0200
@@ -34,11 +34,16 @@
     config = FcConfigCreate ();
     if (!config)
 	goto bail0;
-    if (!FcConfigAddDir (config, (FcChar8 *) FC_DEFAULT_FONTS))
+    char *path = malloc( MAX_PATH + 1 );
+    GetWindowsDirectory( path, MAX_PATH + 1 );
+    strcat( path, "\\fonts" );
+//    printf( "Fonts dir: %s\n", path );
+    if (!FcConfigAddDir (config, (FcChar8 *) path))
 	goto bail1;
     char dir[MAX_PATH+5];
     SHGetFolderPath( NULL, CSIDL_APPDATA, NULL, 0, dir );
     strncat( dir, "\\vlc", 4 );
+//    printf( "Cache dir: %s\n", dir );
     if (!FcConfigAddCacheDir (config, (FcChar8 *)dir))
 	goto bail1;
     return config;
