From 20bc88f0cda8325cf5d22ee470f7f93c87891a91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C3=ABl=20Carr=C3=A9?= <funman@videolan.org>
Date: Sun, 8 Dec 2013 08:44:07 +0100
Subject: [PATCH] Add --disable-binaries option to build only the library

---
 configure.ac | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 07ebff2..01dbfba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -109,7 +109,13 @@ AC_MSG_RESULT($has_visibility)
 
 AC_CHECK_HEADERS(sys/soundcard.h sys/audioio.h)
 
-XIPH_PATH_OGG([src="src"], [src=""])
+AC_ARG_ENABLE(binaries, [  --disable-binaries      Do not build the encoder and decoder binaries, only the library])
+if test "$enable_binaries" != no; then
+    XIPH_PATH_OGG([src="src"], [src=""])
+else
+    src=""
+fi
+
 AC_SUBST(src)
 
 LT_LIB_M
@@ -339,7 +345,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_OUTPUT
 
 
-if test "x$src" = "x"; then 
+if test "x$src" = "x" -a "$enable_binaries" != no; then
 echo "**IMPORTANT**"
 echo "You don't seem to have the development package for libogg (libogg-devel) installed. Only the Speex library (libspeex) will be built (no encoder/decoder executable)"
 echo "You can download libogg from http://downloads.xiph.org/releases/ogg/"
-- 
1.8.3.2

