--- speex/configure.ac.orig	2013-10-16 12:58:06.707579608 +0200
+++ speex/configure.ac	2013-10-16 12:58:21.423579879 +0200
@@ -1,32 +1,29 @@
 dnl Process this file with autoconf to produce a configure script. -*-m4-*-
 
-AC_INIT(libspeex/speex.c)
-
-AM_CONFIG_HEADER([config.h])
-
 SPEEX_MAJOR_VERSION=1
 SPEEX_MINOR_VERSION=1
 SPEEX_MICRO_VERSION=16
 SPEEX_EXTRA_VERSION=
 #SPEEX_VERSION=
 #SPEEX_VERSION=$SPEEX_MAJOR_VERSION.$SPEEX_MINOR_VERSION.$SPEEX_MICRO_VERSION$SPEEX_EXTRA_VERSION
+AC_INIT([speex], 1.2rc1)
 SPEEX_VERSION="1.2rc1"
 
 SPEEX_LT_CURRENT=6
 SPEEX_LT_REVISION=0
 SPEEX_LT_AGE=5
 
+AC_CONFIG_SRCDIR(libspeex/speex.c)
+AM_INIT_AUTOMAKE
+
+AM_CONFIG_HEADER([config.h])
+
 AC_SUBST(SPEEX_LT_CURRENT)
 AC_SUBST(SPEEX_LT_REVISION)
 AC_SUBST(SPEEX_LT_AGE)
 
-# For automake.
-VERSION=$SPEEX_VERSION
-PACKAGE=speex
-
 AC_SUBST(SPEEX_VERSION)
 
-AM_INIT_AUTOMAKE($PACKAGE, $VERSION, no-define)
 AM_MAINTAINER_MODE
 
 AC_CANONICAL_HOST
--- speex/libspeex/speex.c.orig	2013-10-16 13:02:22.643584325 +0200
+++ speex/libspeex/speex.c	2013-10-16 13:01:25.995583281 +0200
@@ -222,6 +222,7 @@
 {
    switch (request)
    {
+#if 0
       case SPEEX_LIB_GET_MAJOR_VERSION:
          *((int*)ptr) = SPEEX_MAJOR_VERSION;
          break;
@@ -237,6 +238,7 @@
       case SPEEX_LIB_GET_VERSION_STRING:
          *((const char**)ptr) = SPEEX_VERSION;
          break;
+#endif
       /*case SPEEX_LIB_SET_ALLOC_FUNC:
          break;
       case SPEEX_LIB_GET_ALLOC_FUNC:
