diff -ur speex.orig/configure.ac speex/configure.ac
--- speex.orig/configure.ac	2012-01-22 15:23:09.767731000 -0500
+++ speex/configure.ac	2012-01-22 15:24:03.891731001 -0500
@@ -112,7 +112,7 @@
 
 AC_CHECK_HEADERS(sys/soundcard.h sys/audioio.h)
 
-XIPH_PATH_OGG([src="src"], [src=""])
+src=
 AC_SUBST(src)
 
 AC_CHECK_LIB(m, sin)
diff -ur speex.orig/libspeex/Makefile.am speex/libspeex/Makefile.am
--- speex.orig/libspeex/Makefile.am	2012-01-22 15:23:09.759731001 -0500
+++ speex/libspeex/Makefile.am	2012-01-22 15:23:40.799730998 -0500
@@ -11,7 +11,7 @@
 
 EXTRA_DIST=echo_diagnostic.m
 
-INCLUDES = -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@ @FFT_CFLAGS@
+INCLUDES = -I$(top_builddir)/include -I$(top_builddir) @FFT_CFLAGS@
 
 lib_LTLIBRARIES = libspeex.la libspeexdsp.la
 
diff -ur speex.orig/src/Makefile.am speex/src/Makefile.am
--- speex.orig/src/Makefile.am	2012-01-22 15:23:09.775731000 -0500
+++ speex/src/Makefile.am	2012-01-22 15:23:30.351731024 -0500
@@ -11,7 +11,7 @@
   VPSY_LIB=
 endif
 
-INCLUDES = -I$(top_builddir) @OGG_CFLAGS@
+INCLUDES = -I$(top_builddir)
 
 man_MANS = speexenc.1 speexdec.1
 
@@ -24,8 +24,8 @@
 
 speexenc_SOURCES = speexenc.c wav_io.c skeleton.c
 speexenc_LDADD = $(top_builddir)/libspeex/libspeex.la $(top_builddir)/libspeex/libspeexdsp.la \
-	$(OGG_LIBS)  @FFT_LIBS@	
+	@FFT_LIBS@
 
 speexdec_SOURCES = speexdec.c wav_io.c
 speexdec_LDADD = $(top_builddir)/libspeex/libspeex.la $(VPSY_LIB) \
-	$(OGG_LIBS)  @FFT_LIBS@
+	@FFT_LIBS@
