diff -ru clinkcc.orig/src/cybergarage/xml/XML.cpp clinkcc/src/cybergarage/xml/XML.cpp
--- clinkcc.orig/src/cybergarage/xml/XML.cpp	2005-11-14 18:11:07.000000000 +0100
+++ clinkcc/src/cybergarage/xml/XML.cpp	2005-11-14 18:12:27.000000000 +0100
@@ -117,7 +117,7 @@
 	}
 	utf8str[outLen] = '\0';
 	return utf8str;
-#elif defined(HAVE_ICONV) || defined(HAVE_ICONV_H)
+/*#elif defined(HAVE_ICONV) || defined(HAVE_ICONV_H)
 	iconvMutex.lock();
 
 	char *cpbuf = strdup(str);
@@ -166,7 +166,7 @@
 	free(cpbuf);
    	iconvMutex.unlock();
 
-	return unistr;
+	return unistr;*/
 #else
 	outLen = strlen(str);
 	UnicodeStr *utf8str = new UnicodeStr[outLen+1];
