diff -ruN dirac.old/Makefile.in dirac/Makefile.in
--- dirac.old/Makefile.in	2007-03-27 00:34:21.000000000 +0200
+++ dirac/Makefile.in	2007-03-27 00:35:09.000000000 +0200
@@ -230,8 +230,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = libdirac_byteio libdirac_common libdirac_motionest libdirac_encoder libdirac_decoder \
-encoder decoder util $(CPPUNITTESTS_DIR) tests $(DOC_DIR) win32
+SUBDIRS = libdirac_byteio libdirac_common libdirac_motionest libdirac_encoder libdirac_decoder 
 
 @HAVE_DOXYGEN_FALSE@DOC_DIR = 
 @HAVE_DOXYGEN_TRUE@DOC_DIR = doc
