diff --git a/Makefile.am b/Makefile.am
index dfd6b87..e534452 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,17 +3,12 @@
 
 SUBDIRS = libdirac_byteio libdirac_common libdirac_motionest libdirac_encoder \
           libdirac_decoder encoder decoder util $(CPPUNITTESTS_DIR) \
-          tests $(DOC_DIR) win32
+          win32
 
 .PHONY: valgrind-check
 valgrind-check:
 	cd tests && $(MAKE) $(AM_MAKEFLAGS) $@
 
-if HAVE_DOXYGEN
-DOC_DIR = doc
-else
-DOC_DIR =
-endif
 
 if HAVE_LATEX
 DOC_DIR = doc
diff --git a/configure.ac b/configure.ac
index a4d0d84..975a83d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -149,7 +149,7 @@ dnl Checks for optional files in doc directory.
 dnl algorithm documentation is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/algorithm/algorithm/index.htm,[HAVE_ALGO="true"], [HAVE_ALGO="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/algorithm/index.htm,[HAVE_ALGO="true"], [HAVE_ALGO="false"])
 AM_CONDITIONAL(HAVE_ALGO, $HAVE_ALGO)
 
 dnl -----------------------------------------------
@@ -157,7 +157,7 @@ dnl Checks for optional files in doc directory.
 dnl Enhancements is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/algorithm/enhancements/index.htm,[HAVE_ENHANCE="true"], [HAVE_ENHANCE="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/enhancements/index.htm,[HAVE_ENHANCE="true"], [HAVE_ENHANCE="false"])
 AM_CONDITIONAL(HAVE_ENHANCE, $HAVE_ENHANCE)
 
 dnl -----------------------------------------------
@@ -165,7 +165,7 @@ dnl Checks for optional files in doc directory.
 dnl Global Motion documentation is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/algorithm/global_motion/index.htm,[HAVE_GM="true"], [HAVE_GM="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/global_motion/index.htm,[HAVE_GM="true"], [HAVE_GM="false"])
 AM_CONDITIONAL(HAVE_GM, $HAVE_GM)
 
 dnl -----------------------------------------------
@@ -173,7 +173,7 @@ dnl Checks for optional files in doc directory.
 dnl Upconversion doc is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/algorithm/upconversion/index.htm,[HAVE_UPCONV="true"], [HAVE_UPCONV="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/upconversion/index.htm,[HAVE_UPCONV="true"], [HAVE_UPCONV="false"])
 AM_CONDITIONAL(HAVE_UPCONV, $HAVE_UPCONV)
 
 dnl -----------------------------------------------
@@ -181,7 +181,7 @@ dnl Checks for optional files in doc directory.
 dnl programmer's guide is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/code/programmers_guide/index.htm,[HAVE_PROG="true"], [HAVE_PROG="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/code/programmers_guide/index.htm,[HAVE_PROG="true"], [HAVE_PROG="false"])
 AM_CONDITIONAL(HAVE_PROG, $HAVE_PROG)
 
 dnl -----------------------------------------------
@@ -189,7 +189,7 @@ dnl Checks for optional files in doc directory.
 dnl Diagnostics tools docs are is not distributed in
 dnl the release. It is available only from CVS
 dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/tools/diagnostics/index.htm,[HAVE_DIAG="true"], [HAVE_DIAG="false"])
+dnl AC_CHECK_FILE($srcdir/doc/documentation/tools/diagnostics/index.htm,[HAVE_DIAG="true"], [HAVE_DIAG="false"])
 AM_CONDITIONAL(HAVE_DIAG, $HAVE_DIAG)
 
 dnl -----------------------------------------------
diff --git a/libdirac_common/band_codec.cpp b/libdirac_common/band_codec.cpp
index f750906..bfe6e75 100644
--- a/libdirac_common/band_codec.cpp
+++ b/libdirac_common/band_codec.cpp
@@ -433,7 +433,7 @@ inline int BandCodec::ChooseFollowContext( const int bin_number ) const
         }
 
     }
-
+    return 0; // avoid warning
 }
 
 inline int BandCodec::ChooseInfoContext() const
--- dirac/configure.ac	2008-06-28 02:43:15.000000000 +0200
+++ dirac.new/configure.ac	2008-06-28 02:38:28.000000000 +0200
@@ -145,70 +145,6 @@
 AM_CONDITIONAL(OVERLAY, test x$overlay = xtrue)
 
 dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl algorithm documentation is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/algorithm/index.htm,[HAVE_ALGO="true"], [HAVE_ALGO="false"])
-AM_CONDITIONAL(HAVE_ALGO, $HAVE_ALGO)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Enhancements is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/enhancements/index.htm,[HAVE_ENHANCE="true"], [HAVE_ENHANCE="false"])
-AM_CONDITIONAL(HAVE_ENHANCE, $HAVE_ENHANCE)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Global Motion documentation is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/global_motion/index.htm,[HAVE_GM="true"], [HAVE_GM="false"])
-AM_CONDITIONAL(HAVE_GM, $HAVE_GM)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Upconversion doc is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/algorithm/upconversion/index.htm,[HAVE_UPCONV="true"], [HAVE_UPCONV="false"])
-AM_CONDITIONAL(HAVE_UPCONV, $HAVE_UPCONV)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl programmer's guide is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/code/programmers_guide/index.htm,[HAVE_PROG="true"], [HAVE_PROG="false"])
-AM_CONDITIONAL(HAVE_PROG, $HAVE_PROG)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Diagnostics tools docs are is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-dnl AC_CHECK_FILE($srcdir/doc/documentation/tools/diagnostics/index.htm,[HAVE_DIAG="true"], [HAVE_DIAG="false"])
-AM_CONDITIONAL(HAVE_DIAG, $HAVE_DIAG)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Publications documentation is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/publications/diracoverview-ibc2006.pdf,[HAVE_PUBLICATIONS="true"], [HAVE_PUBLICATIONS="false"])
-AM_CONDITIONAL(HAVE_PUBLICATIONS, $HAVE_PUBLICATIONS)
-
-dnl -----------------------------------------------
-dnl Checks for optional files in doc directory.
-dnl Support documentation is not distributed in
-dnl the release. It is available only from CVS
-dnl -----------------------------------------------
-AC_CHECK_FILE($srcdir/doc/documentation/support/test.html,[HAVE_SUPPORT="true"], [HAVE_SUPPORT="false"])
-AM_CONDITIONAL(HAVE_SUPPORT, $HAVE_SUPPORT)
-
-dnl -----------------------------------------------
 dnl Setup for the testsuite
 dnl -----------------------------------------------
 AC_CONFIG_TESTDIR([tests], [tests:encoder:decoder:unit_tests:util/instrumentation])
@@ -452,20 +388,6 @@
           util/conversion/common/Makefile \
 	  util/instrumentation/Makefile \
 	  util/instrumentation/libdirac_instrument/Makefile \
-          doc/Makefile \
-          doc/documentation/Makefile \
-          doc/documentation/algorithm/Makefile \
-          doc/documentation/algorithm/algorithm/Makefile \
-          doc/documentation/algorithm/enhancements/Makefile \
-          doc/documentation/algorithm/global_motion/Makefile \
-          doc/documentation/algorithm/upconversion/Makefile \
-          doc/documentation/code/Makefile \
-          doc/documentation/code/api/Makefile \
-          doc/documentation/code/programmers_guide/Makefile \
-          doc/documentation/publications/Makefile \
-          doc/documentation/support/Makefile \
-          doc/documentation/tools/Makefile \
-          doc/documentation/tools/diagnostics/Makefile \
           libdirac_byteio/Makefile \
           libdirac_common/Makefile \
           libdirac_motionest/Makefile \
