--- dirac/Makefile.am	2008-10-01 07:33:56.000000000 +0200
+++ dirac.new/Makefile.am	2010-01-09 02:05:45.000000000 +0100
@@ -2,7 +2,7 @@
 #
 
 SUBDIRS = libdirac_byteio libdirac_common libdirac_motionest libdirac_encoder \
-          libdirac_decoder encoder decoder util $(CPPUNITTESTS_DIR) \
+          libdirac_decoder encoder decoder $(CPPUNITTESTS_DIR) \
           tests $(DOC_DIR) win32
 
 .PHONY: valgrind-check
