--- dirac/Makefile.am.orig	2010-11-01 01:31:21.638194470 +0100
+++ dirac/Makefile.am	2010-11-01 01:31:25.174196742 +0100
@@ -1,6 +1,8 @@
 # $Id: Makefile.am,v 1.22 2008/10/01 05:33:56 asuraparaju Exp $
 #
 
+ACLOCAL_AMFLAGS= -I m4
+
 SUBDIRS = libdirac_byteio libdirac_common libdirac_motionest libdirac_encoder \
           libdirac_decoder encoder decoder $(CPPUNITTESTS_DIR) \
           tests $(DOC_DIR) win32
--- dirac/configure.ac.orig	2010-11-01 01:26:21.058194985 +0100
+++ dirac/configure.ac	2010-11-01 01:32:22.062185299 +0100
@@ -57,8 +57,6 @@
 AM_PROG_LIBTOOL
 AC_SYS_LARGEFILE
 
-ACLOCAL_AMFLAGS="-I m4 $ACLOCAL_AMFLAGS"
-AC_SUBST(ACLOCAL_AMFLAGS)
 
 
 dnl -------------------------------------------------------------------------
