diff -ru libebml-original/ebml/c/libebml_t.h libebml/ebml/c/libebml_t.h
--- libebml-original/ebml/c/libebml_t.h	Thu Apr 15 01:16:31 2004
+++ libebml/ebml/c/libebml_t.h	Mon Aug 16 14:29:14 2004
@@ -72,6 +72,8 @@
     typedef uint16_t uint16;
     typedef uint8_t uint8;
 #endif // __GNUC__
+#elif defined(__BEOS__)
+#include <SupportDefs.h>
 #else
 #ifdef DJGPP				/* SL : DJGPP doesn't support POSIX types ???? */
     typedef signed long long int64;
