+codebook/hcb_9.h codebook/hcb_10.h codebook/hcb_11.h codebook/hcb_sf.h
Index: libfaad/common.h
===================================================================
RCS file: /cvsroot/faac/faad2/libfaad/common.h,v
retrieving revision 1.66
diff -u -r1.66 common.h
--- libfaad/common.h	1 Feb 2005 13:23:35 -0000	1.66
+++ libfaad/common.h	13 May 2005 09:07:19 -0000
@@ -309,8 +309,10 @@
   }
 
 
-  #if defined(_WIN32) && !defined(__MINGW32__)
+  #ifdef __MINGW32__
     #define HAS_LRINTF
+  #elif defined(_WIN32)
+   #define HAS_LRINTF
     static INLINE int lrintf(float f)
     {
         int i;
