diff -ru ffmpeg.old/configure ffmpeg/configure
--- ffmpeg.old/configure	Sun Sep 28 17:26:39 2003
+++ ffmpeg/configure	Tue Nov 11 17:17:59 2003
@@ -172,18 +172,18 @@
 extralibs=""
 darwin="yes"
 strip="strip -x"
-LDFLAGS="-d"
+LDFLAGS="$LDFLAGS -d"
 FFSLDFLAGS=-Wl,-bind_at_load
 gcc_version="$($cc -v 2>&1 | grep version | cut -d ' ' -f3-)"
 case "$gcc_version" in
 *2.95*) 
-CFLAGS="-no-cpp-precomp -pipe -fomit-frame-pointer"
+CFLAGS="$CFLAGS -no-cpp-precomp -pipe -fomit-frame-pointer"
 ;;
 *3.3*) 
-CFLAGS="-no-cpp-precomp -pipe -fomit-frame-pointer -mdynamic-no-pic -force_cpusubtype_ALL"
+CFLAGS="$CFLAGS -no-cpp-precomp -pipe -fomit-frame-pointer -mdynamic-no-pic -force_cpusubtype_ALL"
 ;;
 *) 
-CFLAGS="-no-cpp-precomp -pipe -fomit-frame-pointer -mdynamic-no-pic"
+CFLAGS="$CFLAGS -no-cpp-precomp -pipe -fomit-frame-pointer -mdynamic-no-pic"
 ;;
 esac
 ;;
diff -ru ffmpeg.old/libavcodec/dsputil.h ffmpeg/libavcodec/dsputil.h
--- ffmpeg.old/libavcodec/dsputil.h	Sun Sep 28 17:26:39 2003
+++ ffmpeg/libavcodec/dsputil.h	Tue Nov 11 17:16:28 2003
@@ -477,6 +477,8 @@
           +name8(s, dst+8+8*stride, src+8+8*stride, stride);\
 }
 
+/* Meuuh power */
+#undef HAVE_LRINTF
 #ifndef HAVE_LRINTF
 /* XXX: add ISOC specific test to avoid specific BSD testing. */
 /* better than nothing implementation. */
diff -ru ffmpeg.old/libavcodec/eval.c ffmpeg/libavcodec/eval.c
--- ffmpeg.old/libavcodec/eval.c	Sun Sep 28 17:26:39 2003
+++ ffmpeg/libavcodec/eval.c	Tue Nov 11 17:16:28 2003
@@ -34,6 +34,8 @@
 #include <string.h>
 #include <math.h>
 
+/* Meuuh power */
+#undef NAN
 #ifndef NAN
   #define NAN 0
 #endif
diff -ru ffmpeg.old/libavcodec/libpostproc/Makefile ffmpeg/libavcodec/libpostproc/Makefile
--- ffmpeg.old/libavcodec/libpostproc/Makefile	Sun Sep 28 17:26:40 2003
+++ ffmpeg/libavcodec/libpostproc/Makefile	Tue Nov 11 17:16:28 2003
@@ -55,6 +55,9 @@
 	ldconfig || true
 	mkdir -p $(prefix)/include/postproc
 	install -m 644 postprocess.h $(prefix)/include/postproc/postprocess.h
+else
+	mkdir -p $(prefix)/include/libpostproc
+	install -m 644 postprocess.h $(prefix)/include/libpostproc/postprocess.h
 endif
 
 
diff -ru ffmpeg.old/libavcodec/resample.c ffmpeg/libavcodec/resample.c
--- ffmpeg.old/libavcodec/resample.c	Sun Sep 28 17:26:39 2003
+++ ffmpeg/libavcodec/resample.c	Tue Nov 11 17:16:28 2003
@@ -49,7 +49,7 @@
 static void init_mono_resample(ReSampleChannelContext *s, float ratio)
 {
     ratio = 1.0 / ratio;
-    s->iratio = (int)floorf(ratio);
+    s->iratio = (int)floor(ratio);
     if (s->iratio == 0)
         s->iratio = 1;
     s->incr = (int)((ratio / s->iratio) * FRAC);
diff -ru ffmpeg.old/libavformat/rtsp.c ffmpeg/libavformat/rtsp.c
--- ffmpeg.old/libavformat/rtsp.c	Sun Sep 28 17:26:40 2003
+++ ffmpeg/libavformat/rtsp.c	Tue Nov 11 17:16:28 2003
@@ -20,6 +20,7 @@
 
 #include <unistd.h> /* for select() prototype */
 #include <sys/time.h>
+#include <unistd.h>
 #include <netinet/in.h>
 #include <sys/socket.h>
 #ifndef __BEOS__
diff -ru ffmpeg.old/libavformat/tcp.c ffmpeg/libavformat/tcp.c
--- ffmpeg.old/libavformat/tcp.c	Sun Sep 28 17:26:40 2003
+++ ffmpeg/libavformat/tcp.c	Tue Nov 11 17:16:28 2003
@@ -32,6 +32,7 @@
 #include <netdb.h>
 #include <sys/time.h>
 #include <fcntl.h>
+#define socklen_t int
 
 typedef struct TCPContext {
     int fd;
