Index: swscale.c
===================================================================
--- swscale.c	(revision 26515)
+++ swscale.c	(working copy)
@@ -880,7 +880,7 @@
 #endif
 
 #ifdef ARCH_POWERPC
-#if (defined (HAVE_ALTIVEC) || defined (RUNTIME_CPUDETECT)) && defined (CONFIG_GPL)
+#if (defined (HAVE_ALTIVEC)) && defined (CONFIG_GPL)
 #define COMPILE_ALTIVEC
 #endif //HAVE_ALTIVEC
 #endif //ARCH_POWERPC
@@ -1553,11 +1553,13 @@
 
 #else
 #ifdef ARCH_POWERPC
+#ifdef COMPILE_ALTIVEC
     if (flags & SWS_CPU_CAPS_ALTIVEC)
         return swScale_altivec;
     else
         return swScale_C;
 #endif
+#endif
     return swScale_C;
 #endif /* defined(ARCH_X86) */
 #else //RUNTIME_CPUDETECT
