Index: ffmpeg/configure
===================================================================
--- ffmpeg/configure	(revision 5790)
+++ ffmpeg/configure	(working copy)
@@ -2124,7 +2124,7 @@
 # build pkg-config files libav*.pc and libpostproc.pc
 # libavutil.pc
 cat <<EOF >libavutil.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2139,7 +2139,7 @@
 EOF
 
 cat <<EOF >libavutil-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavutil
 includedir=\${pcfiledir}/libavutil
@@ -2155,7 +2155,7 @@
 
 # libavcodec.pc
 cat <<EOF >libavcodec.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2170,7 +2170,7 @@
 EOF
 
 cat <<EOF >libavcodec-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavcodec
 includedir=\${pcfiledir}/libavcodec
@@ -2186,7 +2186,7 @@
 
 # libavformat.pc
 cat <<EOF >libavformat.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2201,7 +2201,7 @@
 EOF
 
 cat <<EOF >libavformat-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavformat
 includedir=\${pcfiledir}/libavformat
@@ -2218,7 +2218,7 @@
 
 # libpostproc.pc
 cat <<EOF >libpostproc.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2233,7 +2233,7 @@
 EOF
 
 cat <<EOF >libpostproc-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libpostproc
 includedir=\${pcfiledir}/libpostproc
