Index: ffmpeg/libavformat/utils.c
===================================================================
--- ffmpeg/libavformat/utils.c	(révision 7907)
+++ ffmpeg/libavformat/utils.c	(copie de travail)
@@ -282,7 +282,7 @@
     fmt = NULL;
     score_max = 0;
     for(fmt1 = first_iformat; fmt1 != NULL; fmt1 = fmt1->next) {
-        if (!is_opened == !(fmt1->flags & AVFMT_NOFILE))
+        if (!is_opened && !(fmt1->flags & AVFMT_NOFILE))
             continue;
         score = 0;
         if (fmt1->read_probe) {
