--- ffmpeg/configure.orig	2007-11-23 09:07:23.000000000 +0100
+++ ffmpeg/configure	2007-11-23 10:54:15.000000000 +0100
@@ -130,6 +130,7 @@
   echo "  --disable-ffmpeg         disable ffmpeg build"
   echo "  --disable-ffserver       disable ffserver build"
   echo "  --disable-ffplay         disable ffplay build"
+  echo "  --enable-avdevices       enable avdevices build"
   echo "  --enable-small           optimize for size instead of speed"
   echo "  --enable-memalign-hack   emulate memalign, interferes with memory debuggers"
   echo "  --disable-encoder=NAME   disables encoder NAME"
@@ -603,6 +604,7 @@
     ffmpeg
     ffplay
     ffserver
+    avdevices
     gpl
     gprof
     gray
--- ffmpeg/Makefile.orig	2007-11-23 11:06:07.000000000 +0100
+++ ffmpeg/Makefile	2007-11-23 11:09:55.000000000 +0100
@@ -61,8 +61,13 @@
 VHOOKCFLAGS += $(VHOOKCFLAGS-yes)
 
 SRCS = $(addsuffix .c, $(PROGS-yes)) cmdutils.c
-LDFLAGS := -L$(BUILD_ROOT)/libavdevice -L$(BUILD_ROOT)/libavformat -L$(BUILD_ROOT)/libavcodec -L$(BUILD_ROOT)/libavutil $(LDFLAGS)
-EXTRALIBS := -lavdevice$(BUILDSUF) -lavformat$(BUILDSUF) -lavcodec$(BUILDSUF) -lavutil$(BUILDSUF) $(EXTRALIBS)
+LDFLAGS := -L$(BUILD_ROOT)/libavformat -L$(BUILD_ROOT)/libavcodec -L$(BUILD_ROOT)/libavutil $(LDFLAGS)
+EXTRALIBS := -lavformat$(BUILDSUF) -lavcodec$(BUILDSUF) -lavutil$(BUILDSUF) $(EXTRALIBS)
+
+ifeq ($(CONFIG_AVDEVICES),yes)
+LDFLAGS+=-L$(BUILD_ROOT)/libavdevice 
+EXTRALIBS+=-lavdevice$(BUILDSUF) 
+endif
 
 ifeq ($(CONFIG_SWSCALER),yes)
 LDFLAGS+=-L$(BUILD_ROOT)/libswscale
@@ -75,7 +80,9 @@
 	$(MAKE) -C libavutil   all
 	$(MAKE) -C libavcodec  all
 	$(MAKE) -C libavformat all
+ifeq ($(CONFIG_AVDEVICES),yes)
 	$(MAKE) -C libavdevice all
+endif
 ifeq ($(CONFIG_PP),yes)
 	$(MAKE) -C libpostproc all
 endif
@@ -167,7 +174,9 @@
 	$(MAKE) -C libavutil   install-libs
 	$(MAKE) -C libavcodec  install-libs
 	$(MAKE) -C libavformat install-libs
+ifeq ($(CONFIG_AVDEVICES),yes)
 	$(MAKE) -C libavdevice install-libs
+endif
 ifeq ($(CONFIG_PP),yes)
 	$(MAKE) -C libpostproc install-libs
 endif
@@ -183,7 +192,9 @@
 	$(MAKE) -C libavutil   install-headers
 	$(MAKE) -C libavcodec  install-headers
 	$(MAKE) -C libavformat install-headers
+ifeq ($(CONFIG_AVDEVICES),yes)
 	$(MAKE) -C libavdevice install-headers
+endif
 ifeq ($(CONFIG_PP),yes)
 	$(MAKE) -C libpostproc install-headers
 endif
@@ -224,7 +235,9 @@
 	$(MAKE) -C libavutil   depend
 	$(MAKE) -C libavcodec  depend
 	$(MAKE) -C libavformat depend
+ifeq ($(CONFIG_AVDEVICES),yes)
 	$(MAKE) -C libavdevice depend
+endif
 ifeq ($(CONFIG_PP),yes)
 	$(MAKE) -C libpostproc depend
 endif
