Index: libavcodec/allcodecs.c
===================================================================
--- libavcodec/allcodecs.c	(revision 5823)
+++ libavcodec/allcodecs.c	(working copy)
@@ -229,11 +229,9 @@
 #ifdef CONFIG_VC1_DECODER
     register_avcodec(&vc1_decoder);
 #endif //CONFIG_VC1_DECODER
-/* Reenable when it stops crashing on every file, causing bug report spam.
 #ifdef CONFIG_WMV3_DECODER
     register_avcodec(&wmv3_decoder);
 #endif //CONFIG_WMV3_DECODER
-*/
 #ifdef CONFIG_H263I_DECODER
     register_avcodec(&h263i_decoder);
 #endif //CONFIG_H263I_DECODER
Index: libavcodec/avcodec.h
===================================================================
--- libavcodec/avcodec.h	(revision 5852)
+++ libavcodec/avcodec.h	(working copy)
@@ -17,8 +17,8 @@
 #define AV_STRINGIFY(s)         AV_TOSTRING(s)
 #define AV_TOSTRING(s) #s
 
-#define LIBAVCODEC_VERSION_INT  ((51<<16)+(10<<8)+0)
-#define LIBAVCODEC_VERSION      51.10.0
+#define LIBAVCODEC_VERSION_INT  ((51<<16)+(10<<8)+1)
+#define LIBAVCODEC_VERSION      51.10.1
 #define LIBAVCODEC_BUILD        LIBAVCODEC_VERSION_INT
 
 #define LIBAVCODEC_IDENT        "Lavc" AV_STRINGIFY(LIBAVCODEC_VERSION)
