Index: Makefile
===================================================================
RCS file: /cvsroot/ffmpeg/ffmpeg/Makefile,v
retrieving revision 1.99
diff -u -r1.99 Makefile
--- Makefile	13 Feb 2006 14:28:32 -0000	1.99
+++ Makefile	14 Feb 2006 13:39:47 -0000
@@ -151,6 +151,7 @@
 
 install-headers:
 	install -d "$(incdir)"
+	install -d "$(prefix)/include/postproc"
 	install -d "$(libdir)/pkgconfig"
 	$(MAKE) -C libavutil   install-headers
 	$(MAKE) -C libavcodec  install-headers
