Index: libpostproc/postprocess.c
===================================================================
--- libpostproc/postprocess.c	(révision 14661)
+++ libpostproc/postprocess.c	(copie de travail)
@@ -552,9 +552,6 @@
 
 //Note: we have C, MMX, MMX2, 3DNOW version there is no 3DNOW+MMX2 one
 //Plain C versions
-#if !defined (HAVE_MMX) || defined (RUNTIME_CPUDETECT)
-#define COMPILE_C
-#endif
 
 #ifdef HAVE_ALTIVEC
 #define COMPILE_ALTIVEC
@@ -562,17 +559,8 @@
 
 #if defined(ARCH_X86)
 
-#if (defined (HAVE_MMX) && !defined (HAVE_3DNOW) && !defined (HAVE_MMX2)) || defined (RUNTIME_CPUDETECT)
-#define COMPILE_MMX
-#endif
+#define COMPILE_C
 
-#if defined (HAVE_MMX2) || defined (RUNTIME_CPUDETECT)
-#define COMPILE_MMX2
-#endif
-
-#if (defined (HAVE_3DNOW) && !defined (HAVE_MMX2)) || defined (RUNTIME_CPUDETECT)
-#define COMPILE_3DNOW
-#endif
 #endif /* defined(ARCH_X86) */
 
 #undef HAVE_MMX
@@ -640,15 +628,7 @@
     // someone might exchange the CPU whithout restarting MPlayer ;)
 #ifdef RUNTIME_CPUDETECT
 #if defined(ARCH_X86)
-    // ordered per speed fastest first
-    if(c->cpuCaps & PP_CPU_CAPS_MMX2)
-        postProcess_MMX2(src, srcStride, dst, dstStride, width, height, QPs, QPStride, isColor, c);
-    else if(c->cpuCaps & PP_CPU_CAPS_3DNOW)
-        postProcess_3DNow(src, srcStride, dst, dstStride, width, height, QPs, QPStride, isColor, c);
-    else if(c->cpuCaps & PP_CPU_CAPS_MMX)
-        postProcess_MMX(src, srcStride, dst, dstStride, width, height, QPs, QPStride, isColor, c);
-    else
-        postProcess_C(src, srcStride, dst, dstStride, width, height, QPs, QPStride, isColor, c);
+    postProcess_C(src, srcStride, dst, dstStride, width, height, QPs, QPStride, isColor, c);
 #else
 #ifdef HAVE_ALTIVEC
     if(c->cpuCaps & PP_CPU_CAPS_ALTIVEC)
Index: libswscale/yuv2rgb.c
===================================================================
--- libswscale/yuv2rgb.c	(révision 27432)
+++ libswscale/yuv2rgb.c	(copie de travail)
@@ -148,6 +148,7 @@
 };
 #endif
 
+#if 0
 #ifdef HAVE_MMX
 
 /* hope these constant values are cache line aligned */
@@ -181,6 +182,7 @@
 #include "yuv2rgb_template.c"
 
 #endif /* HAVE_MMX */
+#endif
 
 const int32_t Inverse_Table_6_9[8][4] = {
     {117504, 138453, 13954, 34903}, /* no sequence_display_extension */
Index: libswscale/swscale.c
===================================================================
--- libswscale/swscale.c	(révision 27432)
+++ libswscale/swscale.c	(copie de travail)
@@ -820,9 +820,7 @@
 
 //Note: we have C, X86, MMX, MMX2, 3DNOW version therse no 3DNOW+MMX2 one
 //Plain C versions
-#if !defined (HAVE_MMX) || defined (RUNTIME_CPUDETECT) || !defined(CONFIG_GPL)
 #define COMPILE_C
-#endif
 
 #ifdef ARCH_POWERPC
 #if (defined (HAVE_ALTIVEC) || defined (RUNTIME_CPUDETECT)) && defined (CONFIG_GPL)
@@ -830,6 +828,7 @@
 #endif //HAVE_ALTIVEC
 #endif //ARCH_POWERPC
 
+#if 0
 #if defined(ARCH_X86)
 
 #if ((defined (HAVE_MMX) && !defined (HAVE_3DNOW) && !defined (HAVE_MMX2)) || defined (RUNTIME_CPUDETECT)) && defined (CONFIG_GPL)
@@ -844,6 +843,7 @@
 #define COMPILE_3DNOW
 #endif
 #endif //ARCH_X86 || ARCH_X86_64
+#endif
 
 #undef HAVE_MMX
 #undef HAVE_MMX2
@@ -1490,15 +1490,8 @@
 #if defined(RUNTIME_CPUDETECT) && defined (CONFIG_GPL)
 #if defined(ARCH_X86)
     // ordered per speed fastest first
-    if (flags & SWS_CPU_CAPS_MMX2)
-        return swScale_MMX2;
-    else if (flags & SWS_CPU_CAPS_3DNOW)
-        return swScale_3DNow;
-    else if (flags & SWS_CPU_CAPS_MMX)
-        return swScale_MMX;
-    else
-        return swScale_C;
-
+    /* VLC Hack: only C */
+    return swScale_C;
 #else
 #ifdef ARCH_POWERPC
     if (flags & SWS_CPU_CAPS_ALTIVEC)
Index: libswscale/rgb2rgb.c
===================================================================
--- libswscale/rgb2rgb.c	(révision 27432)
+++ libswscale/rgb2rgb.c	(copie de travail)
@@ -158,6 +158,7 @@
 #define RENAME(a) a ## _C
 #include "rgb2rgb_template.c"
 
+#if 0
 #if defined(ARCH_X86) && defined(CONFIG_GPL)
 
 //MMX versions
@@ -188,6 +189,7 @@
 #include "rgb2rgb_template.c"
 
 #endif //ARCH_X86 || ARCH_X86_64
+#endif
 
 /*
  RGB15->RGB16 original by Strepto/Astral
