Index: Makefile
===================================================================
--- Makefile    (revision 8597)
+++ Makefile    (working copy)
@@ -149,7 +149,9 @@
 ifeq ($(CONFIG_PP),yes)
 	(MAKE) -C libpostproc install-headers
 endif
+ifeq ($(CONFIG_SWSCALER),yes)
 	$(MAKE) -C libswscale  install-headers
+endif

 uninstall: uninstall-progs uninstall-libs uninstall-headers uninstall-man uninstall-vhook

