Index: configure
===================================================================
--- ffmpeg/configure	(revision 5823)
+++ ffmpeg/configure	(working copy)
@@ -2118,7 +2118,7 @@
 # build pkg-config files libav*.pc and libpostproc.pc
 # libavutil.pc
 cat <<EOF >libavutil.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2133,7 +2133,7 @@
 EOF
 
 cat <<EOF >libavutil-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavutil
 includedir=\${pcfiledir}/libavutil
@@ -2149,7 +2149,7 @@
 
 # libavcodec.pc
 cat <<EOF >libavcodec.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2164,7 +2164,7 @@
 EOF
 
 cat <<EOF >libavcodec-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavcodec
 includedir=\${pcfiledir}/libavcodec
@@ -2180,7 +2180,7 @@
 
 # libavformat.pc
 cat <<EOF >libavformat.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2195,7 +2195,7 @@
 EOF
 
 cat <<EOF >libavformat-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libavformat
 includedir=\${pcfiledir}/libavformat
@@ -2212,7 +2212,7 @@
 
 # libpostproc.pc
 cat <<EOF >libpostproc.pc
-prefix=$PREFIX
+PREFIX=$PREFIX
 exec_prefix=\${PREFIX}
 libdir=\${exec_prefix}/lib
 includedir=\${PREFIX}/include
@@ -2227,7 +2227,7 @@
 EOF
 
 cat <<EOF >libpostproc-uninstalled.pc
-prefix=
+PREFIX=
 exec_prefix=
 libdir=\${pcfiledir}/libpostproc
 includedir=\${pcfiledir}/libpostproc
