? win-ffmpeg.sh
Index: libavcodec/libpostproc/postprocess.c
===================================================================
RCS file: /cvsroot/ffmpeg/ffmpeg/libavcodec/libpostproc/postprocess.c,v
retrieving revision 1.110
diff -u -r1.110 postprocess.c
--- libavcodec/libpostproc/postprocess.c	11 May 2005 23:51:44 -0000	1.110
+++ libavcodec/libpostproc/postprocess.c	13 May 2005 09:22:27 -0000
@@ -92,6 +92,8 @@
 
 #include "mangle.h" //FIXME should be supressed
 
+#define RUNTIME_CPUDETECT 1
+
 #ifdef HAVE_ALTIVEC_H
 #include <altivec.h>
 #endif
