Index: libavutil/common.h
===================================================================
RCS file: /cvsroot/ffmpeg/ffmpeg/libavutil/common.h,v
retrieving revision 1.153
diff -u -r1.153 common.h
--- ffmpeg.orig/libavutil/common.h	19 Sep 2005 23:26:47 -0000	1.153
+++ ffmpeg/libavutil/common.h	1 Nov 2005 21:40:31 -0000
@@ -543,7 +543,7 @@
     }\
 }
 
-#ifndef HAVE_LRINTF
+#if !defined(HAVE_LRINTF) && !defined(__APPLE__)
 /* XXX: add ISOC specific test to avoid specific BSD testing. */
 /* better than nothing implementation. */
 /* btw, rintf() is existing on fbsd too -- alex */
