Index: configure
===================================================================
--- configure	(révision 13435)
+++ configure	(copie de travail)
@@ -2169,11 +2169,13 @@
 version=$3
 libs=$4
 requires=$5
+libdir_pc=`echo $libdir|sed s'|$(PREFIX)|${prefix}|'`
+incdir_pc=`echo $incdir|sed 's|$(PREFIX)|${prefix}|'`
 cat <<EOF > $name.pc
 prefix=$prefix
 exec_prefix=\${prefix}
-libdir=$libdir
-includedir=$incdir
+libdir=$libdir_pc
+includedir=$incdir_pc
 
 Name: $name
 Description: $comment
