diff -ruN flac.old/src/libFLAC/bitreader.c flac/src/libFLAC/bitreader.c
--- flac.old/src/libFLAC/bitreader.c	2007-02-17 04:34:31.000000000 +0100
+++ flac/src/libFLAC/bitreader.c	2007-02-17 04:35:31.000000000 +0100
@@ -37,6 +37,8 @@
 #include <string.h> /* for memcpy(), memset() */
 #if defined(_MSC_VER) && _MSC_VER <= 1200
 #include <winsock.h> /* for ntohl() */
+#elif defined(__MINGW32__)
+#include <winsock.h> /* for ntohl() */
 #elif defined FLAC__SYS_DARWIN
 #include <machine/endian.h> /* for ntohl() */
 #else
diff -ruN flac.old/src/libFLAC/bitwriter.c flac/src/libFLAC/bitwriter.c
--- flac.old/src/libFLAC/bitwriter.c	2007-02-17 04:34:31.000000000 +0100
+++ flac/src/libFLAC/bitwriter.c	2007-02-17 04:35:39.000000000 +0100
@@ -37,6 +37,8 @@
 #include <string.h> /* for memcpy(), memset() */
 #if defined(_MSC_VER) && _MSC_VER <= 1200
 #include <winsock.h> /* for ntohl() */
+#elif defined(__MINGW32__)
+#include <winsock.h> /* for ntohl() */
 #elif defined FLAC__SYS_DARWIN
 #include <machine/endian.h> /* for ntohl() */
 #else
