--- fontconfig/src/fcxml.c	2009-08-13 16:07:13.000000000 +0200
+++ fontconfig.new/src/fcxml.c	2009-08-13 16:05:41.000000000 +0200
@@ -23,6 +23,8 @@
  */
 
 #include "fcint.h"
+
+#if 0
 #include <fcntl.h>
 #include <stdarg.h>
 #include <dirent.h>
@@ -58,6 +60,8 @@
 #undef STRICT
 #endif
 
+#endif
+
 static void
 FcExprDestroy (FcExpr *e);
 
@@ -71,6 +75,7 @@
     free (test);
 }
 
+#if 0
 static FcExpr *
 FcExprCreateInteger (FcConfig *config, int i)
 {
@@ -179,7 +184,7 @@
     }
     return e;
 }
-
+#endif
 static void
 FcExprDestroy (FcExpr *e)
 {
@@ -255,6 +260,7 @@
     free (e);
 }
 
+#if 0
 typedef enum _FcElement {
     FcElementNone,
     FcElementFontconfig,
@@ -2373,6 +2379,7 @@
     const FcChar8    *bs = *((FcChar8 **) b);
     return FcStrCmp (as, bs);
 }
+#endif
 
 static FcBool
 FcConfigParseAndLoadDir (FcConfig	*config,
@@ -2380,6 +2387,9 @@
 			 const FcChar8	*dir,
 			 FcBool		complain)
 {
+    return FcFalse;
+}
+#if 0
     DIR		    *d;
     struct dirent   *e;
     FcBool	    ret = FcTrue;
@@ -2589,3 +2599,4 @@
 #define __fcxml__
 #include "fcaliastail.h"
 #undef __fcxml__
+#endif
diff -ruN fontconfig/src/fcxml.c fontconfig.new/src/fcxml.c
--- fontconfig/src/fcxml.c	2009-08-13 16:08:49.000000000 +0200
+++ fontconfig.new/src/fcxml.c	2009-08-19 00:03:35.000000000 +0200
@@ -2466,13 +2466,18 @@
 bail0:
     return ret || !complain;
 }
+#endif
+
 
 FcBool
 FcConfigParseAndLoad (FcConfig	    *config,
 		      const FcChar8 *name,
 		      FcBool	    complain)
 {
+    return FcFalse;
+}
 
+#if 0
     XML_Parser	    p;
     FcChar8	    *filename;
     int		    fd;
