--- fribidi/config.sub.orig	2001-08-23 12:40:51.000000000 +0200
+++ fribidi/config.sub	2007-05-16 12:19:10.000000000 +0200
@@ -260,7 +260,7 @@
 	# FIXME: clean up the formatting here.
 	vax-* | tahoe-* | i*86-* | i860-* | ia64-* | m32r-* | m68k-* | m68000-* \
 	      | m88k-* | sparc-* | ns32k-* | fx80-* | arc-* | c[123]* \
-	      | arm-*  | armbe-* | armle-* | armv*-* | strongarm-* | xscale-* \
+	      | arm-*  | armbe-* | armle-* | armeb-* | armel-*| armv*-* | strongarm-* | xscale-* \
 	      | mips-* | pyramid-* | tron-* | a29k-* | romp-* | rs6000-* \
 	      | power-* | none-* | 580-* | cray2-* | h8300-* | h8500-* | i960-* \
 	      | xmp-* | ymp-* \
@@ -1000,6 +1000,9 @@
 	-unixware*)
 		os=-sysv4.2uw
 		;;
+	-uclibc*)
+		os=-linux
+		;;
 	-gnu/linux*)
 		os=`echo $os | sed -e 's|gnu/linux|linux-gnu|'`
 		;;
@@ -1154,6 +1157,9 @@
 	arm*-semi)
 		os=-aout
 		;;
+	arm*-linux)
+		os=-linux
+		;;
 	pdp10-*)
 		os=-tops20
 		;;
