diff -u fribidi.orig/Makefile.am fribidi/Makefile.am
--- fribidi.orig/Makefile.am	2005-08-25 20:35:34.000000000 +0200
+++ fribidi/Makefile.am	2008-01-28 14:31:19.000000000 +0100
@@ -9,7 +9,7 @@
 ##############################################################################
 
 ## The order of subdirs is important, don't change without a reason.
-SUBDIRS = gen.tab charset lib bin doc test
+SUBDIRS = gen.tab charset lib #bin doc test
 
 EXTRA_DIST = bootstrap ChangeLog.old
 
diff -u fribidi.orig/configure.ac fribidi/configure.ac
--- fribidi.orig/configure.ac	2008-01-16 00:15:46.000000000 +0100
+++ fribidi/configure.ac	2008-01-28 14:32:38.000000000 +0100
@@ -165,38 +165,38 @@
 AM_CONDITIONAL(FRIBIDI_CHARSETS, test x$FRIBIDI_CHARSETS = x1)
 
 # --with[out]-glib
-AC_ARG_WITH(glib,
-              AC_HELP_STRING([--with-glib=@<:@no/auto/yes@:>@],
-                             [use Glib @<:@default=auto@:>@]))
-GLIB_PACKAGE=glib-2.0
-GLIB_MINVERSION=2.4
+#AC_ARG_WITH(glib,
+#              AC_HELP_STRING([--with-glib=@<:@no/auto/yes@:>@],
+#                             [use Glib @<:@default=auto@:>@]))
+#GLIB_PACKAGE=glib-2.0
+#GLIB_MINVERSION=2.4
 GLIB_LIBS=
 GLIB_CFLAGS=
-if test x$with_glib = xyes; then
-	PKG_CHECK_MODULES(GLIB,$GLIB_PACKAGE >= $GLIB_MINVERSION)
-	FRIBIDI_USE_GLIB=1
-else
-if test x$with_glib = xno; then
+#if test x$with_glib = xyes; then
+#	PKG_CHECK_MODULES(GLIB,$GLIB_PACKAGE >= $GLIB_MINVERSION)
+#	FRIBIDI_USE_GLIB=1
+#else
+#if test x$with_glib = xno; then
 	FRIBIDI_USE_GLIB=0
-else
-	PKG_CHECK_MODULES(GLIB,$GLIB_PACKAGE >= $GLIB_MINVERSION,
-		FRIBIDI_USE_GLIB=1,
-		FRIBIDI_USE_GLIB=0)
-fi
-fi
-if test x$FRIBIDI_USE_GLIB = x0; then
+#else
+#	PKG_CHECK_MODULES(GLIB,$GLIB_PACKAGE >= $GLIB_MINVERSION,
+#		FRIBIDI_USE_GLIB=1,
+#		FRIBIDI_USE_GLIB=0)
+#fi
+#fi
+#if test x$FRIBIDI_USE_GLIB = x0; then
 	GLIB_PACKAGE=
-fi
+#fi
 AC_SUBST(FRIBIDI_USE_GLIB)
 AM_CONDITIONAL(FRIBIDI_USE_GLIB, test x$FRIBIDI_USE_GLIB = x1)
 
 
-MISC_CFLAGS="$GLIB_CFLAGS"
-MISC_LIBS="$GLIB_LIBS"
-MISC_PACKAGES="$GLIB_PACKAGE"
-AC_SUBST(MISC_CFLAGS)
-AC_SUBST(MISC_LIBS)
-AC_SUBST(MISC_PACKAGES)
+#MISC_CFLAGS="$GLIB_CFLAGS"
+#MISC_LIBS="$GLIB_LIBS"
+#MISC_PACKAGES="$GLIB_PACKAGE"
+#AC_SUBST(MISC_CFLAGS)
+#AC_SUBST(MISC_LIBS)
+#AC_SUBST(MISC_PACKAGES)
 
 # Generate output
 AC_CONFIG_FILES([fribidi.pc
--- fribidi.orig/fribidi.pc.in	2008-02-02 23:06:05.000000000 +0000
+++ fribidi/fribidi.pc.in	2008-02-02 23:06:39.000000000 +0000
@@ -7,7 +7,6 @@
 
 Name: @PACKAGE_NAME@
 Description: Unicode Bidirectional Algorithm Library
-Requires: @MISC_PACKAGES@
 Version: @VERSION@
 Libs: -L${libdir} -lfribidi
 Cflags: -I${includedir}/@PACKAGE@
