--- gettext-0.17/gettext-runtime/intl/Makefile.in	2007-11-04 21:21:12.000000000 +0000
+++ gettext/gettext-runtime/intl/Makefile.in	2008-03-07 18:14:53.000000000 +0000
@@ -280,15 +280,7 @@
 # so that they work with the sed-3.02 that is shipped with MSYS. We can use
 # GNU bash's $'\n' syntax to obtain such a newline.
 libintl.res: $(srcdir)/libintl.rc
-	nl=$$'\n'; \
-	sed_extract_major='/^[0-9]/{'$${nl}'s/^\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
-	sed_extract_minor='/^[0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
-	sed_extract_subminor='/^[0-9][0-9]*[.][0-9][0-9]*[.][0-9]/{'$${nl}'s/^[0-9]*[.][0-9]*[.]\([0-9]*\).*/\1/p'$${nl}q$${nl}'}'$${nl}'c\'$${nl}0$${nl}q; \
-	$(WINDRES) \
-	  "-DPACKAGE_VERSION_STRING=\\\"$(VERSION)\\\"" \
-	  "-DPACKAGE_VERSION_MAJOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_major"` \
-	  "-DPACKAGE_VERSION_MINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_minor"` \
-	  "-DPACKAGE_VERSION_SUBMINOR="`echo '$(VERSION)' | sed -n -e "$$sed_extract_subminor"` \
+	$(WINDRES) --use-temp-file `$(top_srcdir)/../windows/windres-options --escape $(VERSION)`\
 	  -i $(srcdir)/libintl.rc -o libintl.res --output-format=coff
 
 ref-add.sed: $(srcdir)/ref-add.sin
--- gettext-0.17/gettext-runtime/intl/libintl.rc        2007-05-27 21:46:52.000000000 +0000
+++ gettext/gettext-runtime/intl/libintl.rc     2009-05-18 18:09:12.000000000 +0000
@@ -22,13 +22,13 @@
             VALUE "Comments", "This library is free software; you can redistribute it and/or modify it under the terms of the GNU Library General Public License. You should have received a copy of the GNU Library General Public License along with this library; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA\0"
             VALUE "CompanyName", "Free Software Foundation\0"
             VALUE "FileDescription", "LGPLed libintl for Windows NT/2000/XP/Vista and Windows 95/98/ME\0"
-            VALUE "FileVersion", PACKAGE_VERSION_STRING "\0"
+            VALUE "FileVersion", "\0"
             VALUE "InternalName", "intl.dll\0"
             VALUE "LegalCopyright", "Copyright (C) 1995-2007\0"
             VALUE "LegalTrademarks", "\0"
             VALUE "OriginalFilename", "intl.dll\0"
             VALUE "ProductName", "libintl: accessing NLS message catalogs\0"
-            VALUE "ProductVersion", PACKAGE_VERSION_STRING "\0"
+            VALUE "ProductVersion", "\0"
         END
     END
     BLOCK "VarFileInfo"
