diff -ruN gettext/gettext-runtime/libasprintf/Makefile.am gettext.old/gettext-runtime/libasprintf/Makefile.am
--- gettext/gettext-runtime/libasprintf/Makefile.am	2006-10-20 13:49:02.000000000 +0200
+++ gettext.old/gettext-runtime/libasprintf/Makefile.am	2006-11-03 01:30:44.000000000 +0100
@@ -98,7 +98,7 @@
 #   make autosprintf.pdf
 
 all-local: html-local
-install-data-local: install-html
+install-data-local: 
 installdirs-local: installdirs-html
 uninstall-local: uninstall-html
 
diff -ruN gettext/gettext-runtime/libasprintf/Makefile.in gettext.old/gettext-runtime/libasprintf/Makefile.in
--- gettext/gettext-runtime/libasprintf/Makefile.in	2006-11-03 01:56:38.000000000 +0100
+++ gettext.old/gettext-runtime/libasprintf/Makefile.in	2006-11-03 01:35:12.000000000 +0100
@@ -881,7 +881,7 @@
 #   make autosprintf.pdf
 
 all-local: html-local
-install-data-local: install-html
+install-data-local: 
 installdirs-local: installdirs-html
 uninstall-local: uninstall-html
 # autosprintf.dvi and autosprintf.ps are already known to automake.
diff -ruN gettext/gettext-runtime/man/Makefile.in gettext.old/gettext-runtime/man/Makefile.in
--- gettext/gettext-runtime/man/Makefile.in	2006-11-03 01:56:28.000000000 +0100
+++ gettext.old/gettext-runtime/man/Makefile.in	2006-11-03 02:15:57.000000000 +0100
@@ -695,6 +695,8 @@
 	$(MAN2HTML) $(srcdir)/bind_textdomain_codeset.3.in | sed -e '/CreationDate:/d' > t-$@
 	mv t-$@ $@
 
+install-html: install-html-local
+
 install-html-local:
 	$(mkdir_p) $(DESTDIR)$(htmldir)
 	for file in $(man_HTML); do \
diff -ruN gettext/gettext-tools/man/Makefile.in gettext.old/gettext-tools/man/Makefile.in
--- gettext/gettext-tools/man/Makefile.in	2006-11-03 01:57:09.000000000 +0100
+++ gettext.old/gettext-tools/man/Makefile.in	2006-11-03 01:52:46.000000000 +0100
@@ -802,6 +802,8 @@
 	$(MAN2HTML) `if test -f autopoint.1; then echo .; else echo $(srcdir); fi`/autopoint.1 | sed -e '/CreationDate:/d' > t-$@
 	mv t-$@ $@
 
+install-html: install-html-local
+
 install-html-local:
 	$(mkdir_p) $(DESTDIR)$(htmldir)
 	for file in $(man_HTML); do \
