--- glib/gconvert.c	2009-06-27 05:18:06.000000000 +0200
+++ glib/gconvert.c	2009-07-18 02:09:30.000000000 +0200
@@ -51,9 +51,6 @@
 #if defined(USE_LIBICONV_GNU) && !defined (_LIBICONV_H)
 #error GNU libiconv in use but included iconv.h not from libiconv
 #endif
-#if !defined(USE_LIBICONV_GNU) && defined (_LIBICONV_H)
-#error GNU libiconv not in use but included iconv.h is from libiconv
-#endif
 
 #include "galias.h"
 
