diff -ru goom2k4-0/configure goom/configure
--- goom2k4-0/configure	Mon Feb  7 14:49:28 2005
+++ goom/configure	Wed Feb 23 16:03:25 2005
@@ -21069,8 +21069,8 @@
 
 case "$host" in
 *-apple-darwin*)
-	MACTARGET="yes"
-	MACFOLDER="mac"
+	MACTARGET="no"
+	MACFOLDER=""
 
   	CCAS='$(CC)'
 
--- goom/src/Makefile.am	2005-02-07 14:46:41.000000000 +0100
+++ goom.new/src/Makefile.am	2009-08-19 09:21:57.000000000 +0200
@@ -16,7 +16,7 @@
 goom2_libdir = $(libdir)
 
 goom2_library_includedir=$(includedir)/goom
-goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_tools.h goom_config.h
+goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_config.h
 libgoom2_la_LDFLAGS = -export-dynamic -export-symbols-regex "goom.*" 
 libgoom2_la_SOURCES = \
 	goomsl_yacc.y goomsl_lex.l goomsl.c goomsl_hash.c goomsl_heap.c \
