diff -ur goom.orig/configure.in goom/configure.in
--- goom.orig/configure.in	2007-09-20 17:16:48.000000000 +0200
+++ goom/configure.in	2007-09-20 17:17:36.000000000 +0200
@@ -1,6 +1,5 @@
 AC_INIT(README)
 
-AM_DISABLE_STATIC
 AM_INIT_AUTOMAKE(SDL_Goom, 2k4)
 
 ACLOCAL="$ACLOCAL -I m4"
@@ -19,17 +18,13 @@
 
 dnl *** check for xmms ***
 
-AM_PATH_XMMS2(0.9.5.1, HAVE_XMMS="yes", HAVE_XMMS="no")
+HAVE_XMMS="no"
 AM_CONDITIONAL(HAVE_XMMS,test "x$HAVE_XMMS" = "xyes")
 
 
 dnl *** SDL ***
 
-if test "x$HAVE_XMMS" = "xyes"; then
-  AM_PATH_SDL2(1.2.0, HAVE_SDL="yes", HAVE_SDL="no"])
-else
-  HAVE_SDL="no"
-fi
+HAVE_SDL="no"
 AM_CONDITIONAL(HAVE_SDL,test "x$HAVE_SDL" = "xyes")
 
 
@@ -76,24 +71,7 @@
 AM_CONDITIONAL(MACTARGET,test "x$MACTARGET" = "xyes")
 
 
-AC_CHECK_HEADER(pthread.h,,AC_MSG_ERROR([*** POSIX thread support not installed - please install first ***]))
-
-PTHREAD_LIBS=error
-AC_CHECK_LIB(pthread, pthread_attr_init, PTHREAD_LIBS="-lpthread")
-
-if test "x$PTHREAD_LIBS" = xerror; then
-    AC_CHECK_LIB(pthreads, pthread_attr_init, PTHREAD_LIBS="-lpthreads")
-fi
-
-if test "x$PTHREAD_LIBS" = xerror; then
-    AC_CHECK_LIB(c_r, pthread_attr_init, PTHREAD_LIBS="-lc_r")
-fi
-
-if test "x$PTHREAD_LIBS" = xerror; then
-    AC_CHECK_FUNC(pthread_attr_init, PTHREAD_LIBS="")
-fi
-
-AC_SUBST(PTHREAD_LIBS)
+dnl AC_SUBST(PTHREAD_LIBS)
 
 dnl rm -f mmx_zoom.s
 dnl 	echo -n checking for nasm...
diff -ur goom2k4-0.orig/src/filters.c goom2k4-0/src/filters.c
--- goom.orig/src/filters.c	2005-02-07 14:46:41.000000000 +0100
+++ goom/src/filters.c	2007-09-20 17:17:11.000000000 +0200
@@ -201,8 +201,8 @@
     /* Noise */
     if (data->noisify)
     {
-        vx += (((float)random()) / ((float)RAND_MAX) - 0.5f) / 50.0f;
-        vy += (((float)random()) / ((float)RAND_MAX) - 0.5f) / 50.0f;
+        vx += (((float)rand()) / ((float)RAND_MAX) - 0.5f) / 50.0f;
+        vy += (((float)rand()) / ((float)RAND_MAX) - 0.5f) / 50.0f;
     }
     
     /* Hypercos */
diff -ur goom/src/goom_core.c goom/src/goom_core.c
--- goom/src/goom_core.c	2005-02-07 14:46:41.000000000 +0100
+++ goom/src/goom_core.c	2007-09-20 17:17:11.000000000 +0200
@@ -40,11 +40,11 @@
 static void init_buffers(PluginInfo *goomInfo, int buffsize)
 {
     goomInfo->pixel = (guint32 *) malloc (buffsize * sizeof (guint32) + 128);
-    bzero (goomInfo->pixel, buffsize * sizeof (guint32) + 128);
+    memset (goomInfo->pixel, 0, buffsize * sizeof (guint32) + 128);
     goomInfo->back = (guint32 *) malloc (buffsize * sizeof (guint32) + 128);
-    bzero (goomInfo->back, buffsize * sizeof (guint32) + 128);
+    memset (goomInfo->back, 0, buffsize * sizeof (guint32) + 128);
     goomInfo->conv = (Pixel *) malloc (buffsize * sizeof (guint32) + 128);
-    bzero (goomInfo->conv, buffsize * sizeof (guint32) + 128);
+    memset (goomInfo->conv, 0, buffsize * sizeof (guint32) + 128);
 
     goomInfo->outputBuf = goomInfo->conv;
     
