diff --git a/libass/ass_fontconfig.c b/libass/ass_fontconfig.c
index 006be97..a4a680f 100644
--- a/libass/ass_fontconfig.c
+++ b/libass/ass_fontconfig.c
@@ -449,7 +449,7 @@ FCInstance *fontconfig_init(ASS_Library *library,
     }
 
     priv->config = FcConfigCreate();
-    rc = FcConfigParseAndLoad(priv->config, (unsigned char *) config, FcTrue);
+    rc = 0;//FcConfigParseAndLoad(priv->config, (unsigned char *) config, FcTrue);
     if (!rc) {
         ass_msg(library, MSGL_WARN, "No usable fontconfig configuration "
                 "file found, using fallback.");
