--- libcddb.orig/include/cddb/cddb_regex.h	Mon Jul  5 02:12:12 2004
+++ libcddb/include/cddb/cddb_regex.h	Sun Jul  4 22:05:44 2004
@@ -28,6 +28,7 @@
 
 
 #include <stdlib.h>
+#include <unistd.h>
 #include <regex.h>
 
 
--- libcddb.orig/lib/cddb_net.c	Mon Jul  5 22:07:40 2004
+++ libcddb/lib/cddb_net.c	Mon Jul  5 22:05:53 2004
@@ -24,7 +24,7 @@
 #include <netdb.h>
 #include <setjmp.h>
 #include <signal.h>
-#include <time.h>
+#include <sys/time.h>
 #include <unistd.h>
 #include <sys/select.h>
 #include <sys/socket.h>
