--- libcdio-orig/lib/driver/osx.c	2007-08-12 14:38:25.000000000 +0200
+++ libcdio/lib/driver/osx.c	2008-12-27 00:38:21.000000000 +0100
@@ -58,7 +58,7 @@
 
 #include <mach/mach.h>
 #include <Carbon/Carbon.h>
-#include <IOKit/scsi-commands/SCSITaskLib.h>
+#include <IOKit/scsi/SCSITaskLib.h>
 #include <IOKit/IOCFPlugIn.h>
 #include <mach/mach_error.h>
 
@@ -77,7 +77,7 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
 #include <IOKit/IOBSD.h>
-#include <IOKit/scsi-commands/IOSCSIMultimediaCommandsDevice.h>
+#include <IOKit/scsi/IOSCSIMultimediaCommandsDevice.h>
 #include <IOKit/storage/IOCDTypes.h>
 #include <IOKit/storage/IODVDTypes.h>
 #include <IOKit/storage/IOMedia.h>
