--- libdvbpsi/examples/decode_mpeg.c	2010-02-12 11:17:47.000000000 +0100
+++ libdvbpsi.new/examples/decode_mpeg.c	2011-04-29 02:21:40.758841339 +0200
@@ -529,8 +529,8 @@
 
     int i_fd = -1;
     int i_mtu = 1316; /* (7 * 188) = 1316 < 1500 network MTU */
-#ifdef HAVE_SYS_SOCKET_H
     int i_report = REPORT_UDP; /* REPORT_PCR REPORT_UDP */
+#ifdef HAVE_SYS_SOCKET_H
     int i_port = 0;
     char *ipaddress = NULL;
 #endif
