Index: src/dvd_reader.c
===================================================================
--- src/dvd_reader.c	(revision 1157)
+++ src/dvd_reader.c	(working copy)
@@ -34,6 +34,7 @@
 #include <limits.h>
 #include <dirent.h>
 
+#if 0
 /* misc win32 helpers */
 #ifdef WIN32
 #ifndef HAVE_GETTIMEOFDAY
@@ -52,6 +53,7 @@
 #include <io.h> /* read() */
 #define lseek64 _lseeki64
 #endif
+#endif
 
 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DARWIN__)
 #define SYS_BSD 1
Index: src/dvdread/ifo_types.h
===================================================================
--- src/dvdread/ifo_types.h	(révision 1188)
+++ src/dvdread/ifo_types.h	(copie de travail)
@@ -32,7 +32,7 @@
 
 #if defined(__GNUC__)
 #if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 95)
-#define ATTRIBUTE_PACKED __attribute__ ((packed))
+#define ATTRIBUTE_PACKED __attribute__ ((packed,gcc_struct))
 #define PRAGMA_PACK 0
 #endif
 #endif
