Index: libebml/src/EbmlFloat.cpp
===================================================================
--- libebml/src/EbmlFloat.cpp	(revision 1099)
+++ libebml/src/EbmlFloat.cpp	(revision 1101)
@@ -102,8 +102,8 @@
 		if (Size == 4) {
 			big_int32 TmpRead;
 			TmpRead.Eval(Buffer);
-			float val = *((float *)&(int32(TmpRead)));
-			Value = val;
+			int32 tmpp = int32(TmpRead);
+			Value = *((float *) &tmpp);
 			bValueIsSet = true;
 		} else if (Size == 8) {
 			big_int64 TmpRead;
