diff -ruN libebml/ebml/EbmlCrc32.h libebml.new/ebml/EbmlCrc32.h
--- libebml/ebml/EbmlCrc32.h	2009-08-19 00:48:35.000000000 +0200
+++ libebml.new/ebml/EbmlCrc32.h	2009-08-19 00:49:35.000000000 +0200
@@ -143,7 +143,7 @@
 
 inline bool IsAlignedOn(const void *p, unsigned int alignment)
 {
-	return IsPowerOf2(alignment) ? ModPowerOf2((unsigned long)p, alignment) == 0 : (unsigned long)p % alignment == 0;
+	return IsPowerOf2(alignment) ? ModPowerOf2((uintptr_t)p, alignment) == 0 : (uintptr_t)p % alignment == 0;
 }
 
 template <class T>
