--- libgpg-error/src/gpg-error.c	2005-12-18 15:24:34.000000000 +0000
+++ libgpg-error.patch/src/gpg-error.c	2006-07-27 23:10:46.000000000 +0100
@@ -78,10 +78,14 @@
   locale_dir = get_locale_dir ();
   if (locale_dir)
     {
+#ifdef ENABLE_NLS
       bindtextdomain (PACKAGE, locale_dir);
+#endif
       drop_locale_dir (locale_dir);
     }
+#ifdef ENABLE_NLS
   textdomain (PACKAGE);
+#endif
 }
 
 
