diff -ru libmodplug-orig/src/load_abc.cpp libmodplug/src/load_abc.cpp
--- libmodplug-orig/src/load_abc.cpp	2006-11-02 03:18:26.000000000 +0100
+++ libmodplug/src/load_abc.cpp	2008-12-27 00:20:46.000000000 +0100
@@ -43,6 +43,10 @@
 
 #include "load_pat.h"
 
+#ifndef uint
+#define uint unsigned int
+#endif
+
 #define MAXABCINCLUDES	8
 #define MAXCHORDNAMES 80
 #define ABC_ENV_DUMPTRACKS		"MMABC_DUMPTRACKS"
diff -ru libmodplug-orig/src/load_mid.cpp libmodplug/src/load_mid.cpp
--- libmodplug-orig/src/load_mid.cpp	2006-11-02 02:50:47.000000000 +0100
+++ libmodplug/src/load_mid.cpp	2008-12-27 00:15:30.000000000 +0100
@@ -52,6 +52,10 @@
 
 #include "load_pat.h"
 
+#ifndef uint
+#define uint unsigned int
+#endif
+
 #define ROWSPERNOTE 16
 #define ENV_MMMID_SPEED	"MMMID_SPEED"
 #define ENV_MMMID_DEBUG "MMMID_DEBUG"
diff -ru libmodplug-orig/src/load_pat.cpp libmodplug/src/load_pat.cpp
--- libmodplug-orig/src/load_pat.cpp	2006-09-12 01:57:25.000000000 +0200
+++ libmodplug/src/load_pat.cpp	2008-12-27 00:20:43.000000000 +0100
@@ -47,6 +47,10 @@
 
 #include "load_pat.h"
 
+#ifndef uint
+#define uint unsigned int
+#endif
+
 #ifdef MSC_VER
 #define DIRDELIM		'\\'
 #define TIMIDITYCFG	"C:\\TIMIDITY\\TIMIDITY.CFG"
