--- libmodplug-0.8.4/src/Makefile.am	2006-11-02 04:48:13.000000000 +0100
+++ libmodplug/src/Makefile.am	2008-04-06 07:55:35.000000000 +0200
@@ -32,9 +32,6 @@
                         load_j2b.cpp       \
                         load_mt2.cpp       \
                         load_psm.cpp       \
-                        load_abc.cpp       \
-                        load_mid.cpp       \
-                        load_pat.cpp       \
                         modplug.cpp
 
 libmodplugincludedir = $(includedir)/libmodplug
diff -ru libmodplug/src/sndfile.cpp libmodplug-new/src/sndfile.cpp
--- libmodplug/src/sndfile.cpp	2006-11-02 04:31:26.000000000 +0000
+++ libmodplug-new/src/sndfile.cpp	2008-04-12 20:56:39.000000000 +0100
@@ -143,8 +143,5 @@
 #ifndef MODPLUG_BASIC_SUPPORT
 /* Sequencer File Format Support */
-		 && (!ReadABC(lpStream, dwMemLength))
-		 && (!ReadMID(lpStream, dwMemLength))
-		 && (!ReadPAT(lpStream, dwMemLength))
 		 && (!ReadSTM(lpStream, dwMemLength))
 		 && (!ReadMed(lpStream, dwMemLength))
 		 && (!ReadMTM(lpStream, dwMemLength))
