Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(révision 468)
+++ CMakeLists.txt	(copie de travail)
@@ -23,11 +23,9 @@
 add_subdirectory(libmpcpsy)
 add_subdirectory(libmpcenc)
 add_subdirectory(libwavformat)
-add_subdirectory(mpcgain)
 add_subdirectory(mpcdec)
 add_subdirectory(mpcenc)
 add_subdirectory(mpc2sv8)
 add_subdirectory(mpccut)
-add_subdirectory(mpcchap)
 add_subdirectory(wavcmp)
 add_subdirectory(include)
