diff -ruN liboil.new/m4/as-unaligned-access.m4 liboil/m4/as-unaligned-access.m4
--- liboil.new/m4/as-unaligned-access.m4	2009-08-18 10:59:02.000000000 +0200
+++ liboil/m4/as-unaligned-access.m4	2009-08-18 10:59:32.000000000 +0200
@@ -4,7 +4,7 @@
 AC_DEFUN([AS_UNALIGNED_ACCESS], [
   AC_MSG_CHECKING([if unaligned memory access works correctly])
   if test x"$as_cv_unaligned_access" = x ; then
-    case $host in
+    case $host_cpu in
       alpha*|arm*|hp*|mips*|sh*|sparc*|ia64*)
         _AS_ECHO_N([(blacklisted) ])
         as_cv_unaligned_access=no
