--- scripts/makefile.cygwin	2006-06-27 22:22:40.000000000 +0200
+++ scripts/makefile.cygwin	2006-07-12 13:18:15.000000000 +0200
@@ -35,7 +35,6 @@
 # $CFLAGS, and include pnggccrd.o in $OBJS, below, and in the dependency
 # list at the bottom of this makefile.
 
-CC=gcc
 ifdef MINGW
 MINGW_CCFLAGS=-mno-cygwin -I/usr/include/mingw
 MINGW_LDFLAGS=-mno-cygwin -L/usr/lib/mingw
@@ -43,14 +42,13 @@
 
 # Where "make install" puts libpng*.a, *png*.dll, png.h, and pngconf.h
 ifndef prefix
-prefix=/usr
-$(warning You haven't specified a 'prefix=' location. Defaulting to "/usr")
+prefix=??PREFIX??
 endif
 exec_prefix=$(prefix)
 
 # Where the zlib library and include files are located
-ZLIBLIB= /usr/lib
-ZLIBINC=
+ZLIBLIB= ??PREFIX??/lib
+ZLIBINC= ??PREFIX??/include
 #ZLIBLIB=../zlib
 #ZLIBINC=../zlib
 
@@ -95,7 +93,9 @@
 LDEXTRA=-Wl,--out-implib=$(IMPLIB) $(addprefix -L,$(ZLIBLIB)) -lz
 
 MKDIR_P=/bin/mkdir -pv
+ifndef RANLIB
 RANLIB=ranlib
+endif
 #RANLIB=echo
 
 INCPATH=$(prefix)/include
@@ -174,7 +174,7 @@
 	$(CC) -c $(CFLAGS) -DPNG_BUILD_DLL -o $@ $<
 
 $(STATLIB): $(OBJS)
-	ar rc $@ $(OBJS)
+	$(AR) rc $@ $(OBJS)
 	$(RANLIB) $@
 
 $(SHAREDDEF): scripts/pngw32.def
