diff -ruN libpng-1.2.12/scripts/makefile.darwin libpng/scripts/makefile.darwin
--- libpng-1.2.12/scripts/makefile.darwin	2006-06-27 22:22:40.000000000 +0200
+++ libpng/scripts/makefile.darwin	2006-07-27 12:07:18.000000000 +0200
@@ -7,14 +7,14 @@
 # For conditions of distribution and use, see copyright notice in png.h
 
 # where "make install" puts libpng.a, libpng12.dylib, png.h and pngconf.h
-prefix=/usr/local
+prefix=??PREFIX??
 exec_prefix=$(prefix)
 
 # Where the zlib library and include files are located
-#ZLIBLIB=/usr/local/lib
-#ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=/usr/lib
+ZLIBINC=/usr/include
+#ZLIBLIB=../zlib
+#ZLIBINC=../zlib
 
 # Library name:
 LIBNAME = libpng12
@@ -38,8 +38,8 @@
 RANLIB=ranlib
 RM_F=/bin/rm -f
 
-CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops
-LDFLAGS=-L. -L$(ZLIBLIB) -lpng12 -lz
+CFLAGS=??EXTRA_CFLAGS?? -I$(ZLIBINC) -Wall -O3 -funroll-loops
+LDFLAGS=??EXTRA_LDFLAGS?? -L. -L$(ZLIBLIB) -lpng12 -lz
 
 INCPATH=$(prefix)/include
 LIBPATH=$(exec_prefix)/lib
@@ -178,7 +178,7 @@
 	chmod 755 $(DB)/$(LIBNAME)-config
 	(cd $(DB); $(LN_SF) $(LIBNAME)-config libpng-config)
 
-install: install-static install-shared install-man install-config
+install: install-static install-man install-config
 
 # If you installed in $(DESTDIR), test-installed won't work until you
 # move the library to its final location.  Use test-dd to test it
