diff -urN libpng-1.2.41/scripts/makefile.darwin libpng/scripts/makefile.darwin
--- libpng-1.2.41/scripts/makefile.darwin       2009-12-03 04:46:02 +0300
+++ libpng/scripts/makefile.darwin       2009-12-04 14:37:52 +0300
@@ -76,7 +76,7 @@
 .c.pic.o:
        $(CC) -c $(CFLAGS) -fno-common -o $@ $*.c

-all: libpng.a $(LIBSO) pngtest libpng.pc libpng-config
+all: libpng.a $(LIBSO) libpng.pc libpng-config

 libpng.a: $(OBJS)
        $(AR_RC) $@ $(OBJS)
@@ -185,7 +185,7 @@
        chmod 755 $(DB)/$(LIBNAME)-config
        (cd $(DB); $(LN_SF) $(LIBNAME)-config libpng-config)

-install: install-static install-shared install-man install-config
+install: install-static install-man install-config

 # If you installed in $(DESTDIR), test-installed won't work until you
 # move the library to its final location.  Use test-dd to test it
