--- libpng/scripts/makefile.darwin	2006-07-28 16:53:49.000000000 +0200
+++ libpng2/scripts/makefile.darwin	2006-07-28 16:45:50.000000000 +0200
@@ -65,7 +65,7 @@
 	pngread.o pngrio.o pngwio.o pngwrite.o pngrtran.o \
 	pngwtran.o pngmem.o pngerror.o pngpread.o 
 
-OBJSDLL = $(OBJS:.o=.pic.o)
+OBJSDLL = $(OBJS:.o=.pic.o) pnggccrd.pic.o
 
 .SUFFIXES:      .c .o .pic.o
 
@@ -226,5 +226,6 @@
 pngwtran.o pngwtran.pic.o: png.h pngconf.h
 pngwutil.o pngwutil.pic.o: png.h pngconf.h
 pngpread.o pngpread.pic.o: png.h pngconf.h
+pnggccrd.o pnggccrd.pic.o: png.h pngconf.h
 
 pngtest.o: png.h pngconf.h
