--- libpng-1.2.8/scripts/makefile.linux	2004-12-03 01:14:31.000000000 +0100
+++ libpng/scripts/makefile.linux	2006-07-14 10:12:00.000000000 +0200
@@ -13,7 +13,7 @@
 # where "make install" puts libpng12.a, libpng12.so*,
 # libpng12/png.h and libpng12/pngconf.h
 # Prefix must be a full pathname.
-prefix=/usr/local
+prefix=$(PREFIX)
 exec_prefix=$(prefix)
 
 # Where the zlib library and include files are located.
@@ -53,7 +53,7 @@
 # If you're going to install into a temporary location
 # via DESTDIR, $(DESTDIR)$(prefix) must already exist before
 # you execute make install.
-DESTDIR=
+#DESTDIR=
 
 DB=$(DESTDIR)$(BINPATH)
 DI=$(DESTDIR)$(INCPATH)
