diff -ruN libpng-1.2.12/scripts/makefile.darwin libpng-1.2.12b/scripts/makefile.darwin
--- libpng/scripts/makefile.darwin	2006-06-27 22:22:40.000000000 +0200
+++ libpng/scripts/makefile.darwin	2006-07-27 09:57:03.000000000 +0200
@@ -63,7 +63,7 @@
 
 OBJS = png.o pngset.o pngget.o pngrutil.o pngtrans.o pngwutil.o \
 	pngread.o pngrio.o pngwio.o pngwrite.o pngrtran.o \
-	pngwtran.o pngmem.o pngerror.o pngpread.o
+	pngwtran.o pngmem.o pngerror.o pngpread.o pnggccrd.o
 
 OBJSDLL = $(OBJS:.o=.pic.o)
 
@@ -226,5 +226,6 @@
 pngwtran.o pngwtran.pic.o: png.h pngconf.h
 pngwutil.o pngwutil.pic.o: png.h pngconf.h
 pngpread.o pngpread.pic.o: png.h pngconf.h
+pnggccrd.o pnggccrd.pic.o: png.h pngconf.h
 
 pngtest.o: png.h pngconf.h
