Index: src/libprojectM/MilkdropPresetFactory/CMakeLists.txt
===================================================================
--- src/libprojectM/MilkdropPresetFactory/CMakeLists.txt	(revision 1287)
+++ src/libprojectM/MilkdropPresetFactory/CMakeLists.txt	(working copy)
@@ -3,7 +3,7 @@
 
 SET(MilkdropPresetFactory_SOURCES BuiltinFuncs.cpp Func.cpp MilkdropPreset.cpp Param.hpp PresetFrameIO.cpp CustomShape.cpp  Eval.cpp MilkdropPresetFactory.cpp PerPixelEqn.cpp BuiltinParams.cpp InitCond.cpp Parser.cpp CustomWave.cpp Expr.cpp PerPointEqn.cpp Param.cpp PerFrameEqn.cpp IdlePreset.cpp)
 
-SET (CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} -fPIC)
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 
 INCLUDE_DIRECTORIES(${projectM_SOURCE_DIR} ${Renderer_SOURCE_DIR})
 LINK_DIRECTORIES(${projectM_BINARY_DIR} ${Renderer_BINARY_DIR})
Index: src/libprojectM/NativePresetFactory/CMakeLists.txt
===================================================================
--- src/libprojectM/NativePresetFactory/CMakeLists.txt	(revision 1287)
+++ src/libprojectM/NativePresetFactory/CMakeLists.txt	(working copy)
@@ -2,7 +2,7 @@
 cmake_minimum_required(VERSION 2.4.0)
 
 SET(NativePresetFactory_SOURCES NativePresetFactory.cpp)
-SET (CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} -fPIC)
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
  
 INCLUDE_DIRECTORIES(${projectM_SOURCE_DIR} ${Renderer_SOURCE_DIR})
 ADD_LIBRARY(NativePresetFactory STATIC ${NativePresetFactory_SOURCES})
Index: src/libprojectM/Renderer/CMakeLists.txt
===================================================================
--- src/libprojectM/Renderer/CMakeLists.txt	(revision 1287)
+++ src/libprojectM/Renderer/CMakeLists.txt	(working copy)
@@ -7,8 +7,8 @@
 Filters.cpp PerlinNoise.cpp PipelineContext.cpp  Renderable.cpp BeatDetect.cpp Shader.cpp TextureManager.cpp VideoEcho.cpp 
 RenderItemDistanceMetric.cpp RenderItemMatcher.cpp ${SOIL_SOURCES})
 
-SET (CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} -fPIC)
-SET (CMAKE_C_FLAGS ${CMAKE_C_FLAGS} -fPIC)
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 
 INCLUDE_DIRECTORIES(${projectM_SOURCE_DIR})
 ADD_LIBRARY(Renderer STATIC ${Renderer_SOURCES})
