diff -Nur live.orig/config.armlinux.rej live/config.armlinux.rej
--- live.orig/config.armlinux.rej	1970-01-01 01:00:00.000000000 +0100
+++ live/config.armlinux.rej	2007-09-17 01:49:49.000000000 +0200
@@ -0,0 +1,13 @@
+***************
+*** 1,5 ****
+  CROSS_COMPILE=		arm-elf-
+- COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -DNO_STRSTREAM=1 -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+  C =			c
+  C_COMPILER =		$(CROSS_COMPILE)gcc
+  C_FLAGS =		$(COMPILE_OPTS)
+--- 1,5 ----
+  CROSS_COMPILE=		arm-elf-
++ COMPILE_OPTS =          $(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -DNO_STRSTREAM=1 -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DUSE_SYSTEM_RANDOM=1
+  C =			c
+  C_COMPILER =		$(CROSS_COMPILE)gcc
+  C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.cygwin live/config.cygwin
--- live.orig/config.cygwin	2007-08-03 18:38:58.000000000 +0200
+++ live/config.cygwin	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOCKLEN_T=socklen_t
+COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOCKLEN_T=socklen_t -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		gcc
 C_FLAGS =		$(COMPILE_OPTS) -DUSE_OUR_BZERO=1 -D__CYGWIN__
diff -Nur live.orig/config.freebsd live/config.freebsd
--- live.orig/config.freebsd	2007-08-03 18:38:58.000000000 +0200
+++ live/config.freebsd	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=socklen_t -DHAVE_SOCKADDR_LEN=1
+COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=socklen_t -DHAVE_SOCKADDR_LEN=1 -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.linux live/config.linux
--- live.orig/config.linux	2007-08-03 18:38:58.000000000 +0200
+++ live/config.linux	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.linux-gdb live/config.linux-gdb
--- live.orig/config.linux-gdb	2007-08-03 18:38:58.000000000 +0200
+++ live/config.linux-gdb	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOCKLEN_T=socklen_t -g -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOCKLEN_T=socklen_t -g -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.macosx-before-version-10.4 live/config.macosx-before-version-10.4
--- live.orig/config.macosx-before-version-10.4	2007-08-03 18:38:58.000000000 +0200
+++ live/config.macosx-before-version-10.4	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=int
+COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=int -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.mingw live/config.mingw
--- live.orig/config.mingw	2007-08-03 18:38:58.000000000 +0200
+++ live/config.mingw	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =         $(INCLUDES) -I. -O -DSOCKLEN_T=int
+COMPILE_OPTS =         $(INCLUDES) -I. -O -DSOCKLEN_T=int -DUSE_SYSTEM_RANDOM=1
 C =                    c
 C_COMPILER =           gcc
 C_FLAGS =              $(COMPILE_OPTS) -DUSE_OUR_BZERO=1 -D__MINGW32__
diff -Nur live.orig/config.openbsd live/config.openbsd
--- live.orig/config.openbsd	2007-08-03 18:38:58.000000000 +0200
+++ live/config.openbsd	2007-09-17 01:49:49.000000000 +0200
@@ -1,5 +1,5 @@
 .SUFFIXES: .cpp
-COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=socklen_t
+COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=socklen_t -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.solaris live/config.solaris
--- live.orig/config.solaris	2007-08-03 18:38:58.000000000 +0200
+++ live/config.solaris	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOLARIS -DSOCKLEN_T=socklen_t
+COMPILE_OPTS =		$(INCLUDES) -I. -O -DSOLARIS -DSOCKLEN_T=socklen_t -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -Nur live.orig/config.uClinux live/config.uClinux
--- live.orig/config.uClinux	2007-08-03 18:38:58.000000000 +0200
+++ live/config.uClinux	2007-09-17 01:49:49.000000000 +0200
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DUSE_SYSTEM_RANDOM=1
 C =			c
 C_COMPILER =		$(CC)
 CFLAGS +=		$(COMPILE_OPTS)
