diff -ru live.orig/config.mingw live/config.mingw
--- live.orig/config.mingw	2005-10-23 08:50:31.000000000 +0200
+++ live/config.mingw	2005-10-23 17:09:35.000000000 +0200
@@ -1,15 +1,15 @@
 COMPILE_OPTS =         $(INCLUDES) -I. -O -DSOCKLEN_T=int
 C =                    c
-C_COMPILER =           gcc
-C_FLAGS =              $(COMPILE_OPTS) -DUSE_OUR_BZERO=1 -D__MINGW32__
+C_COMPILER =           $(CC)
+C_FLAGS =              $(COMPILE_OPTS) -DUSE_OUR_BZERO=1 -D__MINGW32__ -DUSE_LOCALE
 CPP =                  cpp
-CPLUSPLUS_COMPILER =   c++
-CPLUSPLUS_FLAGS =      $(COMPILE_OPTS) -D__MINGW32__ -Wall -Wno-deprecated
+CPLUSPLUS_COMPILER =   $(CXX)
+CPLUSPLUS_FLAGS =      $(COMPILE_OPTS) -D__MINGW32__ -DUSE_LOCALE -Wall -Wno-deprecated 
 OBJ =                  o
-LINK =                 c++ -o 
+LINK =                 $(CXX) -o 
 LINK_OPTS =            -L.
 CONSOLE_LINK_OPTS =    $(LINK_OPTS)
-LIBRARY_LINK =         ld -o 
+LIBRARY_LINK =         $(LD) -o 
 LIBRARY_LINK_OPTS =    $(LINK_OPTS) -r -Bstatic
 LIB_SUFFIX =                   a
 LIBS_FOR_CONSOLE_APPLICATION = -lws2_32
