Copyright (C) 2009 Rémi Denis-Courmont. Licensed under GPLv2 or higher.
diff -Nru live.orig/liveMedia/RTSPClient.cpp live/liveMedia/RTSPClient.cpp
--- live.orig/liveMedia/RTSPClient.cpp	2009-04-05 20:38:11.000000000 +0300
+++ live/liveMedia/RTSPClient.cpp	2009-04-05 20:39:43.000000000 +0300
@@ -1835,7 +1835,7 @@
       }
       if (connect(fInputSocketNum, (struct sockaddr*) &remoteName, sizeof remoteName) != 0) {
 #if defined(__WIN32__) || defined(_WIN32)
-        if (errno != WSAEINPROGRESS) {
+        if (errno != WSAEWOULDBLOCK) {
 #else
         if (errno != EINPROGRESS) {
 #endif
