diff -ruN live.o/Makefile.tail live/Makefile.tail
--- live.o/Makefile.tail	2007-02-20 01:15:46.000000000 +0100
+++ live/Makefile.tail	2007-02-20 01:16:36.000000000 +0100
@@ -14,9 +14,7 @@
 
 ALL =	$(LIVEMEDIA_LIB) \
 	$(GROUPSOCK_LIB) \
-	$(USAGE_ENVIRONMENT_LIB) \
-	$(BASIC_USAGE_ENVIRONMENT_LIB) \
-	$(TESTPROGS_APP)
+	$(BASIC_USAGE_ENVIRONMENT_LIB)
 all:	$(ALL)
 
 
diff -ruN live.o/config.macosx live/config.macosx
--- live.o/config.macosx	2007-02-20 01:15:46.000000000 +0100
+++ live/config.macosx	2007-02-20 01:17:02.000000000 +0100
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -DBSD=1 -O -DSOCKLEN_T=socklen_t -DHAVE_SOCKADDR_LEN=1
+COMPILE_OPTS =		$(INCLUDES) -I. $(EXTRA_LDFLAGS) -DBSD=1 -O -DSOCKLEN_T=socklen_t -DHAVE_SOCKADDR_LEN=1
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
