diff -ru libmatroska-original/make/linux/Makefile libmatroska/make/linux/Makefile
--- libmatroska-original/make/linux/Makefile	Mon Jan  5 12:28:43 2004
+++ libmatroska/make/linux/Makefile	Mon Aug 16 11:56:22 2004
@@ -93,7 +93,7 @@
 depend: $(sources)
 	$(DEPEND) $(DEPENDFLAGS) $(sources)
 
-test: test6 test8 test9
+test: test6 test9
 
 test6:	test6.o $(LIBRARY)
 	$(LD) -o $@ $(LDFLAGS) $< $(MUX_LIBS)
