Index: mpcdec/configure.ac
===================================================================
--- mpcdec/configure.ac (revision 276)
+++ mpcdec/configure.ac (revision 289)
@@ -59,11 +59,6 @@
 fi

-AC_FUNC_MEMCMP
-if test "x$ac_cv_func_memcmp_working" = "xno" ; then
-	AC_MSG_ERROR([working memcmp is not available.])
-fi
-
-AC_CHECK_FUNCS([memmove memset], [],
-        AC_MSG_ERROR([memset or memmove is missing.]))
+AC_CHECK_FUNCS([memmove memset memcmp], [],
+        AC_MSG_ERROR([memmove memset or memcmp is missing.]))
 
  dnl Make substitutions
