--- mpcdec/configure.ac	2007-04-17 21:05:32.000000000 +0200
+++ mpcdec/configure.ac	2007-09-06 23:52:09.000000000 +0200
@@ -8,6 +8,9 @@
 
 CFLAGS="$CFLAGS -O3 -fomit-frame-pointer -fPIC"
 
+AC_PROG_CC
+AC_PROG_CXX
+
 AC_C_BIGENDIAN(,CFLAGS="$CFLAGS -DMPC_LITTLE_ENDIAN",)
 
 AC_HEADER_STDC
@@ -58,10 +58,15 @@
 	ac_cv_c_int64_t=int64_t
 fi
 
+if test "x$cross_compiling" = "xyes" ; then
+AC_CHECK_FUNCS([memcmp], [],
+        AC_MSG_ERROR([memcmp is missing.]))
+else
 AC_FUNC_MEMCMP
 if test "x$ac_cv_func_memcmp_working" = "xno" ; then
 	AC_MSG_ERROR([working memcmp is not available.])
 fi
+fi
 
 AC_CHECK_FUNCS([memmove memset], [],
         AC_MSG_ERROR([memset or memmove is missing.]))
