diff -ruN mpeg2dec.orig/libmpeg2/cpu_accel.c mpeg2dec/libmpeg2/cpu_accel.c
--- mpeg2dec.orig/libmpeg2/cpu_accel.c	2004-04-10 07:51:44.000000000 +0200
+++ mpeg2dec/libmpeg2/cpu_accel.c	2005-04-08 23:45:11.000000000 +0200
@@ -145,7 +145,7 @@
 #endif /* ACCEL_DETECT && (ARCH_PPC || ARCH_SPARC) */
 
 #ifdef ARCH_PPC
-static inline uint32_t arch_accel (uint32_t accel)
+static uint32_t arch_accel (uint32_t accel)
 {
 #ifdef ACCEL_DETECT
     if ((accel & (MPEG2_ACCEL_PPC_ALTIVEC | MPEG2_ACCEL_DETECT)) ==
