--- configure.in.orig	2006-11-05 19:40:10.000000000 +0000
+++ configure.in	2006-11-05 19:42:10.000000000 +0000
@@ -62,10 +62,10 @@
 
 AC_MSG_CHECKING([for Win32])
 case "$host" in
-  *-*-mingw*)
-    native_win32=yes
-    AC_DEFINE(USE_INSTALLED_GLIB, 1, [We are using an installed GLib])
-  ;;
+dnl  *-*-mingw*)
+dnl    native_win32=yes
+dnl    AC_DEFINE(USE_INSTALLED_GLIB, 1, [We are using an installed GLib])
+dnl  ;;
   *)
     native_win32=no
   ;;
