diff -uN portaudio/Makefile.in portaudio.new/Makefile.in
--- portaudio/Makefile.in	2008-01-01 23:02:42.000000000 +0100
+++ portaudio.new/Makefile.in	2008-03-07 05:22:21.000000000 +0100
@@ -120,7 +120,7 @@
 SUBDIRS =
 @ENABLE_CXX_TRUE@SUBDIRS += bindings/cpp
 
-all: lib/$(PALIB) all-recursive tests
+all: lib/$(PALIB) all-recursive
 
 tests: bin-stamp $(TESTS)
 
diff -uN portaudio/configure.in portaudio.new/configure.in
--- portaudio/configure.in	2008-01-27 03:03:30.000000000 +0100
+++ portaudio.new/configure.in	2008-03-07 05:38:44.000000000 +0100
@@ -99,9 +99,10 @@
 
 dnl sizeof checks: we will need a 16-bit and a 32-bit type
 
-AC_CHECK_SIZEOF(short)
-AC_CHECK_SIZEOF(int)
-AC_CHECK_SIZEOF(long)
+dnl don't work if you crosscompile 
+dnl AC_CHECK_SIZEOF(short)
+dnl AC_CHECK_SIZEOF(int)
+dnl AC_CHECK_SIZEOF(long)
 
 save_LIBS="${LIBS}"
 AC_CHECK_LIB(rt, clock_gettime, [rt_libs=" -lrt"])
@@ -235,7 +236,7 @@
             #CFLAGS="$CFLAGS -I$VC98 -DPA_NO_WMME -DPA_NO_ASIO";
             CFLAGS="$CFLAGS -I\$(top_srcdir)/src/common -I$DXDIR/include -DPA_NO_WMME -DPA_NO_DS -DPA_NO_ASIO";
         else   # WMME default
-            OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
+            OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_waveformat.o src/os/win/pa_win_waveformat.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
             LIBS="-lwinmm -lm -lstdc++ -lole32 -luuid";
             PADLL="portaudio.dll";
 	    THREAD_CFLAGS="-mthreads"
@@ -248,7 +249,7 @@
   cygwin* )
 	dnl Cygwin configuration
 
-	OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
+	OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_waveformat.o src/os/win/pa_win_waveformat.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
 	CFLAGS="$CFLAGS -DPA_NO_DS -DPA_NO_WDMKS -DPA_NO_ASIO -DPA_NO_WASAPI"
 	LIBS="-lwinmm -lm";
 	PADLL="portaudio.dll";
