diff -ru speex-original/libspeex/preprocess.c speex/libspeex/preprocess.c
--- speex-original/libspeex/preprocess.c	Sat Apr 10 00:21:49 2004
+++ speex/libspeex/preprocess.c	Fri Jul 23 10:11:10 2004
@@ -970,8 +970,8 @@
 int speex_preprocess_ctl(SpeexPreprocessState *state, int request, void *ptr)
 {
    SpeexPreprocessState *st;
-   st=(SpeexPreprocessState*)state;
    int i;
+   st=(SpeexPreprocessState*)state;
    switch(request)
    {
    case SPEEX_PREPROCESS_SET_DENOISE:
