diff -ruN speex-1.1.12/Makefile.am speex/Makefile.am
--- speex-1.1.12/Makefile.am	2005-11-23 12:42:56.000000000 +0100
+++ speex/Makefile.am	2006-07-26 15:38:04.000000000 +0200
@@ -13,9 +13,9 @@
 EXTRA_DIST = Speex.spec Speex.spec.in Speex.kdevelop speex.m4 speex.pc.in README.blackfin README.symbian README.TI-DSP
 
 #Fools KDevelop into including all files
-SUBDIRS = libspeex include @src@ doc win32 symbian ti
+SUBDIRS = libspeex include doc win32 symbian ti
 
-DIST_SUBDIRS = libspeex include src doc win32 symbian ti
+DIST_SUBDIRS = libspeex include doc win32 symbian ti
 
 rpm: dist
 	rpmbuild -ta ${PACKAGE}-${VERSION}.tar.gz
diff -ruN speex-1.1.12/Makefile.in speex/Makefile.in
--- speex-1.1.12/Makefile.in	2006-02-10 11:13:16.000000000 +0100
+++ speex/Makefile.in	2006-07-26 15:38:04.000000000 +0200
@@ -194,8 +194,8 @@
 EXTRA_DIST = Speex.spec Speex.spec.in Speex.kdevelop speex.m4 speex.pc.in README.blackfin README.symbian README.TI-DSP
 
 #Fools KDevelop into including all files
-SUBDIRS = libspeex include @src@ doc win32 symbian ti
-DIST_SUBDIRS = libspeex include src doc win32 symbian ti
+SUBDIRS = libspeex include doc win32 symbian ti
+DIST_SUBDIRS = libspeex include doc win32 symbian ti
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
