Index: os.h
===================================================================
--- os.h	(revision 10420)
+++ os.h	(working copy)
@@ -20,6 +20,14 @@
 #include <math.h>
 #include "os_types.h"
 
+#ifdef _LOW_ACCURACY_
+#  define X(n) (((((n)>>22)+1)>>1) - ((((n)>>22)+1)>>9))
+#  define LOOKUP_T const unsigned char
+#else
+#  define X(n) (n)
+#  define LOOKUP_T const ogg_int32_t
+#endif
+
 #ifndef _V_IFDEFJAIL_H_
 #  define _V_IFDEFJAIL_H_
 
Index: synthesis.c
===================================================================
--- synthesis.c	(revision 10420)
+++ synthesis.c	(working copy)
@@ -33,7 +33,7 @@
  
   /* first things first.  Make sure decode is ready */
   _vorbis_block_ripcord(vb);
-  oggpack_readinit(opb,op->packet);
+  oggpack_readinit(opb,op->packet,op->bytes);
 
   /* Check the packet type */
   if(oggpack_read(opb,1)!=0){
@@ -86,7 +86,7 @@
   oggpack_buffer       opb;
   int                  mode;
  
-  oggpack_readinit(&opb,op->packet);
+  oggpack_readinit(&opb,op->packet,op->bytes);
 
   /* Check the packet type */
   if(oggpack_read(&opb,1)!=0){
Index: info.c
===================================================================
--- info.c	(revision 10420)
+++ info.c	(working copy)
@@ -299,7 +299,7 @@
   oggpack_buffer opb;
   
   if(op){
-    oggpack_readinit(&opb,op->packet);
+    oggpack_readinit(&opb,op->packet,op->bytes);
 
     /* Which of the three types of header is this? */
     /* Also verify header-ness, vorbis */
Index: ivorbiscodec.h
===================================================================
--- ivorbiscodec.h	(revision 10420)
+++ ivorbiscodec.h	(working copy)
@@ -23,7 +23,7 @@
 {
 #endif /* __cplusplus */
 
-#include "ogg.h"
+#include <ogg/ogg.h>
 
 typedef struct vorbis_info{
   int version;
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 10420)
+++ Makefile.am	(working copy)
@@ -8,11 +8,11 @@
                         synthesis.c info.c \
                         floor1.c floor0.c vorbisfile.c \
                         res012.c mapping0.c registry.c codebook.c \
-			sharedbook.c framing.c bitwise.c \
+			sharedbook.c \
                         codebook.h misc.h mdct_lookup.h\
                         os.h mdct.h ivorbisfile.h lsp_lookup.h\
                         registry.h window.h window_lookup.h\
-                        codec_internal.h backends.h ogg.h \
+                        codec_internal.h backends.h \
 			asm_arm.h ivorbiscodec.h
 libvorbisidec_la_LDFLAGS = -version-info @V_LIB_CURRENT@:@V_LIB_REVISION@:@V_LIB_AGE@
 
@@ -25,7 +25,7 @@
 
 includedir = $(prefix)/include/tremor
 
-include_HEADERS = ivorbiscodec.h ivorbisfile.h ogg.h os_types.h config_types.h
+include_HEADERS = ivorbiscodec.h ivorbisfile.h os_types.h config_types.h
 
 example:
 	-ln -fs . vorbis
Index: configure.in
===================================================================
--- orig/configure.in	2006-02-01 12:56:54.000000000 +0100
+++ configure.in	2006-02-01 13:02:36.000000000 +0100
@@ -54,8 +54,7 @@ else
 
         case $host in 
         arm-*-*)
-                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char 
--D_ARM_ASSEM_"
+                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -D_ARM_ASSEM_"
                 CFLAGS="-O2 -D_ARM_ASSEM_ -fsigned-char"
                 PROFILE="-W -pg -g -O2 -D_ARM_ASSEM_ -fsigned-char -fno-inline-functions";;
 
