--- libvorbis/Makefile.in	2010-03-26 08:00:09.000000000 +0100
+++ libvorbis.new/Makefile.in	2010-03-29 21:39:50.000000000 +0200
@@ -80,7 +80,7 @@
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = m4 include vq lib test doc examples
+DIST_SUBDIRS = m4 include vq lib 
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
