--- configure	2008-04-06 09:07:25.000000000 +0200
+++ configure	2008-04-06 09:10:52.000000000 +0200
@@ -343,6 +343,7 @@
 # autodetect options that weren't forced nor disabled
 
 libpthread=""
+extradef=""
 if test "$pthread" = "auto" ; then
     pthread="no"
     case $SYS in
@@ -350,13 +351,13 @@
             pthread="yes"
             ;;
         MINGW)
-            if cc_check pthread.h -lpthread "pthread_create(0,0,0,0);" ; then
+            if cc_check pthread.h -lpthread "pthread_create(0,0,0,0); pthread_num_processors_np();" ; then
                 pthread="yes"
                 libpthread="-lpthread"
-            elif cc_check pthread.h -lpthreadGC2 "pthread_create(0,0,0,0);" ; then
+            elif cc_check pthread.h -lpthreadGC2 "pthread_create(0,0,0,0); pthread_num_processors_np(); " ; then
                 pthread="yes"
                 libpthread="-lpthreadGC2"
-            elif cc_check pthread.h "-lpthreadGC2 -lwsock32 -DPTW32_STATIC_LIB" "pthread_create(0,0,0,0);" ; then
+            elif cc_check pthread.h "-lpthreadGC2 -lwsock32 -DPTW32_STATIC_LIB" "pthread_create(0,0,0,0); pthread_num_processors_np();" ; then
                 pthread="yes"
                 libpthread="-lpthreadGC2 -lwsock32"
                 CFLAGS="$CFLAGS -DPTW32_STATIC_LIB"
@@ -443,6 +444,10 @@
 ASFLAGS=$ASFLAGS
 GTK=$gtk
 EXE=$EXE
+AR=$AR
+STRIP=$STRIP
+AROPTS=$AROPTS
+RANLIB=$RANLIB
 VIS=$vis
 HAVE_GETOPT_LONG=$HAVE_GETOPT_LONG
 DEVNULL=$DEVNULL
@@ -495,7 +500,7 @@
 Description: H.264 (MPEG4 AVC) encoder library
 Version: $(grep POINTVER < config.h | sed -e 's/.* "//; s/".*//')
 Libs: $pclibs
-Cflags: -I$includedir
+Cflags: -I$includedir $extradef
 EOF
 
 
