Index: configure
===================================================================
--- configure	(revision 715)
+++ configure	(working copy)
@@ -396,7 +396,7 @@
     CFLAGS="$CFLAGS -fPIC"
     ASFLAGS="$ASFLAGS -D__PIC__"
     # resolve textrels in the x86 asm
-    cc_check stdio.h -Wl,-Bsymbolic && LDFLAGS="$LDFLAGS -Wl,-Bsymbolic"
+    cc_check stdio.h -Wl && LDFLAGS="$LDFLAGS -Wl"
 fi
 
 if [ "$debug" != "yes" -a "$gprof" != "yes" ]; then

Index: common/common.h
===================================================================
--- common/common.h	(revision 701)
+++ common/common.h	(working copy)
@@ -67,7 +67,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <assert.h>
-#include "x264.h"
+#include "../x264.h"
 #include "bs.h"
 #include "set.h"
 #include "predict.h"
