diff -ruN x264.orig/Makefile x264/Makefile
--- x264.orig/Makefile	Mon May 30 21:47:02 2005
+++ x264/Makefile	Sun Jun 26 03:09:05 2005
@@ -40,7 +40,7 @@
 OBJS = $(SRCS:%.c=%.o)
 DEP  = depend
 
-default: $(DEP) x264$(EXE)
+default: $(DEP) libx264.a
 
 libx264.a: .depend $(OBJS) $(OBJASM)
 	ar rc libx264.a $(OBJS) $(OBJASM)
@@ -61,7 +61,7 @@
 .depend: config.mak config.h
 	rm -f .depend
 # Hacky - because gcc 2.9x doesn't have -MT
-	$(foreach SRC, $(SRCS) x264.c, ( echo -n "`dirname $(SRC)`/" && $(CC) $(CFLAGS) $(SRC) -MM -g0 ) 1>> .depend;)
+	$(foreach SRC, $(SRCS), ( echo -n "`dirname $(SRC)`/" && $(CC) $(CFLAGS) $(SRC) -MM -g0 ) 1>> .depend;)
 
 config.h: $(wildcard .svn/entries */.svn/entries */*/.svn/entries)
 	./version.sh
@@ -78,11 +78,10 @@
 distclean: clean
 	rm -f config.mak vfw/build/cygwin/config.mak
 
-install: x264
+install: libx264.a
 	install -d $(DESTDIR)$(bindir) $(DESTDIR)$(libdir) $(DESTDIR)$(includedir)
 	install -m 644 x264.h $(DESTDIR)$(includedir)
 	install -m 644 libx264.a $(DESTDIR)$(libdir)
-	install x264 $(DESTDIR)$(bindir)
 
 etags: TAGS
 
