Index: configure
===================================================================
--- configure	(revision 715)
+++ configure	(working copy)
@@ -396,7 +396,7 @@
     CFLAGS="$CFLAGS -fPIC"
     ASFLAGS="$ASFLAGS -D__PIC__"
     # resolve textrels in the x86 asm
-    cc_check stdio.h -Wl,-Bsymbolic && LDFLAGS="$LDFLAGS -Wl,-Bsymbolic"
+    cc_check stdio.h -Wl && LDFLAGS="$LDFLAGS -Wl"
 fi
 
 if [ "$debug" != "yes" -a "$gprof" != "yes" ]; then

