Index: common/i386/dct-a.asm
===================================================================
--- common/i386/dct-a.asm	(revision 270)
+++ common/i386/dct-a.asm	(working copy)
@@ -125,27 +125,6 @@
 ;%endmacro
 
 ;=============================================================================
-; Local Data (Read Only)
-;=============================================================================
-
-%ifdef FORMAT_COFF
-SECTION .rodata data
-%else
-SECTION .rodata data align=16
-%endif
-
-;-----------------------------------------------------------------------------
-; Various memory constants (trigonometric values or rounding values)
-;-----------------------------------------------------------------------------
-
-ALIGN 16
-x264_mmx_1:
-  dw 1, 1, 1, 1
-
-x264_mmx_32:
-  dw 32, 32, 32, 32
-
-;=============================================================================
 ; Code
 ;=============================================================================
 
@@ -311,3 +290,23 @@
 
     ret
 
+;=============================================================================
+; Local Data (Read Only)
+;=============================================================================
+
+%ifdef FORMAT_COFF
+SECTION .rodata data
+%else
+SECTION .rodata data align=16
+%endif
+
+;-----------------------------------------------------------------------------
+; Various memory constants (trigonometric values or rounding values)
+;-----------------------------------------------------------------------------
+
+ALIGN 16
+x264_mmx_1:
+  dw 1, 1, 1, 1
+
+x264_mmx_32:
+  dw 32, 32, 32, 32
Index: common/i386/mc-a.asm
===================================================================
--- common/i386/mc-a.asm	(revision 270)
+++ common/i386/mc-a.asm	(working copy)
@@ -46,22 +46,6 @@
 %endmacro
 
 ;=============================================================================
-; Local Data (Read Only)
-;=============================================================================
-
-%ifdef FORMAT_COFF
-SECTION .rodata data
-%else
-SECTION .rodata data align=16
-%endif
-
-;-----------------------------------------------------------------------------
-; Various memory constants (trigonometric values or rounding values)
-;-----------------------------------------------------------------------------
-
-ALIGN 16
-
-;=============================================================================
 ; Code
 ;=============================================================================
 
@@ -487,3 +471,20 @@
 .finish
     pop     edi
     ret
+
+;=============================================================================
+; Local Data (Read Only)
+;=============================================================================
+
+%ifdef FORMAT_COFF
+SECTION .rodata data
+%else
+SECTION .rodata data align=16
+%endif
+
+;-----------------------------------------------------------------------------
+; Various memory constants (trigonometric values or rounding values)
+;-----------------------------------------------------------------------------
+
+ALIGN 16
+
Index: common/i386/mc-a2.asm
===================================================================
--- common/i386/mc-a2.asm	(revision 270)
+++ common/i386/mc-a2.asm	(working copy)
@@ -33,22 +33,6 @@
     %endif
 %endmacro
 
-;=============================================================================
-; Read only data
-;=============================================================================
-
-SECTION .rodata data align=16
-
-ALIGN 16
-mmx_dw_one:
-    times 4 dw 16
-mmx_dd_one:
-    times 2 dd 512
-mmx_dw_20:
-    times 4 dw 20
-mmx_dw_5:
-    times 4 dw -5
-
 %assign twidth  0
 %assign theight 4
 %assign tdstp1  8
@@ -60,7 +44,6 @@
 %assign toffset 32
 %assign tbuffer 36
 
-
 ;=============================================================================
 ; Macros
 ;=============================================================================
@@ -375,3 +358,23 @@
     pop         edi
 
     ret
+
+;=============================================================================
+; Read only data
+;=============================================================================
+
+%ifdef FORMAT_COFF
+SECTION .rodata data
+%else
+SECTION .rodata data align=16
+%endif
+
+ALIGN 16
+mmx_dw_one:
+    times 4 dw 16
+mmx_dd_one:
+    times 2 dd 512
+mmx_dw_20:
+    times 4 dw 20
+mmx_dw_5:
+    times 4 dw -5
Index: build/cygwin/Makefile
===================================================================
--- build/cygwin/Makefile	(revision 270)
+++ build/cygwin/Makefile	(working copy)
@@ -7,6 +7,8 @@
 # $Id: Makefile,v 1.4 2004/06/18 02:00:40 chenm001 Exp $
 ##############################################################################
 
+CC=i586-mingw32msvc-gcc
+
 # Current dir
 DIR_CUR=$(shell pwd)
 
@@ -18,7 +20,7 @@
 # Sources
 SRC_C= common/mc.c common/predict.c common/pixel.c common/macroblock.c \
        common/frame.c common/dct.c common/cpu.c common/cabac.c \
-       common/common.c common/mdate.c common/csp.c \
+       common/common.c common/mdate.c common/csp.c common/set.c \
        encoder/analyse.c encoder/me.c encoder/ratecontrol.c \
        encoder/set.c encoder/macroblock.c encoder/cabac.c encoder/cavlc.c \
        encoder/encoder.c encoder/eval.c \
@@ -33,7 +35,7 @@
 ##############################################################################
 # PFLAGS
 ##############################################################################
-PFLAGS=-DARCH_X86 -DHAVE_MMXEXT -UHAVE_SSE2
+PFLAGS=-DHAVE_PTHREAD -DARCH_X86 -DHAVE_MMXEXT -UHAVE_SSE2
 
 ##############################################################################
 # CFLAGS
@@ -107,7 +109,7 @@
 libx264.a: $(DIR_BUILD) $(OBJECTS)
 	@echo " L: $(@F)"
 	@cd $(DIR_BUILD) && \
-	ar rc libx264.a $(OBJECTS) && \
+	i586-mingw32msvc-ar rc libx264.a $(OBJECTS) && \
 	cp -f libx264.a $(DIR_SRC)
 
 x264.exe: $(DIR_BUILD) $(OBJECTS) x264.obj
