Index: x264/Makefile
===================================================================
--- x264/Makefile	(revision 381)
+++ x264/Makefile	(working copy)
@@ -60,7 +60,7 @@
 DEP  = depend
 
 .PHONY: default fprofiled clean distclean install uninstall
-default: $(DEP) x264$(EXE)
+default: $(DEP) libx264.a
 
 libx264.a: .depend $(OBJS) $(OBJASM)
 	ar rc libx264.a $(OBJS) $(OBJASM)
@@ -128,13 +128,12 @@
 distclean: clean
 	rm -f config.mak vfw/build/cygwin/config.mak x264.pc
 
-install: x264
+install: libx264.a
 	install -d $(DESTDIR)$(bindir) $(DESTDIR)$(includedir)
 	install -d $(DESTDIR)$(libdir) $(DESTDIR)$(libdir)/pkgconfig
 	install -m 644 x264.h $(DESTDIR)$(includedir)
 	install -m 644 libx264.a $(DESTDIR)$(libdir)
 	install -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
-	install x264 $(DESTDIR)$(bindir)
 	ranlib $(DESTDIR)$(libdir)/libx264.a
 
 uninstall:
