--- zlib/gzio.c	2005-07-11 21:31:48.000000000 +0100
+++ zlib/gzio.c	2005-11-22 22:26:26.000000000 +0000
@@ -35,6 +35,10 @@
    FILE *fdopen(int, const char *);
 #endif
 
+#ifdef UNDER_CE
+#   define fdopen(fd, fmode) 0
+#endif
+
 #ifndef STDC
 extern voidp  malloc OF((uInt size));
 extern void   free   OF((voidpf ptr));
===================================================================
--- zlib/zutil.h	2005-11-27 20:05:31.000000000 +0000
+++ zlib/zutil.h	2005-11-27 20:06:16.000000000 +0000
@@ -36,6 +36,9 @@
 #else
 #  ifndef _WIN32_WCE
 #    include <errno.h>
+#  else
+#    define errno z_errno
+     extern int errno;
 #  endif
 #endif
 
