--- game-music-emu/CMakeLists.txt       2013-03-17 03:13:46.000000000 +0100
+++ game-music-emu.patched/CMakeLists.txt       2013-09-07 13:53:36.000000000 +0200
@@ -67,14 +67,14 @@
       string (REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")

       # gcc <4.1 had poor support for symbol visibility
-      if ((${_gcc_version} VERSION_GREATER "4.1") OR (${_gcc_version} VERSION_EQUAL "4.1"))
+      if (("${_gcc_version}" VERSION_GREATER "4.1") OR ("${_gcc_version}" VERSION_EQUAL "4.1"))
          set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
          set (ENABLE_VISIBILITY ON)
          add_definitions (-DLIBGME_VISIBILITY)

          # GCC >= 4.2 also correctly supports making inline members have hidden
          # visibility by default.
-         if ((${_gcc_version} VERSION_GREATER "4.2") OR (${_gcc_version} VERSION_EQUAL "4.2"))
+         if (("${_gcc_version}" VERSION_GREATER "4.2") OR ("${_gcc_version}" VERSION_EQUAL "4.2"))
             set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility-inlines-hidden")
          endif()
       endif()
