--- game-music-emu-0.6.0/gme/CMakeLists.txt.orig	2012-05-27 22:48:53.000000000 +0200
+++ game-music-emu-0.6.0/gme/CMakeLists.txt	2013-07-04 19:48:23.031372405 +0200
@@ -143,7 +143,7 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 # Add library to be compiled.
-add_library(gme SHARED ${libgme_SRCS})
+add_library(gme STATIC ${libgme_SRCS})
 
 # The version is the release.  The "soversion" is the API version.  As long
 # as only build fixes are performed (i.e. no backwards-incompatible changes
