--- libvpx/vpx/vpx_encoder.h.orig	2016-09-29 13:16:43.275060163 +0200
+++ libvpx/vpx/vpx_encoder.h	2016-09-29 13:17:25.606960648 +0200
@@ -83,10 +83,6 @@
    */
 #define VPX_CODEC_CAP_OUTPUT_PARTITION  0x20000
 
-/*! Can support input images at greater than 8 bitdepth.
- */
-#define VPX_CODEC_CAP_HIGHBITDEPTH  0x40000
-
   /*! \brief Initialization-time Feature Enabling
    *
    *  Certain codec features must be known at initialization time, to allow

--- libvpx/vpx/vpx_codec.h.orig	2016-09-29 13:17:21.970969167 +0200
+++ libvpx/vpx/vpx_codec.h	2016-09-29 13:17:57.098887090 +0200
@@ -154,6 +154,9 @@
 #define VPX_CODEC_CAP_DECODER 0x1 /**< Is a decoder */
 #define VPX_CODEC_CAP_ENCODER 0x2 /**< Is an encoder */
 
+/*! Can support input images at greater than 8 bitdepth.
+ */
+#define VPX_CODEC_CAP_HIGHBITDEPTH  0x40000
 
   /*! \brief Initialization-time Feature Enabling
    *
