diff -ru libvpx.orig/configure libvpx/configure
--- libvpx.orig/configure	2011-07-04 19:54:18.000000000 +0300
+++ libvpx/configure	2011-07-04 19:56:19.000000000 +0300
@@ -117,6 +117,7 @@
 all_platforms="${all_platforms} x86_64-linux-gcc"
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
+all_platforms="${all_platforms} x86_64-win64-gcc"
 all_platforms="${all_platforms} x86_64-win64-vs8"
 all_platforms="${all_platforms} x86_64-win64-vs9"
 all_platforms="${all_platforms} universal-darwin8-gcc"
--- libvpx/vpx_ports/x86_abi_support.asm.orig	2011-11-06 20:40:14.131254698 -0500
+++ libvpx/vpx_ports/x86_abi_support.asm	2011-11-06 20:40:16.975268804 -0500
@@ -88,6 +88,8 @@
 %define sym(x) x
 %elifidn __OUTPUT_FORMAT__,x64
 %define sym(x) x
+%elifidn __OUTPUT_FORMAT__,win64
+%define sym(x) x
 %else
 %define sym(x) _ %+ x
 %endif
