--- libvpx-v1.2.0/build/make/configure.sh.orig	2013-10-14 14:22:30.218929454 +0200
+++ libvpx-v1.2.0/build/make/configure.sh	2013-10-14 14:23:38.986930059 +0200
@@ -782,8 +782,8 @@
             check_add_asflags --defsym ARCHITECTURE=${arch_int}
             tune_cflags="-mtune="
             if [ ${tgt_isa} == "armv7" ]; then
-                check_add_cflags  -march=armv7-a -mfloat-abi=softfp
-                check_add_asflags -march=armv7-a -mfloat-abi=softfp
+                check_add_cflags  -march=armv7-a
+                check_add_asflags -march=armv7-a
 
                 if enabled neon
                 then
