--- libvpx-v1.1.0/build/make/configure.sh.orig	2012-05-09 01:14:00.000000000 +0200
+++ libvpx.new/build/make/configure.sh	2013-03-10 16:48:48.562884896 +0100
@@ -738,8 +737,8 @@
                     check_add_cflags -mfpu=neon #-ftree-vectorize
                     check_add_asflags -mfpu=neon
                 fi
-                check_add_cflags -march=armv7-a -mcpu=cortex-a8 -mfloat-abi=softfp
-                check_add_asflags -mcpu=cortex-a8 -mfloat-abi=softfp  #-march=armv7-a
+                check_add_cflags -march=armv7-a -mcpu=cortex-a8
+                check_add_asflags -mcpu=cortex-a8  #-march=armv7-a
             else
                 check_add_cflags -march=${tgt_isa}
                 check_add_asflags -march=${tgt_isa}
