--- libvpx-1.13.0/build/make/configure.sh	2023-02-07 20:33:51.000000000 -0500
+++ libvpx-1.13.0-new/build/make/configure.sh	2023-03-24 11:41:25.477967730 -0400
@@ -1119,7 +1119,7 @@
 
         darwin)
           if ! enabled external_build; then
-            XCRUN_FIND="xcrun --sdk iphoneos --find"
+            XCRUN_FIND="xcrun --find"
             CXX="$(${XCRUN_FIND} clang++)"
             CC="$(${XCRUN_FIND} clang)"
             AR="$(${XCRUN_FIND} ar)"
@@ -1138,7 +1138,7 @@
             add_cflags -arch ${tgt_isa}
             add_ldflags -arch ${tgt_isa}
 
-            alt_libc="$(show_darwin_sdk_path iphoneos)"
+            alt_libc="$(show_darwin_sdk_path macosx)"
             if [ -d "${alt_libc}" ]; then
               add_cflags -isysroot ${alt_libc}
             fi
