diff -ru libvpx.orig/configure libvpx/configure
--- libvpx.orig/configure	2011-07-04 19:54:18.000000000 +0300
+++ libvpx/configure	2011-07-04 19:56:19.000000000 +0300
@@ -117,6 +117,7 @@
 all_platforms="${all_platforms} x86_64-linux-gcc"
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
+all_platforms="${all_platforms} x86_64-win64-gcc"
 all_platforms="${all_platforms} x86_64-win64-vs8"
 all_platforms="${all_platforms} x86_64-win64-vs9"
 all_platforms="${all_platforms} universal-darwin8-gcc"
diff --git a/build/make/configure.sh b/build/make/configure.sh
From 75340193f720eceea5f52b00fcda83aef651b6e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C3=ABl=20Carr=C3=A9?= <funman@videolan.org>
Date: Mon, 7 Nov 2011 19:45:34 -0500
Subject: [PATCH] win64: use -f x64 in asflags instead of -f win64

Change-Id: Id1be1a91c7f0fd286c403489ff447837029ba17f
---
 build/make/configure.sh |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

index 1bb50af..1279f78 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -901,8 +901,12 @@ process_common_toolchain() {
         [ "${AS##*/}" = nasm ] && add_asflags -Ox
         AS_SFX=.asm
         case  ${tgt_os} in
-            win*)
-                add_asflags -f win${bits}
+            win32)
+                add_asflags -f win32
+                enabled debug && add_asflags -g cv8
+            ;;
+            win64)
+                add_asflags -f x64
                 enabled debug && add_asflags -g cv8
             ;;
             linux*|solaris*)
-- 
1.7.5.4

