From db37f83129cd777b588f28a50223d5158e50446e Mon Sep 17 00:00:00 2001
From: Marvin Scholz <epirat07@gmail.com>
Date: Mon, 10 Aug 2020 16:01:05 +0200
Subject: [PATCH 5/6] add patch to use macOS SDK on macOS aarch64

---
 build/make/configure.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 5638e3367..e7d57f342 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1176,7 +1176,7 @@ EOF
 
         darwin)
           if ! enabled external_build; then
-            XCRUN_FIND="xcrun --sdk iphoneos --find"
+            XCRUN_FIND="xcrun --find"
             CXX="$(${XCRUN_FIND} clang++)"
             CC="$(${XCRUN_FIND} clang)"
             AR="$(${XCRUN_FIND} ar)"
-- 
2.52.0.windows.1

