--- libvpx-1.8.0/build/make/configure.sh.orig	2019-05-09 08:39:28.062680326 -0400
+++ libvpx-1.8.0/build/make/configure.sh	2019-05-09 08:40:19.974415124 -0400
@@ -1031,9 +1031,14 @@
         android*)
           if [ -n "${sdk_path}" ]; then
             SDK_PATH=${sdk_path}
+            if [ ${tgt_isa} != "arm64" ]; then
+                tools_prefix=arm-linux-androideabi-
+            else
+                tools_prefix=aarch64-linux-android-
+            fi
             COMPILER_LOCATION=`find "${SDK_PATH}" \
-              -name "arm-linux-androideabi-gcc*" -print -quit`
-            TOOLCHAIN_PATH=${COMPILER_LOCATION%/*}/arm-linux-androideabi-
+              -name "${tools_prefix}gcc*" -print -quit`
+            TOOLCHAIN_PATH=${COMPILER_LOCATION%/*}/${tools_prefix}
             CC=clang
             CXX=clang++
             AR=llvm-ar
