--- libvpx/configure.orig	2011-08-15 17:02:45.000000000 -0400
+++ libvpx/configure	2011-11-20 21:29:52.241612178 -0500
@@ -107,6 +107,7 @@
 all_platforms="${all_platforms} x86-darwin8-icc"
 all_platforms="${all_platforms} x86-darwin9-gcc"
 all_platforms="${all_platforms} x86-darwin9-icc"
+all_platforms="${all_platforms} x86-darwin10-gcc"
 all_platforms="${all_platforms} x86-linux-gcc"
 all_platforms="${all_platforms} x86-linux-icc"
 all_platforms="${all_platforms} x86-solaris-gcc"
