--- libvpx-1.13.0/build/make/configure.sh	2023-02-07 20:33:51.000000000 -0500
+++ libvpx-1.13.0-new/build/make/configure.sh	2023-03-24 11:19:11.230089669 -0400
@@ -1166,12 +1166,6 @@
                 fi
                 ;;
             esac
-
-            if [ "$(show_darwin_sdk_major_version iphoneos)" -gt 8 ]; then
-              check_add_cflags -fembed-bitcode
-              check_add_asflags -fembed-bitcode
-              check_add_ldflags -fembed-bitcode
-            fi
           fi
 
           asm_conversion_cmd="${source_path_mk}/build/make/ads2gas_apple.pl"
