--- libvpx-1.4.0/build/make/configure.sh.orig	2015-05-05 16:03:00.504713016 +0200
+++ libvpx-1.4.0/build/make/configure.sh	2015-05-05 16:05:52.358003947 +0200
@@ -751,6 +751,12 @@
       ;;
   esac
 
+  if [ -d "${sdk_path}" ]; then
+      case "${sdk_path}" in
+         darwin*) osx_sdk_dir=${sdk_path} ;;
+      esac
+  fi
+
   case ${toolchain} in
     *-darwin8-*)
       add_cflags  "-mmacosx-version-min=10.4"
--- libvpx-1.4.0/build/make/configure.sh.orig	2015-05-05 16:06:31.502087047 +0200
+++ libvpx-1.4.0/build/make/configure.sh	2015-05-05 16:09:41.525727635 +0200
@@ -651,6 +651,12 @@
         ;;
     esac
 
+    if [ -d "${sdk_path}" ]; then
+        case "${sdk_path}" in
+           darwin*) osx_sdk_dir=${sdk_path} ;;
+        esac
+    fi
+
     # detect tgt_os
     case "$gcctarget" in
       *darwin8*)
@@ -751,12 +757,6 @@
       ;;
   esac
 
-  if [ -d "${sdk_path}" ]; then
-      case "${sdk_path}" in
-         darwin*) osx_sdk_dir=${sdk_path} ;;
-      esac
-  fi
-
   case ${toolchain} in
     *-darwin8-*)
       add_cflags  "-mmacosx-version-min=10.4"
