diff --git a/build/make/configure.sh b/build/make/configure.sh
index d05d0fa12..0ebf74039 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1129,12 +1129,6 @@ EOF
                 fi
                 ;;
             esac
-
-            if [ "$(show_darwin_sdk_major_version iphoneos)" -gt 8 ]; then
-              check_add_cflags -fembed-bitcode
-              check_add_asflags -fembed-bitcode
-              check_add_ldflags -fembed-bitcode
-            fi
           fi

           asm_conversion_cmd="${source_path}/build/make/ads2gas_apple.pl"
