From 8e5ef374c8ff3603a8ed4e1f48d4d5399e2e8044 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 25 Aug 2025 16:11:15 +0200
Subject: [PATCH] Link with gdi32 when using freetype in Windows

It's calling EnumFontFamiliesExW().
---
 src/meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/meson.build b/src/meson.build
index 07152660..255fbc47 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -65,6 +65,12 @@ if host_machine.system() == 'windows'
         'file/file_win32.c',
         'file/mount.c',
     )
+
+    if freetype_dependency.found()
+        # for EnumFontFamiliesExW
+        extra_dependencies += cc.find_library('gdi32')
+    endif
+
 elif host_machine.system() == 'darwin'
     libbluray_src += files(
         'file/dir_posix.c',
-- 
2.45.1.windows.1

