--- libmatroska-1.4.0/make/linux/Makefile.orig	2013-07-04 19:12:54.531784786 +0200
+++ libmatroska-1.4.0/make/linux/Makefile	2013-07-04 19:13:25.958887357 +0200
@@ -67,7 +67,7 @@
 
 objects_so:=$(objects)
 
-WARNINGFLAGS=-Wall -Wextra -Wno-unknown-pragmas -ansi -fno-gnu-keywords -D_GNU_SOURCE \
+WARNINGFLAGS=-Wall -Wextra -Wno-unknown-pragmas -fno-gnu-keywords -D_GNU_SOURCE \
 		-Wshadow
 COMPILEFLAGS=$(DEBUGFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(WARNINGFLAGS) $(INCLUDE)
 LINKFLAGS=-L. -L$(LIBEBML_LIB_DIR) $(LDFLAGS)
