--- configure.in	Fri Oct  4 13:45:02 2002
+++ configure.gpe.in	Sun Oct  6 22:24:16 2002
@@ -1508,5 +1508,23 @@
       PLUGINS="${PLUGINS} familiar"
     fi
+    GPE_PATH=""
+    AC_ARG_WITH(gpe,
+    [    --with-gpe=PATH gpe path (default search in \$PATH)],
+    [ if test "x${with_gpe}" != "xno"
+      then
+        GPE_PATH="${with_gpe}:${PATH}"
+      fi ])
+    ac_cv_gpe_headers=yes
+    AC_CHECK_HEADERS(gpe/init.h, , [
+      ac_cv_gpe_headers=no
+      echo "Cannot find gpe development headers."
+    ])
+    if test "x${ac_cv_gtk_headers}" = "xyes"
+    then
+      familiar_CFLAGS="${familiar_CFLAGS} -I${GPE_PATH}/libgpewidget"
+      familiar_LDFLAGS="${familiar_LDFLAGS} -L${GPE_PATH}/libgpewidget -lgpewidget"
+    fi
     CPPFLAGS="${save_CPPFLAGS}"
   fi
 fi
