--- familiar.c	Wed Oct  2 23:41:50 2002
+++ familiar-gpe.c	Sun Oct  6 22:01:23 2002
@@ -32,6 +32,7 @@
 #include <videolan/vlc.h>
 
 #include <gtk/gtk.h>
+#include <gpe/init.h>
 
 #include "stream_control.h"
 #include "input_ext-intf.h"
@@ -178,12 +179,9 @@
     int    i_args   = 1;
     int    i_dummy  = 0;
 
-    /* Initialize Gtk+ */
-    gtk_set_locale ();
-
-    /* gtk_init will register stuff with g_atexit, so we need to take
-     * the global lock if we want to be able to intercept the calls */
-    gtk_init( &i_args, &pp_args );
+   /* Initialize GPE interface */
+   if (gpe_application_init (&argc, &argv) == FALSE)
+        exit (1);	
 
     /* Create some useful widgets that will certainly be used */
 // FIXME: magic path
