--- plugins/sdl/vout_sdl.c	2001/12/03 16:18:37	1.66
+++ plugins/sdl/vout_sdl.c	2001/12/04 13:29:29
@@ -329,12 +329,16 @@
             break;
 
         case SDL_MOUSEBUTTONUP:
+#if 0
             switch( event.button.button )
             {
             case SDL_BUTTON_RIGHT:
                 p_main->p_intf->b_menu_change = 1;
                 break;
             }
+#else
+            //intf_ProcessKey( p_main->p_intf, SDLK_q );
+#endif
             break;
 
         case SDL_MOUSEBUTTONDOWN:

