--- SDL_image/Makefile.am.orig	2009-11-08 05:55:32.000000000 -0500
+++ SDL_image/Makefile.am	2011-11-22 22:17:14.234702385 -0500
@@ -47,7 +47,6 @@
 else
 libSDL_image_la_LDFLAGS = 	\
 	-no-undefined		\
-	-release $(LT_RELEASE)	\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libSDL_image_la_LIBADD = $(IMG_LIBS)
 endif
@@ -58,7 +57,7 @@
 %.o : %.rc
 	$(WINDRES) $< $@
 
-noinst_PROGRAMS = showimage
+#noinst_PROGRAMS = showimage
 
 showimage_LDADD = libSDL_image.la
 
--- SDL_image/Makefile.in.orig	2009-11-12 01:42:10.000000000 -0500
+++ SDL_image/Makefile.in	2011-11-22 22:17:10.714684927 -0500
@@ -41,7 +41,7 @@
 host_triplet = @host@
 @USE_VERSION_RC_FALSE@libSDL_image_la_DEPENDENCIES =  \
 @USE_VERSION_RC_FALSE@	$(am__DEPENDENCIES_1)
-noinst_PROGRAMS = showimage$(EXEEXT)
+#noinst_PROGRAMS = showimage$(EXEEXT)
 subdir = .
 DIST_COMMON = README $(am__configure_deps) \
 	$(libSDL_imageinclude_HEADERS) $(srcdir)/Makefile.am \
@@ -297,12 +297,10 @@
 
 @USE_VERSION_RC_FALSE@libSDL_image_la_LDFLAGS = \
 @USE_VERSION_RC_FALSE@	-no-undefined		\
-@USE_VERSION_RC_FALSE@	-release $(LT_RELEASE)	\
 @USE_VERSION_RC_FALSE@	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 @USE_VERSION_RC_TRUE@libSDL_image_la_LDFLAGS = \
 @USE_VERSION_RC_TRUE@	-no-undefined		\
-@USE_VERSION_RC_TRUE@	-release $(LT_RELEASE)	\
 @USE_VERSION_RC_TRUE@	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -Wl,version.o
 
 @USE_VERSION_RC_FALSE@libSDL_image_la_LIBADD = $(IMG_LIBS)
