From 0a07b23a280b3a80dc933edc0db3aafd2bb5626f Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 5 Nov 2015 14:16:51 +0100
Subject: [PATCH 3/3] config: use stdint.h and stdbool.h

---
 soxr-config.h.in | 31 ++-----------------------------
 1 file changed, 2 insertions(+), 29 deletions(-)

diff --git a/soxr-config.h.in b/soxr-config.h.in
index 227bcfd..539efe0 100644
--- a/soxr-config.h.in
+++ b/soxr-config.h.in
@@ -13,34 +13,7 @@
 #define WORDS_BIGENDIAN       @WORDS_BIGENDIAN@
 
 #include <limits.h>
-
-#undef bool
-#undef false
-#undef true
-#define bool int
-#define false 0
-#define true 1
-
-#undef int16_t
-#undef int32_t
-#undef int64_t
-#undef uint32_t
-#undef uint64_t
-#define int16_t short
-#if LONG_MAX > 2147483647L
-  #define int32_t int
-  #define int64_t long
-#elif LONG_MAX < 2147483647L
-#error this library requires that 'long int' has at least 32-bits
-#else
-  #define int32_t long
-  #if defined _MSC_VER
-    #define int64_t __int64
-  #else
-    #define int64_t long long
-  #endif
-#endif
-#define uint32_t unsigned int32_t
-#define uint64_t unsigned int64_t
+#include <stdint.h>
+#include <stdbool.h>
 
 #endif
-- 
2.1.4

